---
title: "Data download and preprocessing"
author: "Tao Wu"
date: "`r Sys.Date()`"
output:   
  rmdformats::readthedown:
    highlight: kate
    lightbox: false
    toc_depth: 3
    mathjax: true
---

```{r pre-process-setup, include=FALSE}
options(max.print = "75")
knitr::opts_chunk$set(echo = TRUE, comment = "#>", eval = TRUE, collapse = TRUE,cache = FALSE)
knitr::opts_knit$set(width = 75)
```

```{r lib,echo=TRUE,eval=TRUE,include=FALSE}
library(tidyverse)
library(ggpubr)
library(cowplot)
library(viridis)
library(NeoEnrichment)
library(parallel)
```
This part will describe how and where the data that this project used is downloaded and pre-processed.

## TCGA pancancer data download and clean

### Mutation data
Pre-compiled curated somatic mutations (called by MuTect2) for TCGA cohorts were downloaded from [Xena](https://xenabrowser.net/datapages/?dataset=GDC-PANCAN.mutect2_snv.tsv&host=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443) and only keep missense variants for following analysis

```{r mut_data,eval=FALSE}
mutect2 <- data.table::fread("~/useful_data/GDC-PANCAN.mutect2_snv.tsv") %>% 
  dplyr::filter(effect=="missense_variant" & filter=="PASS")
saveRDS(mutect2,file = "~/Immunoediting/data/pancancer_mutation.rds")
```
view this file:

```{r view_mut,eval=TRUE}
pancancer_mutation <- readRDS("~/Immunoediting/data/pancancer_mutation.rds")
DT::datatable(head(pancancer_mutation),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)

```

We also downloaded xena non-silent mutation data from [xena](https://xenabrowser.net/datapages/?cohort=TCGA%20Pan-Cancer%20(PANCAN)&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443). This file records mutations as 1 for non_slient mutation and 0 for slient mutation. 


### Copy number data
Absolute copy number data was download from [TCGA GDC Data Portal](https://portal.gdc.cancer.gov/repository?filters=%7B%22op%22%3A%22and%22%2C%22content%22%3A%5B%7B%22op%22%3A%22in%22%2C%22content%22%3A%7B%22field%22%3A%22files.data_type%22%2C%22value%22%3A%5B%22Allele-specific%20Copy%20Number%20Segment%22%5D%7D%7D%5D%7D), then we converted the aliquot id of those files to TCGA sample barcode using `aliquot.tsv` (check Biosepcimen botton in the GDC Cart to download),the code used to convert id is in `code/R/convert_aliquot_id.R`

```{r convert_id,eval=FALSE}
###merge all data
file <- dir("~/neo_dep/ssNEA/data/copy_number/")
dt <- data.frame(sample=NA,Chromosome=NA,Start=NA,End=NA,Copy_Number=NA)
dt <- dt[-1,]
for (i in file){
  tmp <- readRDS(paste("~/neo_dep/ssNEA/data/copy_number/",i,sep = "")) %>%
    select(sample_submitter_id,Chromosome,Start,End,Copy_Number) %>%
    rename(sample=sample_submitter_id)
  dt <- rbind(dt,tmp)
}
saveRDS(dt,file = "~/Immunoediting/data/pancancer_copy_number.rds")
```

view this file:

```{r view_cn,eval=TRUE}
pancancer_copy_number <- readRDS("~/Immunoediting/data/pancancer_copy_number.rds")
DT::datatable(head(pancancer_copy_number),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)

```

### Clinical and purity data
Clinical and tumor purity data was obtained from [TCGA PanCanAtlas Publications](https://gdc.cancer.gov/about-data/publications/pancanatlas)

```{r survial_purity,eval=FALSE}
###Clinical
survial <- data.table::fread("~/neo_dep/afteranalysis/survial/TCGA-CDR-survial.csv")
saveRDS(survial,file = "~/Immunoediting/data/pancancer_survial.rds")
###purity
purity <- data.table::fread(
  file = "~/neo_dep/envolution/TCGA_mastercalls.abs_tables_JSedit.fixed.txt")
saveRDS(purity,file = "~/Immunoediting/data/pancaner_purity.rds")
```

Clean the clinical data.

```{r clean_clinical,eval=FALSE}
pancancer_survial <- readRDS("~/Immunoediting/data/pancancer_survial.rds") %>%
  rename(
    sample = bcr_patient_barcode,
    age = age_at_initial_pathologic_diagnosis
  ) %>%
  mutate(gender = case_when(
    gender == "FEMALE" ~ "Female",
    gender == "MALE" ~ "Male",
    TRUE ~ NA_character_
  ), Tumor_stage = case_when(
    ajcc_pathologic_tumor_stage == "Stage 0" ~ "0",
    ajcc_pathologic_tumor_stage %in% c("Stage I", "Stage IA", "Stage IB") ~ "I",
    ajcc_pathologic_tumor_stage %in% c("Stage II", "Stage IIA", "Stage IIB", "Stage IIC") ~ "II",
    ajcc_pathologic_tumor_stage %in% c("Stage IIIA", "Stage IIIB", "Stage IIIC") ~ "III",
    ajcc_pathologic_tumor_stage %in% c("Stage IV", "Stage IVA", "Stage IVB", "Stage IVC") ~ "IV",
    ajcc_pathologic_tumor_stage == "Stage X" ~ "X",
    TRUE ~ NA_character_
  )) %>% 
  select(sample,age,gender,Tumor_stage,OS,OS.time)
saveRDS(pancancer_survial,file = "~/Immunoediting/data/pancancer_survial.rds")
```

View the clinical data:
```{r view_clinical,eval=TRUE}
pancancer_survial <- readRDS("~/Immunoediting/data/pancancer_survial.rds")
DT::datatable(head(pancancer_survial),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)

```


### HLA typing data
HLA typing data was obtained from [The immune landscape of cancer](The Immune Landscape of Cancer)
For samples which do not have mutation data in HLA typing data set (1168 samples) or samples which do not have HLA typing data in mutation data set (2404 samples), we download raw bam files from TCGA to call mutatioins and type HLA genotypes.

### Immune infiltration data

Immune infiltration estimation data for all TCGA tumors was downloaded from [TIMER2.0](http://timer.comp-genomics.org/) and we used CIBERSORT data, which has 22 immune cell types. In addition, the Leukocyte Fraction (LF) data reported in Vesteinn Thorsson et.al was downloaded from [The immune landscape of cancer](The Immune Landscape of Cancer).

```{r immune,eval=FALSE}
###Immune infiltration estimation
immune <- data.table::fread("~/useful_data/infiltration_estimation_for_tcga.csv") %>% 
  rename(sample=cell_type) %>% 
  select(sample,ends_with("CIBERSORT")) %>%
  filter(substr(sample,14,15) != "11") %>% ## checked the 14-15 of sample names
  mutate(
    Macrophage=`Monocyte_CIBERSORT`+`Macrophage M0_CIBERSORT`+`Macrophage M1_CIBERSORT`,
    Dendritic_cell=`Myeloid dendritic cell resting_CIBERSORT`+`Myeloid dendritic cell activated_CIBERSORT`,
    Mast_cell=`Mast cell activated_CIBERSORT`+`Mast cell resting_CIBERSORT`,
    Neutrophils=`Eosinophil_CIBERSORT`,
    Eosinophils=`Neutrophil_CIBERSORT`,
    T_cells_CD8=`T cell CD8+_CIBERSORT`,
    T_cells_CD4=`T cell CD4+ memory resting_CIBERSORT`+`T cell CD4+ memory activated_CIBERSORT`+
      `T cell CD4+ naive_CIBERSORT`,
    B_cells=`B cell naive_CIBERSORT`+`B cell memory_CIBERSORT`,
    NK_cells=`NK cell resting_CIBERSORT`+`NK cell activated_CIBERSORT`,
    Supress_cell=`Macrophage M2_CIBERSORT`+`T cell regulatory (Tregs)_CIBERSORT`
  ) %>% select(1,24:33)
saveRDS(immune,file = "~/Immunoediting/data/immue_estimate.rds")

###LF
lf <- data.table::fread("~/useful_data/immune_landscape.csv") %>% 
  select(1:32)
saveRDS(lf,file = "~/Immunoediting/data/immune_landscape.rds")
```

view those files:
```{r view_immune,eval=TRUE}
immue_estimate <- readRDS("~/Immunoediting/data/immue_estimate.rds")
DT::datatable(head(immue_estimate),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
```

```{r view_lf,eval=TRUE}
immune_landscape <- readRDS("~/Immunoediting/data/immune_landscape.rds")
DT::datatable(head(immune_landscape),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
```

In addition, we also obtain the data from [Wang S et.al, 2019 elife](10.7554/eLife.49020), and use APM score , Aggregate TIS (T cell infiltration score) and IIS (Immune cell infiltration score).


```{r wang_gsva,eval=FALSE}
load("~/Immunoediting/data/df_combine_gsva_clinical.rdata")
df.gsva <- df.gsva %>% 
  select(Project,Tumor_Sample_Barcode,APM,TIS,IIS)
saveRDS(df.gsva,file = "~/Immunoediting/data/wang_2019_elife.rds")
```

view the data:
```{r view_gsva,eval=TRUE}
wang_2019_elife <- readRDS("~/Immunoediting/data/wang_2019_elife.rds")
DT::datatable(head(wang_2019_elife),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
```


### RNA-seq data

The normalized gene-level RNA-seq data (TPM,Transcripts Per Million) for 33 TCGA cohorts were downloaded from [Xena](https://xenabrowser.net/).We chose GDC TCGA cohort and `HTSeq - FPKM` data, then we transfrom FPKM to TPM.(Those file is too large, so I don't push it to github)

```{r rna-seq,eval=FALSE}
fpkmToTpm <- function(fpkm)
{
  exp(log(fpkm) - log(sum(fpkm)) + log(1e6))
}

fpkm2tpm <- function(cancertype){
  ##read fpkm data
  f2 = paste("~/neo_dep/RNA-seq/TCGA_pancancer/",cancertype,"/",cancertype,".htseq_fpkm.tsv",sep = "")
  cancer_exp <- data.table::fread(
    file = f2,
    stringsAsFactors = FALSE,
    verbose = FALSE,
    data.table = TRUE,
    showProgress = TRUE,
    header = T,
    quote = ""
  ) 
  
  ##tranform fpkm to tpm
  #the data in the matrix is log2(fpkm+1)
  cancer_exp[,2:ncol(cancer_exp)] <- 2^(cancer_exp[,2:ncol(cancer_exp)])-1
  cancer_exp <- as.data.frame(cancer_exp)
  cancer_exp[,2:ncol(cancer_exp)] <- apply(cancer_exp[,2:ncol(cancer_exp)],2,fpkmToTpm)
  
  ##remove normal samples
  ##for get normal samples expression, see code/R/get_expmatrix_normal.R
  i = which(substr(colnames(cancer_exp),14,15)%in%c("01","02","03","04","05",
                                                    "06","07","08","09","10"))
  cancer_exp = cancer_exp %>% select(c(1,i))
  ## add gene symbol
  map = data.table::fread(
    file = "~/neo_dep/RNA-seq/TCGA_pancancer/acc/gencode.v22.annotation.gene.probeMap",
    stringsAsFactors = FALSE,
    verbose = FALSE,
    data.table = TRUE,
    showProgress = TRUE,
    header = T,
    quote = ""
  )
  cancer_exp <- left_join(cancer_exp,map %>% select(id,gene),by=c("Ensembl_ID"="id")) %>% 
    select(Ensembl_ID,gene,everything())
  save(cancer_exp,file = paste("~/neo_dep/RNA-seq/TCGA_pancancer/exp_matrix_tpm/",cancertype,"_exp.Rdata",sep = "")) 
}

cancer <- read.table("~/neo_dep/RNA-seq/TCGA_pancancer/cancer")
for (i in cancer$V1){
  fpkm2tpm(i)
}
```

Based on RNA-seq data, we can also calculate CYT values. Cytolytic activity (CYT) was calculated as the geometric mean of GZMA and PRF1 (as expressed in TPM, 0.01 offset)(Michael S. Rooney et.al, 2015 Cell)

The code used to calculate CYT can be found in `code/R/cal_CYT`

## TCGA pancancer data pre-process

This part describes how to calculate CCF, NES (based on CCF and copynumber) and find escape samples based on several immune escape mechanisms.

### Calculate ccf and add expression

In this project, we quantified Neoantigen Enrichment in the aspect of ccf and expression of neoantigen. So, we need calculate ccf and add expression information for each mutation.

The proportion of cancer cells carrying a particular mutation (cancer cell fraction, CCF) was calculated from the VAF of the mutation, sample purity (tumour content), and copy number (CN) of the mutation’s genomic locus as follows:
$$
CCF = \frac{VAF*CN}{purity}
$$
CCF values above 1 (arising from sequencing noise and copy-neutral loss-of-heterozygosity events) were assumed to be 1.

```{r cal_ccf,eval=FALSE}
###load data
pancancer_copy_number <- readRDS("~/Immunoediting/data/pancancer_copy_number.rds")
pancaner_purity <- readRDS("~/Immunoediting/data/pancaner_purity.rds")

pancancer_copy_number <- left_join(pancancer_copy_number,
                                   pancaner_purity %>%
                                     mutate(sample=substr(sample,1,16)) %>%
                                     select(sample,purity),
                                   by="sample")
###add copy number to mutation data
pancancer_mutation <- readRDS("~/Immunoediting/data/pancancer_mutation.rds")

dt <- pancancer_mutation %>%
  filter(effect=="missense_variant" & filter=="PASS") %>%
  mutate(chr_sample=paste(chrom,Sample_ID,sep = ","),
         pos=start)%>%
  select(chr_sample,pos,gene,dna_vaf) %>% rename(x=chr_sample)

copynumber_dt <- pancancer_copy_number %>%
  mutate(chr_sample=paste(Chromosome,sample,sep = ","))%>%
  select(chr_sample,Start,End,Copy_Number,purity) %>% rename(x=chr_sample)

dt <- setDT(dt)
copynumber_dt <- setDT(copynumber_dt)
overlap <- dt[copynumber_dt, .(x, pos=x.pos,Start,End,Copy_Number,purity,gene,dna_vaf),
              on=.(x, pos>=Start, pos<=End), nomatch=F,allow.cartesian=TRUE]
overlap <- overlap[,index:=paste(x,pos,sep = ",")]
overlap <- overlap %>%
  mutate(ccf=dna_vaf*Copy_Number/purity) %>%
  mutate(ccf_cn_assume=ifelse(ccf>1,1,ccf)) %>%
  separate(col = x,into = c("chromosome","sample"),sep = ",")
overlap$cancer_type <- NeoEnrichment::getcancer_type(overlap$sample,cores = 12,par = TRUE)
```

Add RNA expression to each mutation:

```{r add_exp,eval=FALSE}
cancer_type <- names(table(overlap$cancer_type))

with_tpm <- vector("list",33)
names(with_tpm) <- cancer_type

for(i in seq_along(with_tpm)){
  load(paste("~/neo_dep/RNA-seq/TCGA_pancancer/exp_matrix_tpm/",
             tolower(gsub("TCGA-","",names(with_tpm)[i])),"_exp.Rdata",sep = ""))
  a <- overlap %>% filter(cancer_type==names(with_tpm)[i])
  with_tpm[[i]] <- dplyr::left_join(
    a %>% mutate(sample=substr(sample, 1, 15)),
    cancer_exp %>%
      tidyr::pivot_longer(cols = dplyr::starts_with("TCGA"), names_to = "sample", values_to = "exp") %>%
      dplyr::mutate(sample = substr(sample, 1, 15)),
    by = c("sample", "gene")
  )
}
overlap_add_tpm <- bind_rows(with_tpm)

###remove dupliated gene
overlap_add_tpm <- overlap_add_tpm[!duplicated(overlap_add_tpm$index),]

saveRDS(overlap_add_tpm,file = "~/Immunoediting/data/pancancer_ccf_exp.rds")
```

### Calculate Neoantigen Enrichment score
We use pvactools to predict mutated peptides binding affinity, see methods for more details.

Firstly, we need add the results of neoantigen prediction (aggregated IC50) to mutation data.
```{r add_IC50,eval=FALSE}
###add IC50
pancancer_ccf_exp <- readRDS("~/Immunoediting/data/pancancer_ccf_exp.rds")
neoantigen <- readRDS("~/neo_dep/ssNEA/data/update_data.rds")
neoantigen <- neoantigen %>%
  mutate(index=paste(chr,Tumor_Barcode,Stop,sep = ","))
mut_with_IC50 <- left_join(neoantigen,
                         pancancer_ccf_exp %>%
                           select(index,ccf_cn_assume,cancer_type,exp),
                         by="index")
saveRDS(mut_with_IC50,file = "~/Immunoediting/data/pancancer_neoantigen_exp_ccf.rds")
```

Then we can calculate the Enrichment score for ccf and expression of neoantigens based on package `NeoEnrichment` (This package is developed for this project and can be installed by `devtools::install_github("wt12318/NeoEnrichment")`). The detail of method, please check the package document or our paper.

```{r cal_nes,eval=FALSE}
pancancer_neoantigen_exp_ccf <- readRDS("~/Immunoediting/data/pancancer_neoantigen_exp_ccf.rds")
pancancer_neoantigen_exp_ccf <- pancancer_neoantigen_exp_ccf %>% 
  rename(sample=Tumor_Barcode,chromosome=chr,position=Stop,MT_mean=MT_har_mean) %>% 
  as.data.frame(stringsAsFactors=FALSE)

###ccf
results_ccf <- vector("list",length = length(unique(pancancer_neoantigen_exp_ccf$sample)))
names(results_ccf) <- unique(pancancer_neoantigen_exp_ccf$sample)

cl <- makeCluster(getOption("cl.cores", 12),type="FORK")
set.seed(202011105)
results_ccf <- parSapply(cl=cl,names(results_ccf),
                         function(x){
                           NeoEnrichment::cales_t(pancancer_neoantigen_exp_ccf,x,
                                                  calp=TRUE,cal_type="ccf",mhc_type="I")
                         })
stopCluster(cl)

results_ccf <- Filter(function(x){length(x)>1},results_ccf)
pancancer_nes_ccf <- bind_rows(results_ccf,.id = "sample")
saveRDS(pancancer_nes_ccf,file = "~/Immunoediting/data/pancancer_nes_ccf.rds")

###exp
results_exp <- vector("list",length = length(unique(pancancer_neoantigen_exp_ccf$sample)))
names(results_exp) <- unique(pancancer_neoantigen_exp_ccf$sample)

cl <- makeCluster(getOption("cl.cores", 10),type="FORK")
set.seed(202011106)
results_exp <- parSapply(cl=cl,names(results_exp),
                         function(x){
                           NeoEnrichment::cales_t(pancancer_neoantigen_exp_ccf,x,
                                                  calp=TRUE,cal_type="exp",mhc_type="I")
                         })
stopCluster(cl)

results_exp <- Filter(function(x){length(x)>1},results_exp)
pancancer_nes_exp <- bind_rows(results_exp,.id = "sample")
saveRDS(pancancer_nes_exp,file = "~/Immunoediting/data/pancancer_nes_exp.rds")
```

View the results:
```{r view_nes,eval=TRUE}
pancancer_nes_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds")
pancancer_nes_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds")
DT::datatable(head(pancancer_nes_ccf),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
DT::datatable(head(pancancer_nes_exp),
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
```

### Process immune escape sample

We consider following immune escape mechanisms:

- Suppress the transcription/expression of genome alterations encoding high antigenicity (quantified by NES(EXP))
- Antigen presentation pathway gene mutation
- Up-regulate the expression of immune suppressive signaling, like PD-L1, CTLA-4

As for antigen presentation pathway gene, we use KEGG Antigen processing and presentation pathway: [hsa04612](https://www.kegg.jp/dbget-bin/www_bget?path:hsa04612). In addition, we included APM genes used in Rosenthal et.al and wang et.al. We then classified samples into “escape” if there was a non-slient mutation in one of these escape genes

```{r process_escape_TPM,eval=FALSE}
library("KEGGREST")
gs<-keggGet('hsa04612')
gene <- str_extract(gs[[1]]$GENE,".+; ") %>% na.omit()
gene <- gsub("; ","",gene)
another_apm <- c("CIITA","IRF1","PSME1","PSME2","PSME3","ERAP1","ERAP2","PSMA7",
         "HSPBP1","TAP1","TAP2","TAPBP","CALR","CANX","PDIA3","B2M","PSMB5","PSMB6","PSMB7",
         "PSMB8","PSMB9","PSMB10")
apm_gene <- union(gene,another_apm)
saveRDS(apm_gene,file = "~/Immunoediting/data/apm_gene.rds")

pancancer_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds")
pancancer_exp$adj_p <- p.adjust(pancancer_exp$p_value,method = "fdr")

pancancer_exp <- pancancer_exp %>% 
  mutate(exp_escape=ifelse(nes<0 & adj_p<0.25,"yes","no"))
#exp_escape_sample <- pancancer_exp[pancancer_exp$exp_escape=="yes","sample"] %>% substr(.,1,15)

###apm mutate sample
apm_gene <- readRDS("~/Immunoediting/data/apm_gene.rds")

##non-slient mutation
mut <- data.table::fread("~/PD_L1_ubiquitination/data/mc3.v0.2.8.PUBLIC.nonsilentGene.xena")
mut <- mut %>% 
  filter(sample %in% apm_gene) %>% 
  select(any_of(substr(pancancer_exp$sample,1,15)))

pancancer_exp$apm_mut <- ifelse(substr(pancancer_exp$sample,1,15) %in% colnames(mut),1,NA)

apm_mutation_sample <- apply(mut[,1:ncol(mut)],2,sum)
apm_mutation_sample <- names(apm_mutation_sample[apm_mutation_sample>=1])

pancancer_exp$apm_mut <- ifelse(is.na(pancancer_exp$apm_mut),NA,
                                ifelse(substr(pancancer_exp$sample,1,15) %in% apm_mutation_sample,"yes","no"))
saveRDS(pancancer_exp,file = "~/Immunoediting/data/pancancer_nes_exp.rds")
```
Immune checkpoint overexpression was assessed using RNA-seq data. Normal expression values (in transcripts per million mapped reads (TPM)) of PD-L1 and CTLA-4 were established for each cohort from the TCGA based on RNA-seq counts of the two proteins in solid tissue normal samples. Checkpoint overexpression was called if either PD-L1 or CTLA-4 expression in the tumor was higher than the mean plus two s.d. of normal expression :

```{r process_escape_CHECK,eval=FALSE}
###normal
cancer <- read.table("~/neo_dep/RNA-seq/TCGA_pancancer/cancer")
cancer <- cancer %>% 
  mutate(mean_pdl1=NA,
         mean_ctla4=NA,
         sd_pdl1=NA,
         sd_ctla4=NA) %>% rename(caner=V1)
file <- dir("~/neo_dep/RNA-seq/TCGA_pancancer/exp_tpm_mat_normal/")
for(i in file){
  a <- readRDS(paste("~/neo_dep/RNA-seq/TCGA_pancancer/exp_tpm_mat_normal/",i,sep = ""))
  a <- a %>% filter(gene %in% c("CD274","CTLA4"))
  b <- gsub("_exp.rds","",i)
  cancer[cancer$caner==b,"mean_pdl1"] <- mean(as.numeric(a[a$gene=="CD274",3:ncol(a)]))
  cancer[cancer$caner==b,"mean_ctla4"] <- mean(as.numeric(a[a$gene=="CTLA4",3:ncol(a)]))
  cancer[cancer$caner==b,"sd_pdl1"] <- sd(as.numeric(a[a$gene=="CD274",3:ncol(a)]))
  cancer[cancer$caner==b,"sd_ctla4"] <- sd(as.numeric(a[a$gene=="CTLA4",3:ncol(a)]))
}
saveRDS(cancer,file = "~/Immunoediting/data/normal_checkpoint.rds")

###cancer
a <- data.frame(CTLA4=NA,CD274=NA,sample=NA)
for (i in cancer$V1) {
  load(paste("~/neo_dep/RNA-seq/TCGA_pancancer/exp_matrix_tpm/",i,"_exp.Rdata",sep = ""))
  cancer_exp <- cancer_exp %>% filter(gene%in%c("CD274","CTLA4")) %>% select(-1)
  rownames(cancer_exp) <- cancer_exp$gene
  cancer_exp <- cancer_exp[,-1]
  cancer_exp <- as.data.frame(t(cancer_exp),stringsAsFactors = F)
  cancer_exp$sample <- rownames(cancer_exp)
  a <- rbind(a,cancer_exp)
}
a <- na.omit(a)
a$sample <- substr(a$sample,1,15)
a <- a[!duplicated(a$sample),]
saveRDS(a,file = "~/Immunoediting/data/cancer_checkpoint.rds")

###find overexpression
cancer_checkpoint_exp <- readRDS("~/Immunoediting/data/cancer_checkpoint.rds")
normal_checkpoint_exp <- readRDS("~/Immunoediting/data/normal_checkpoint.rds")

cancer_checkpoint_exp$cancer <- tolower(gsub("TCGA-","",getcancer_type(cancer_checkpoint_exp$sample)))
sample_checkpoint_exp <- left_join(cancer_checkpoint_exp,normal_checkpoint_exp,
                                   by=c("cancer"="caner"))
sample_checkpoint_exp <- sample_checkpoint_exp %>% 
  filter(!(is.na(mean_pdl1) & is.na(sd_pdl1)))

sample_checkpoint_exp <- sample_checkpoint_exp %>% 
  dplyr::mutate(CTLA4_overexpression=mapply(function(x,y,z){
    if(is.na(z)){
     return(ifelse(x>3*y,"yes","no")) 
    }else{
      return(ifelse(x>(y+2*z),"yes","no"))
    }
  },CTLA4,mean_ctla4,sd_ctla4),
  PDL1_overexpression=mapply(function(x,y,z){
    if(is.na(z)){
      return(ifelse(x>3*y,"yes","no")) 
    }else{
      return(ifelse(x>(y+2*z),"yes","no"))
    }
  },CD274,mean_pdl1,sd_pdl1))

sample_checkpoint_exp <- sample_checkpoint_exp %>% 
  select(sample,CTLA4_overexpression,PDL1_overexpression)
saveRDS(sample_checkpoint_exp,"~/Immunoediting/data/checkpoint_overexpression_sample.rds")
```
Then we can find escape samples:

```{r find_escape,eval=FALSE}
checkpoint_overexpression_sample <- readRDS("~/Immunoediting/data/checkpoint_overexpression_sample.rds")

checkpoint_escape_sample <- checkpoint_overexpression_sample[checkpoint_overexpression_sample$CTLA4_overexpression=="yes" | checkpoint_overexpression_sample$PDL1_overexpression=="yes","sample"]

pancancer_exp$checkpoint_overexpression <- ifelse(substr(pancancer_exp$sample,1,15) %in% checkpoint_overexpression_sample$sample,1,NA)

pancancer_exp$checkpoint_overexpression <- ifelse(is.na(pancancer_exp$checkpoint_overexpression),NA,
                                ifelse(substr(pancancer_exp$sample,1,15) %in% checkpoint_escape_sample,"yes","no"))

saveRDS(pancancer_exp,file = "~/Immunoediting/data/pancancer_nes_exp.rds")
```

## Immunotherapy data download and clean

For pan-cancer ICI objective response rates (ORR), we obtained data from wang et.al elife (2019).

```{r orr_ob,eval=FALSE}
ORR <- read.csv("~/neo_dep/ssNEA/data/orr.csv")
## transform cancer names to TCGA projects
ORR$cancer <- c("TCGA-ACC",NA,"TCGA-BRCA",
                "TCGA-CESC",NA,"TCGA-UCEC",
                "TCGA-STAD;TCGA-ESCA",
                "TCGA-TGCT","TCGA-GBM",
                "TCGA-HNSC","TCGA-LIHC",
                "TCGA-SKCM",NA,"TCGA-MESO",
                NA,"TCGA-LUAD;TCGA-LUSC",
                "TCGA-LUAD","TCGA-LUSC",
                "TCGA-OV","TCGA-PAAD",
                "TCGA-PRAD","TCGA-KIRC;TCGA-KIRP",
                "TCGA-SARC",NA,"TCGA-BLCA",
                "TCGA-UVM","TCGA-THYM")
saveRDS(ORR,file = "~/Immunoediting/data/Immunotherapy/ORR.rds")
```

We collected three cohorts of immunotherapy datasets for our analysis. `The Hugo et al. (2016) dataset` was related to anti-PD-1 therapy in metastatic melanoma. This dataset has 37 samples with WES data, 26 were also analyzed by RNA sequencing (RNA-seq). `The Riaz et al. (2017) dataset` was related to anti-PD-1 therapy in metastatic melanoma. This dataset has 56 samples with WES data, 40 with RNA-seq. `The David Liu et.al (2019) cohort` has patients with melanoma treated with anti-PD1 ICB, which 119 samples has WES data, 112 with RNA-seq.

Immune therapy response for patients was defined by RECIST v1.1(CR/PR/SD/PD), responding tumors were derived from patients who have complete or partial responses or stable disease (CR/PR) in response to anti-PD-1 therapy; non-responding tumors were derived from patients who had progressive disease (PD/SD)

The details of mutation calling, neoantigen prediction, copy number analysis and RNA_seq process can be found in Methods. Code can be found in `code/shell/ICI_*` 

## Immunotherapy data pre-process

Caculate NES(ccf) and NES(exp) for immunotherapy dataset:

```{r cal_es_immune,eval=FALSE}
all_sample_exp_ccf <- readRDS("~/Immunoediting/data/Immunotherapy/all_sample_exp_ccf.rds")

###ccf
all_sample_exp_ccf1 <- all_sample_exp_ccf %>% 
  filter(!is.na(ccf_cn_assume))
results_ccf <- vector("list",length = length(unique(all_sample_exp_ccf1$sample)))
names(results_ccf) <- unique(all_sample_exp_ccf1$sample)

cl <- makeCluster(getOption("cl.cores", 10),type="FORK")
set.seed(202011131)
results_ccf <- parSapply(cl=cl,names(results_ccf),
                         function(x){
                           NeoEnrichment::cales_t(all_sample_exp_ccf1,x,
                                                  calp=TRUE,cal_type="ccf",mhc_type="I")
                         })
stopCluster(cl)

results_ccf <- Filter(function(x){length(x)>1},results_ccf)
all_sample_nes_ccf <- bind_rows(results_ccf,.id = "sample")
saveRDS(all_sample_nes_ccf,file = "~/Immunoediting/data/Immunotherapy/all_sample_nes_ccf.rds")

###exp
all_sample_exp_ccf2 <- all_sample_exp_ccf %>% 
  filter(!is.na(exp))
results_exp <- vector("list",length = length(unique(all_sample_exp_ccf2$sample)))
names(results_exp) <- unique(all_sample_exp_ccf2$sample)

cl <- makeCluster(getOption("cl.cores", 10),type="FORK")
set.seed(202011132)
results_exp <- parSapply(cl=cl,names(results_exp),
                         function(x){
                           NeoEnrichment::cales_t(all_sample_exp_ccf2,x,
                                                  calp=TRUE,cal_type="exp",mhc_type="I")
                         })
stopCluster(cl)

results_exp <- Filter(function(x){length(x)>1},results_exp)
all_sample_nes_exp <- bind_rows(results_exp,.id = "sample")
saveRDS(all_sample_nes_exp,file = "~/Immunoediting/data/Immunotherapy/all_sample_nes_exp.rds")

```
