<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta name="generator" content="pandoc" />

        <meta name="author" content="Tao Wu" />
        <meta name="author" content="Xue-Song Liu (Corresponding author)" />
    
        <meta name="date" content="2021-10-25" />
    
    <title>Pan-cancer quantification of neoantigen-mediated immunoediting in cancer evolution</title>

        <script src="report_main_files/header-attrs-2.6/header-attrs.js"></script>
        <script src="report_main_files/jquery-1.12.4/jquery.min.js"></script>
        <meta name="viewport" content="width=device-width, initial-scale=1" />
        <link href="report_main_files/bootstrap-3.3.7/css/bootstrap.min.css" rel="stylesheet" />
        <script src="report_main_files/bootstrap-3.3.7/js/bootstrap.min.js"></script>
        <script src="report_main_files/jqueryui-1.11.4/jquery-ui.min.js"></script>
        <script src="report_main_files/navigation-1.1/tabsets.js"></script>
        <link href="report_main_files/magnific-popup-1.1.0/magnific-popup.css" rel="stylesheet" />
        <script src="report_main_files/magnific-popup-1.1.0/jquery.magnific-popup.min.js"></script>
        <link href="report_main_files/readthedown-0.1/readthedown.css" rel="stylesheet" />
        <link href="report_main_files/readthedown-0.1/readthedown_fonts_embed.css" rel="stylesheet" />
        <script src="report_main_files/readthedown-0.1/readthedown.js"></script>
        <script src="report_main_files/htmlwidgets-1.5.1/htmlwidgets.js"></script>
        <link href="report_main_files/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
        <script src="report_main_files/datatables-binding-0.16/datatables.js"></script>
        <link href="report_main_files/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
        <link href="report_main_files/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
        <script src="report_main_files/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
        <link href="report_main_files/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
        <script src="report_main_files/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
    
    
        <style type="text/css">code{white-space: pre;}</style>
    <style type="text/css">
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
          background-color: #ffffff;
          color: #a0a0a0;
        }
      pre.numberSource { margin-left: 3em; border-left: 1px solid #a0a0a0;  padding-left: 4px; }
      div.sourceCode
        { color: #1f1c1b; background-color: #ffffff; }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      code span { color: #1f1c1b; } /* Normal */
      code span.al { color: #bf0303; background-color: #f7e6e6; font-weight: bold; } /* Alert */
      code span.an { color: #ca60ca; } /* Annotation */
      code span.at { color: #0057ae; } /* Attribute */
      code span.bn { color: #b08000; } /* BaseN */
      code span.bu { color: #644a9b; font-weight: bold; } /* BuiltIn */
      code span.cf { color: #1f1c1b; font-weight: bold; } /* ControlFlow */
      code span.ch { color: #924c9d; } /* Char */
      code span.cn { color: #aa5500; } /* Constant */
      code span.co { color: #898887; } /* Comment */
      code span.cv { color: #0095ff; } /* CommentVar */
      code span.do { color: #607880; } /* Documentation */
      code span.dt { color: #0057ae; } /* DataType */
      code span.dv { color: #b08000; } /* DecVal */
      code span.er { color: #bf0303; text-decoration: underline; } /* Error */
      code span.ex { color: #0095ff; font-weight: bold; } /* Extension */
      code span.fl { color: #b08000; } /* Float */
      code span.fu { color: #644a9b; } /* Function */
      code span.im { color: #ff5500; } /* Import */
      code span.in { color: #b08000; } /* Information */
      code span.kw { color: #1f1c1b; font-weight: bold; } /* Keyword */
      code span.op { color: #1f1c1b; } /* Operator */
      code span.ot { color: #006e28; } /* Other */
      code span.pp { color: #006e28; } /* Preprocessor */
      code span.re { color: #0057ae; background-color: #e0e9f8; } /* RegionMarker */
      code span.sc { color: #3daee9; } /* SpecialChar */
      code span.ss { color: #ff5500; } /* SpecialString */
      code span.st { color: #bf0303; } /* String */
      code span.va { color: #0057ae; } /* Variable */
      code span.vs { color: #bf0303; } /* VerbatimString */
      code span.wa { color: #bf0303; } /* Warning */
    </style>
    
    
    <!-- tabsets -->
    <script>
      $(document).ready(function () {
	  window.buildTabsets("toc");
      });
      $(document).ready(function () {
	  $('.tabset-dropdown > .nav-tabs > li').click(function () {
	      $(this).parent().toggleClass('nav-tabs-open')
	  });
      });
    </script>

    <!-- code folding -->
    
    <!-- code download -->
    
    <!-- tabsets dropdown -->

    <style type="text/css">
      .tabset-dropdown > .nav-tabs {
	  display: inline-table;
	  max-height: 500px;
	  min-height: 44px;
	  overflow-y: auto;
	  background: white;
	  border: 1px solid #ddd;
	  border-radius: 4px;
      }
      
      .tabset-dropdown > .nav-tabs > li.active:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
	  content: "&#xe258;";
	  border: none;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
	  content: "";
	  font-family: 'Glyphicons Halflings';
	  display: inline-block;
	  padding: 10px;
	  border-right: 1px solid #ddd;
      }
      
      .tabset-dropdown > .nav-tabs > li.active {
	  display: block;
      }

      .tabset-dropdown > .nav-tabs > li.active a {
  	  padding: 0 15px !important;
      }

      .tabset-dropdown > .nav-tabs > li > a,
      .tabset-dropdown > .nav-tabs > li > a:focus,
      .tabset-dropdown > .nav-tabs > li > a:hover {
	  border: none;
	  display: inline-block;
	  border-radius: 4px;
	  background-color: transparent;
      }
      
      .tabset-dropdown > .nav-tabs.nav-tabs-open > li {
	  display: block;
	  float: none;
      }
      
      .tabset-dropdown > .nav-tabs > li {
	  display: none;
	  margin-left: 0 !important;
      }
    </style>
    
</head>

<body class="preload">

   	
         <!-- readthedown start -->   
   <div id="content" data-toggle="wy-nav-shift">
     <nav id="nav-top" role="navigation" aria-label="top navigation">
       <a role="button" href="#" data-toggle="wy-nav-top"><span class="glyphicon glyphicon-menu-hamburger"></span></a>
     </nav>
         
   
        
      <h1 class="title">Pan-cancer quantification of neoantigen-mediated immunoediting in cancer evolution</h1>
      
         <!-- readthedown authors -->
   <div id="sidebar">
    <h2><a href="#content">Pan-cancer quantification of neoantigen-mediated immunoediting in cancer evolution</a></h2>
    <div id="toc">
      <ul>
      <li><a href="#dependencies">Dependencies</a></li>
      <li><a href="#data-download-and-preprocessing">Data download and preprocessing</a>
      <ul>
      <li><a href="#tcga-pancancer-data-download-and-clean">TCGA pancancer data download and clean</a>
      <ul>
      <li><a href="#mutation-data">Mutation data</a></li>
      <li><a href="#expression-data">Expression data</a></li>
      <li><a href="#hla-typing-and-immune-cell-infiltration-data">HLA typing and immune cell infiltration data</a></li>
      </ul></li>
      <li><a href="#tcga-pancancer-data-processing">TCGA pancancer data processing</a>
      <ul>
      <li><a href="#hla-typing">HLA typing</a></li>
      <li><a href="#neoantigen-prediction">Neoantigen prediction</a></li>
      </ul></li>
      <li><a href="#immunotherapy-data-download-and-clean">Immunotherapy data download and clean</a></li>
      </ul></li>
      <li><a href="#tcga-pancancer-analysis">TCGA pancancer analysis</a>
      <ul>
      <li><a href="#the-existence-of-significant-immunoediting-signal">The existence of significant immunoediting signal</a></li>
      <li><a href="#neoantigen-enrichment-score-and-immune-negative-selection-strength-quantification">Neoantigen enrichment score and immune negative selection strength quantification</a></li>
      <li><a href="#pan-cancer-features-and-correlations-of-immunoediting-signal">Pan-cancer features and correlations of immunoediting signal</a></li>
      </ul></li>
      <li><a href="#immunotherapy-dataset-analysis">Immunotherapy dataset analysis</a>
      <ul>
      <li><a href="#quantified-immunoediting-elimination-signal-predicts-the-clinical-response-of-cancer-immunotherapy">Quantified immunoediting-elimination signal predicts the clinical response of cancer immunotherapy</a></li>
      </ul></li>
      <li><a href="#supplementary-analyses">Supplementary analyses</a></li>
      </ul>
    </div>
    <div id="postamble" data-toggle="wy-nav-shift" class="status">
                  <p class="author"><span class="glyphicon glyphicon-user"></span> Tao Wu</p>
                        <p class="author"><span class="glyphicon glyphicon-user"></span> Xue-Song Liu (Corresponding author)</p>
                        <p class="date"><span class="glyphicon glyphicon-calendar"></span> 2021-10-25</p>
          </div>
   </div>
     

   
      
   
<!-- Don't indent these lines or it will mess pre blocks indentation --> 
<div id="main">
<div id="dependencies" class="section level1">
<h1>Dependencies</h1>
<p>This project is depended on R software, R packages, shell bash, Julia software and some Julia packages :</p>
<ul>
<li><a href="https://github.com/wt12318/NeoEnrichment">NeoEnrichment</a> - do neoantigen enrichment for this project.</li>
<li><a href="https://www.tidyverse.org/">tidyverse</a> - tidy data</li>
<li><a href="https://github.com/Rdatatable/data.table">data.table</a> - read and tidy data</li>
<li><a href="https://github.com/therneau/survival">survival</a> - survival analysis</li>
<li><a href="https://github.com/kassambara/survminer">survminer</a> - plot survival fit</li>
<li><a href="https://cran.r-project.org/web/packages/DT/index.html">DT</a> - show data table as a table in html</li>
<li><a href="https://github.com/thomasp85/patchwork">patchwork</a> - arrange figures into complex compound figures</li>
<li><a href="https://github.com/PoisonAlien/maftools">maftools</a> - summarize, Analyze and Visualize MAF files</li>
<li><a href="https://github.com/jokergoo/ComplexHeatmap">ComplexHeatmap</a> - make heatmap</li>
<li><a href="https://github.com/ShixiangWang/ezcox">ezcox</a> - operate a batch of univariate or multivariate Cox models and return tidy result</li>
<li><a href="https://github.com/thomas-neitmann/ggcharts">ggcharts</a> - plot pyramid bar plot</li>
<li>parallel - Parallel Computing</li>
<li><a href="https://julialang.org/">Julia</a> - do simulation</li>
<li>knitr, rmdformats - used to compile this file</li>
</ul>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(NeoEnrichment)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parallel)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survminer)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(survival)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(maftools)</span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ezcox)</span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggcharts)</span></code></pre></div>
</div>
<div id="data-download-and-preprocessing" class="section level1">
<h1>Data download and preprocessing</h1>
<p>This part will describe how and where the data that this project used is downloaded and pre-processed.</p>
<div id="tcga-pancancer-data-download-and-clean" class="section level2">
<h2>TCGA pancancer data download and clean</h2>
<div id="mutation-data" class="section level3">
<h3>Mutation data</h3>
<p>Pre-compiled curated somatic mutations (called by MuTect2) for TCGA cohorts were downloaded from <a href="https://xenabrowser.net/datapages/?dataset=GDC-PANCAN.mutect2_snv.tsv&amp;host=https%3A%2F%2Fgdc.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">Xena</a> and only keep missense variants for following analysis</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>mutect2 <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/useful_data/GDC-PANCAN.mutect2_snv.tsv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(effect<span class="sc">==</span><span class="st">&quot;missense_variant&quot;</span> <span class="sc">&amp;</span> filter<span class="sc">==</span><span class="st">&quot;PASS&quot;</span>)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(mutect2,<span class="at">file =</span> <span class="st">&quot;../data/pancancer_mutation.rds&quot;</span>)</span></code></pre></div>
<p>view this file:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>pancancer_mutation <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/pancancer_mutation.rds&quot;</span>)</span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">head</span>(pancancer_mutation),</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">keys =</span> <span class="cn">TRUE</span>), <span class="at">rownames =</span> <span class="cn">FALSE</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div id="htmlwidget-fd65ec6f0842ab086aec" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-fd65ec6f0842ab086aec">{"x":{"filter":"none","data":[["TCGA-P6-A5OH-01A","TCGA-P6-A5OH-01A","TCGA-P6-A5OH-01A","TCGA-P6-A5OH-01A","TCGA-P6-A5OH-01A","TCGA-P6-A5OH-01A"],["WASF2","MACF1","NFIA","CADM3","DUSP27","RGSL1"],["chr1","chr1","chr1","chr1","chr1","chr1"],[27410171,39455094,61088434,159200836,167126702,182556049],[27410171,39455094,61088434,159200836,167126702,182556049],["G","T","C","G","G","G"],["A","A","T","A","C","C"],["p.A287V","p.H6923Q","p.L105F","p.D371N","p.G524A","p.E1075Q"],["missense_variant","missense_variant","missense_variant","missense_variant","missense_variant","missense_variant"],["PASS","PASS","PASS","PASS","PASS","PASS"],[0.25531914893617,0.0720720720720721,0.152671755725191,0.299479166666667,0.0555555555555556,0.2734375]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>Sample_ID<\/th>\n      <th>gene<\/th>\n      <th>chrom<\/th>\n      <th>start<\/th>\n      <th>end<\/th>\n      <th>ref<\/th>\n      <th>alt<\/th>\n      <th>Amino_Acid_Change<\/th>\n      <th>effect<\/th>\n      <th>filter<\/th>\n      <th>dna_vaf<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"keys":true,"columnDefs":[{"className":"dt-right","targets":[3,4,10]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>To predict neoantigen, mutation files were first transformed into single sample VCF format by maf2vcf tools (This Code can be found in <code>code/shell/TCGA_neoantigen</code> folder).</p>
<p>ABSOLUTE-annotated MAF file which contains cancer cell fraction (CCF) information of mutations was downloaded from GDC PanCanAtlas publications (<a href="https://gdc.cancer.gov/node/905/">TCGA_consolidated.abs_mafs_truncated.fixed.txt.gz</a>), and then we used liftover function from R package “rtracklayer” to convert the this hg37 genome coordinates file to hg38 coordinates:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a>absolute_maf <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/useful_data/TCGA_consolidated.abs_mafs_truncated.fixed.txt&quot;</span>)</span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>absolute_maf <span class="ot">&lt;-</span> absolute_maf <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">select</span>(Hugo_Symbol,Chromosome,Start_position,End_position,ccf_hat,sample,</span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>         Protein_Change,Variant_Classification,Variant_Type,Reference_Allele,</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>         Tumor_Seq_Allele2,ref,alt,purity)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GenomicRanges)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>absolute_maf<span class="sc">$</span>Chromosome <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>,absolute_maf<span class="sc">$</span>Chromosome)</span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>absolute_granges <span class="ot">&lt;-</span> <span class="fu">makeGRangesFromDataFrame</span>(absolute_maf,</span>
<span id="cb4-12"><a href="#cb4-12" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">keep.extra.columns=</span><span class="cn">TRUE</span>,</span>
<span id="cb4-13"><a href="#cb4-13" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">ignore.strand=</span><span class="cn">TRUE</span>,</span>
<span id="cb4-14"><a href="#cb4-14" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">seqinfo=</span><span class="cn">NULL</span>,</span>
<span id="cb4-15"><a href="#cb4-15" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">seqnames.field=</span><span class="st">&quot;Chromosome&quot;</span>,</span>
<span id="cb4-16"><a href="#cb4-16" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">start.field=</span><span class="st">&quot;Start_position&quot;</span>,</span>
<span id="cb4-17"><a href="#cb4-17" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">end.field=</span><span class="st">&quot;End_position&quot;</span>,</span>
<span id="cb4-18"><a href="#cb4-18" aria-hidden="true" tabindex="-1"></a>                                             <span class="at">starts.in.df.are.0based=</span><span class="cn">FALSE</span>)</span>
<span id="cb4-19"><a href="#cb4-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rtracklayer)</span>
<span id="cb4-20"><a href="#cb4-20" aria-hidden="true" tabindex="-1"></a>chainObject <span class="ot">&lt;-</span> <span class="fu">import.chain</span>(<span class="st">&quot;~/useful_data/hg19ToHg38.over.chain&quot;</span>)</span>
<span id="cb4-21"><a href="#cb4-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-22"><a href="#cb4-22" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(<span class="fu">liftOver</span>(absolute_granges, chainObject))</span>
<span id="cb4-23"><a href="#cb4-23" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results <span class="sc">%&gt;%</span></span>
<span id="cb4-24"><a href="#cb4-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(width<span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb4-25"><a href="#cb4-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-26"><a href="#cb4-26" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results <span class="sc">%&gt;%</span></span>
<span id="cb4-27"><a href="#cb4-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Variant_Classification<span class="sc">==</span><span class="st">&quot;Missense_Mutation&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-28"><a href="#cb4-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(<span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">16</span>),seqnames,start,Reference_Allele,Tumor_Seq_Allele2,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb4-29"><a href="#cb4-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,index,Hugo_Symbol,Protein_Change,Reference_Allele,Tumor_Seq_Allele2,purity,ccf_hat)</span>
<span id="cb4-30"><a href="#cb4-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(results,<span class="at">file =</span> <span class="st">&quot;../data/all_mut_mis_ccf.rds&quot;</span>)</span></code></pre></div>
<p>Driver mutation data was downloaded from Bailey M H et.al study (<a href="https://gdc.cancer.gov/about-data/publications/pancan-driver">Mutation.CTAT.3D.Scores.txt</a>). This study used three different categories of tools to find driver mutations: (1) tools distinguishing benign versus pathogenic mutations using sequence (CTAT population); (2) tools distinguishing driver versus passenger mutations using sequence (CTAT cancer); and (3) tools discovering statistically significant three-dimensional clusters of missense mutations (structure based). We keep mutations identified by ≥2 approaches as finial high confident driver mutations, including 2165 unique mutations:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>driver <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/useful_data/Mutation.CTAT.3D.Scores.txt&quot;</span>)</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>driver_mutations <span class="ot">&lt;-</span> driver <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Combined flag (strong evidence)</span><span class="st">`</span><span class="sc">==</span><span class="dv">1</span> <span class="sc">|</span> <span class="st">`</span><span class="at">Combined flag (weaker evidence)</span><span class="st">`</span> <span class="sc">==</span><span class="dv">1</span>)</span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(driver_mutations,<span class="at">file =</span> <span class="st">&quot;../data/driver_mutations.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="expression-data" class="section level3">
<h3>Expression data</h3>
<p>The normalized gene-level RNA-seq data (TPM, transcripts per million) for 31 TCGA cohorts were downloaded from Xena (<a href="https://xenabrowser.net/datapages/?dataset=tcga_RSEM_gene_tpm&amp;host=https%3A%2F%2Ftoil.xenahubs.net&amp;removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443">tcga_RSEM_gene_tpm</a>) and transformed from log-form to non-log form (This file is too large to store in Github):</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>TPM_log2 <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/useful_data/xena_RSEM_TPM/tcga_RSEM_gene_tpm.gz&quot;</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>mapping <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="st">&quot;~/useful_data/xena_RSEM_TPM/mapping_probe&quot;</span>)</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>mapping <span class="ot">&lt;-</span> mapping[,<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>TPM_log2_mapping <span class="ot">&lt;-</span> <span class="fu">left_join</span>(TPM_log2 <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">id=</span>sample),mapping) <span class="sc">%&gt;%</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(id,gene,<span class="fu">everything</span>())<span class="do">###log2(tpm+0.001)</span></span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>TPM_log2_mapping <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(TPM_log2_mapping)</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>tpm_trans <span class="ot">&lt;-</span> TPM_log2_mapping</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>tpm_trans[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(tpm_trans)] <span class="ot">&lt;-</span> <span class="fu">apply</span>(TPM_log2_mapping[,<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(TPM_log2_mapping)],<span class="dv">2</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>                                       <span class="cf">function</span>(x){(<span class="dv">2</span><span class="sc">^</span>x) <span class="sc">-</span> <span class="fl">0.001</span>})</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(tpm_trans,<span class="at">file =</span> <span class="st">&quot;~/useful_data/xena_RSEM_TPM/tpm_trans.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="hla-typing-and-immune-cell-infiltration-data" class="section level3">
<h3>HLA typing and immune cell infiltration data</h3>
<p>HLA typing data was downloaded from Thorsson et.al study and transformed to the format required by the neoantigen prediction tools:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>hla <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;~/useful_data/panCancer_hla.tsv&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>hla <span class="ot">&lt;-</span> hla <span class="sc">%&gt;%</span></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> V2,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;HLA-A_1&quot;</span>,<span class="st">&quot;HLA-A_2&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;HLA-B_1&quot;</span>,<span class="st">&quot;HLA-B_2&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>                             <span class="st">&quot;HLA-C_1&quot;</span>,<span class="st">&quot;HLA-C_2&quot;</span>),<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a><span class="do">###no stars</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>hla <span class="ot">&lt;-</span> <span class="fu">t</span>(base<span class="sc">::</span><span class="fu">apply</span>(hla,<span class="dv">1</span>,<span class="cf">function</span>(x){</span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  x[<span class="fu">which</span>(<span class="fu">duplicated</span>(x))] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">*&quot;</span>,<span class="st">&quot;&quot;</span>,x)</span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>  x</span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>})) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>()</span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(hla)[<span class="dv">1</span>] <span class="ot">&lt;-</span> <span class="st">&quot;Patient&quot;</span></span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(hla,<span class="at">file =</span> <span class="st">&quot;~/useful_data/TCGA_HLA_typing.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">quote =</span> F,<span class="at">col.names =</span> T,<span class="at">row.names =</span> F)</span></code></pre></div>
<p>Immune cell infiltration data for all TCGA tumors was downloaded from ImmuneCellAI study (Miao et al., 2020), which estimates the abundance of 24 immune cells comprised of 18 T-cell subtypes and 6 other immune cells. We can only download indivial cancer type data from the web server, so we need combine them all:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;~/test/immune_score/immuneCellAI/&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="fu">length</span>(files))</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(re)){</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/test/immune_score/immuneCellAI/&quot;</span>,files[i]),<span class="at">data.table =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">sample=</span>V1) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">16</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.&quot;</span>,<span class="st">&quot;-&quot;</span>,sample))</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> dt</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(results,<span class="at">file =</span> <span class="st">&quot;../data/pancancer_subtcells.rds&quot;</span>)</span></code></pre></div>
<p>We can view this file:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>pancancer_immune <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/pancancer_subtcells.rds&quot;</span>)</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>DT<span class="sc">::</span><span class="fu">datatable</span>(<span class="fu">head</span>(pancancer_immune),</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">options =</span> <span class="fu">list</span>(<span class="at">scrollX =</span> <span class="cn">TRUE</span>, <span class="at">keys =</span> <span class="cn">TRUE</span>), <span class="at">rownames =</span> <span class="cn">FALSE</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div id="htmlwidget-1fd2c59a30ae3a903aa7" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-1fd2c59a30ae3a903aa7">{"x":{"filter":"none","data":[["TCGA-OR-A5J1-01A","TCGA-OR-A5J2-01A","TCGA-OR-A5J3-01A","TCGA-OR-A5J5-01A","TCGA-OR-A5J6-01A","TCGA-OR-A5J7-01A"],[0,0,0.089,0.014,0,0.042],[0.115,0.114,0.087,0.218,0,0.069],[0.436,0.427,0.192,0.406,0.537,0.218],[0.035,0.08,0,0.003,0.105,0.02],[0.04,0.104,0.033,0.079,0.123,0.174],[0.048,0.025,0,0,0.11,0.013],[0.065,0.245,0,0,0.156,0.003],[0.158,0.152,0.069,0.004,0.148,0.025],[0.334,0.219,0,0.082,0.325,0],[0.057,0.044,0.253,0.114,0.203,0.112],[0.075,0.115,0,0.055,0.03,0.043],[0,0.087,0.061,0,0,0.041],[0.027,0.012,0,0,0.031,0],[0,0.143,0.07,0.105,0,0.043],[0.278,0.163,0.045,0.173,0.17,0.051],[0.15,0.207,0.149,0.088,0.224,0.13],[0.026,0.068,0.067,0.114,0.038,0.14],[0.104,0.095,0.076,0.148,0.172,0.086],[0.182,0.102,0.085,0.085,0.275,0.112],[0.162,0.003,0.146,0.254,0.377,0.179],[0.115,0.106,0.162,0.122,0.115,0.157],[0.089,0.07,0.13,0.151,0.09,0.06],[0.102,0.084,0.123,0.087,0.058,0.094],[0.201,0.211,0.046,0.156,0.206,0.048],[0.606,0.508,0.496,0.613,0.851,0.549]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>sample<\/th>\n      <th>CD4_naive<\/th>\n      <th>CD8_naive<\/th>\n      <th>Tc<\/th>\n      <th>Tex<\/th>\n      <th>Tr1<\/th>\n      <th>nTreg<\/th>\n      <th>iTreg<\/th>\n      <th>Th1<\/th>\n      <th>Th2<\/th>\n      <th>Th17<\/th>\n      <th>Tfh<\/th>\n      <th>Tcm<\/th>\n      <th>Tem<\/th>\n      <th>NKT<\/th>\n      <th>MAIT<\/th>\n      <th>DC<\/th>\n      <th>B_cell<\/th>\n      <th>Monocyte<\/th>\n      <th>Macrophage<\/th>\n      <th>NK<\/th>\n      <th>Neutrophil<\/th>\n      <th>Tgd<\/th>\n      <th>CD4_T<\/th>\n      <th>CD8_T<\/th>\n      <th>InfiltrationScore<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"scrollX":true,"keys":true,"columnDefs":[{"className":"dt-right","targets":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25]}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
</div>
</div>
<div id="tcga-pancancer-data-processing" class="section level2">
<h2>TCGA pancancer data processing</h2>
<p>This part describes HLA typing and neoantigen prediction of TCGA data.</p>
<div id="hla-typing" class="section level3">
<h3>HLA typing</h3>
<p>For samples which don’t have corresponding HLA infromation data in Thorsson et.al study, HLA genotyping was performed with Optitype (Szolek et al., 2014), using default parameters. This corresponding code can be found in <code>code/python/HLA_typing</code> (using snakemake pipeline tools).</p>
</div>
<div id="neoantigen-prediction" class="section level3">
<h3>Neoantigen prediction</h3>
<p>Mutect2 mutation files were first transformed into VCF format by maf2vcf tools, and we used NeoPredPipe to predict neoantigen (Schenck et al., 2019). Single-nucleotide variants leading to a single amino acid change are the focus of this study. From the output results, if the IC50 of a novel peptide is less than 50, the bind level is SB (strong binder, rank is less than 0.5%), and the expression level (TPM) is greater than 1, then this peptide is labeled as neoantigen. A mutation is considered neoantigenic if there is at least one peptide derived from the mutated site is predicted as neoantigen. The neoantigen prediction code using NeoPrePipe can be found in <code>code/shell/TCGA_neoantigen</code>. Then we add CCF and mRNA expression information to neoantigen prediction data:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;/public/slst/home/wutao2/TCGA_neopredpipe/batch_results&quot;</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;files&quot;</span>)</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>){</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> data.table<span class="sc">::</span><span class="fu">fread</span>(<span class="fu">paste</span>(files<span class="sc">$</span>V1[i],<span class="st">&quot;/batch_&quot;</span>,files<span class="sc">$</span>V1[i],<span class="st">&quot;.neoantigens.unfiltered.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>),<span class="at">fill=</span><span class="cn">TRUE</span>)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span><span class="sc">:</span><span class="dv">11</span>,<span class="dv">21</span><span class="sc">:</span><span class="dv">28</span>))</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(mut) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>,<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;position&quot;</span>,</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;ref&quot;</span>,<span class="st">&quot;alt&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;exp&quot;</span>,<span class="st">&quot;res_pos&quot;</span>,</span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;hla&quot;</span>,<span class="st">&quot;score_el&quot;</span>,<span class="st">&quot;rank_el&quot;</span>,<span class="st">&quot;score_ba&quot;</span>,</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;rank_ba&quot;</span>,<span class="st">&quot;IC50&quot;</span>,<span class="st">&quot;candidate&quot;</span>,<span class="st">&quot;bindlevel&quot;</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;novelty&quot;</span>)</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">novelty=</span><span class="fu">ifelse</span>(<span class="fu">is.na</span>(novelty),<span class="dv">0</span>,novelty)) <span class="sc">%&gt;%</span></span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(sample,chr,position,ref,alt,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>))</span>
<span id="cb10-17"><a href="#cb10-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">saveRDS</span>(mut,<span class="at">file =</span> <span class="fu">paste</span>(files<span class="sc">$</span>V1[i],<span class="st">&quot;.rds&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb10-18"><a href="#cb10-18" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-19"><a href="#cb10-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-20"><a href="#cb10-20" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;~/test/data/2021_03_31/&quot;</span>)</span>
<span id="cb10-21"><a href="#cb10-21" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">100</span>)</span>
<span id="cb10-22"><a href="#cb10-22" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-23"><a href="#cb10-23" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">100</span>){</span>
<span id="cb10-24"><a href="#cb10-24" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste</span>(<span class="st">&quot;~/test/data/2021_03_31/&quot;</span>,files[i],<span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb10-25"><a href="#cb10-25" aria-hidden="true" tabindex="-1"></a>  neo <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb10-26"><a href="#cb10-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(novelty <span class="sc">==</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-27"><a href="#cb10-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IC50<span class="sc">&lt;</span><span class="dv">50</span> <span class="sc">&amp;</span> bindlevel<span class="sc">==</span><span class="st">&quot;SB&quot;</span> <span class="sc">&amp;</span> novelty<span class="sc">==</span><span class="dv">1</span> ) <span class="sc">%&gt;%</span></span>
<span id="cb10-28"><a href="#cb10-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(index,<span class="at">.keep_all =</span> T)</span>
<span id="cb10-29"><a href="#cb10-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-30"><a href="#cb10-30" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb10-31"><a href="#cb10-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(index,<span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb10-32"><a href="#cb10-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">neo =</span> <span class="fu">ifelse</span>(index <span class="sc">%in%</span> neo<span class="sc">$</span>index , <span class="st">&quot;neo&quot;</span>,<span class="st">&quot;not_neo&quot;</span>))</span>
<span id="cb10-33"><a href="#cb10-33" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> mut</span>
<span id="cb10-34"><a href="#cb10-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb10-35"><a href="#cb10-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-36"><a href="#cb10-36" aria-hidden="true" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb10-37"><a href="#cb10-37" aria-hidden="true" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> all_mut <span class="sc">%&gt;%</span></span>
<span id="cb10-38"><a href="#cb10-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(sample,chr,position,ref,alt,neo,gene,exp)</span>
<span id="cb10-39"><a href="#cb10-39" aria-hidden="true" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> all_mut <span class="sc">%&gt;%</span></span>
<span id="cb10-40"><a href="#cb10-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">:.+&quot;</span>,<span class="st">&quot;&quot;</span>,gene))</span>
<span id="cb10-41"><a href="#cb10-41" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut,<span class="at">file =</span> <span class="st">&quot;~/test/data/2021_04_17/all_mut.rds&quot;</span>)</span>
<span id="cb10-42"><a href="#cb10-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-43"><a href="#cb10-43" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/useful_data/xena_RSEM_TPM/tpm_trans.rds&quot;</span>)</span>
<span id="cb10-44"><a href="#cb10-44" aria-hidden="true" tabindex="-1"></a>tpm <span class="ot">&lt;-</span> tpm[<span class="sc">!</span><span class="fu">duplicated</span>(tpm<span class="sc">$</span>gene),]</span>
<span id="cb10-45"><a href="#cb10-45" aria-hidden="true" tabindex="-1"></a>all_mut<span class="sc">$</span>tpm_exp <span class="ot">&lt;-</span> <span class="fu">mapply</span>(<span class="cf">function</span>(sample,gene){</span>
<span id="cb10-46"><a href="#cb10-46" aria-hidden="true" tabindex="-1"></a>  tpm[tpm<span class="sc">$</span>gene<span class="sc">==</span>gene,<span class="fu">substr</span>(sample,<span class="dv">1</span>,<span class="dv">15</span>)]</span>
<span id="cb10-47"><a href="#cb10-47" aria-hidden="true" tabindex="-1"></a>},all_mut<span class="sc">$</span>sample,all_mut<span class="sc">$</span>gene)</span>
<span id="cb10-48"><a href="#cb10-48" aria-hidden="true" tabindex="-1"></a>all_mut1 <span class="ot">&lt;-</span> all_mut <span class="sc">%&gt;%</span></span>
<span id="cb10-49"><a href="#cb10-49" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">lengths</span>(tpm_exp)<span class="sc">!=</span><span class="dv">0</span>)</span>
<span id="cb10-50"><a href="#cb10-50" aria-hidden="true" tabindex="-1"></a>all_mut1<span class="sc">$</span>tpm_exp <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(all_mut1<span class="sc">$</span>tpm_exp)</span>
<span id="cb10-51"><a href="#cb10-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-52"><a href="#cb10-52" aria-hidden="true" tabindex="-1"></a>all_mut1 <span class="ot">&lt;-</span> all_mut1 <span class="sc">%&gt;%</span></span>
<span id="cb10-53"><a href="#cb10-53" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo2=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span> <span class="sc">&amp;</span> tpm_exp<span class="sc">&gt;</span><span class="dv">1</span>,<span class="st">&quot;neo&quot;</span>,<span class="st">&quot;not_neo&quot;</span>))</span>
<span id="cb10-54"><a href="#cb10-54" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut1,<span class="at">file =</span> <span class="st">&quot;~/test/all_mut_tpm.rds&quot;</span>)</span>
<span id="cb10-55"><a href="#cb10-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-56"><a href="#cb10-56" aria-hidden="true" tabindex="-1"></a>all_mut1 <span class="ot">&lt;-</span> all_mut1 <span class="sc">%&gt;%</span></span>
<span id="cb10-57"><a href="#cb10-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>neo,<span class="sc">-</span>exp) <span class="sc">%&gt;%</span></span>
<span id="cb10-58"><a href="#cb10-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">neo=</span>neo2)</span>
<span id="cb10-59"><a href="#cb10-59" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut1,<span class="at">file =</span> <span class="st">&quot;../data/all_mut_tpm_not_filter.rds&quot;</span>)<span class="do">##this file is used to calculte EXP-ES</span></span>
<span id="cb10-60"><a href="#cb10-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-61"><a href="#cb10-61" aria-hidden="true" tabindex="-1"></a><span class="do">##add ccf</span></span>
<span id="cb10-62"><a href="#cb10-62" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/test/data/2021_04_05/all_mut_mis_ccf.rds&quot;</span>)</span>
<span id="cb10-63"><a href="#cb10-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-64"><a href="#cb10-64" aria-hidden="true" tabindex="-1"></a>all_mut_tpm <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/test/all_mut_tpm.rds&quot;</span>)</span>
<span id="cb10-65"><a href="#cb10-65" aria-hidden="true" tabindex="-1"></a><span class="co">#all_mut_tpm &lt;- readRDS(&quot;~/test/all_mut_tpm_not_filter.rds&quot;)</span></span>
<span id="cb10-66"><a href="#cb10-66" aria-hidden="true" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> all_mut_tpm <span class="sc">%&gt;%</span></span>
<span id="cb10-67"><a href="#cb10-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(sample,chr,position,ref,alt,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb10-68"><a href="#cb10-68" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="at">ref_allele=</span>ref,<span class="at">alt_allele=</span>alt)</span>
<span id="cb10-69"><a href="#cb10-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-70"><a href="#cb10-70" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb10-71"><a href="#cb10-71" aria-hidden="true" tabindex="-1"></a>  all_mut,</span>
<span id="cb10-72"><a href="#cb10-72" aria-hidden="true" tabindex="-1"></a>  results <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>sample),</span>
<span id="cb10-73"><a href="#cb10-73" aria-hidden="true" tabindex="-1"></a>  <span class="at">by=</span><span class="st">&quot;index&quot;</span></span>
<span id="cb10-74"><a href="#cb10-74" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb10-75"><a href="#cb10-75" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf[<span class="sc">!</span><span class="fu">duplicated</span>(all_mut_ccf<span class="sc">$</span>index),]</span>
<span id="cb10-76"><a href="#cb10-76" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut_ccf,<span class="at">file =</span> <span class="st">&quot;../data/all_mut_ccf_tpm.rds&quot;</span>)<span class="do">##This file is used to calculate CCF-ES</span></span></code></pre></div>
</div>
</div>
<div id="immunotherapy-data-download-and-clean" class="section level2">
<h2>Immunotherapy data download and clean</h2>
<p>We collected three cohorts of immunotherapy datasets for our analysis. <code>The Hugo et al. (2016) dataset</code> was related to anti-PD-1 therapy in metastatic melanoma. This dataset has 37 samples with WES data, 26 were also analyzed by RNA sequencing (RNA-seq). <code>The Riaz et al. (2017) dataset</code> was related to anti-PD-1 therapy in metastatic melanoma. This dataset has 56 samples with WES data, 40 with RNA-seq. <code>The David Liu et.al (2019) cohort</code> has patients with melanoma treated with anti-PD1 ICB, which 119 samples has WES data, 112 with RNA-seq.</p>
<p>Immune therapy response for patients was defined by RECIST v1.1(CR/PR/SD/PD), responding tumors were derived from patients who have complete or partial responses (CR/PR) in response to anti-PD-1 therapy; non-responding tumors were derived from patients who had progressive disease or stable disease (PD/SD).</p>
<p>The details of mutation calling, producing copy number segment file (using GATK4) and RNA_seq process can be found in Methods. Code can be found in <code>code/shell/ICI</code>. The HLA typing and neoantigen prediction was the same as the process in TCGA data as previously described. The code for caculating CCF by ABSOLUTE can be found in <code>code/R/ICI_ABSOLUTE.R</code></p>
<p>After got neoantiogen prediction and ABSOLUTE CCF information, we can combine these data for downstream analysis:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="do">####combine neoantigen prediction data</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;immunetherapy/&quot;</span>)</span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> files[<span class="fu">grepl</span>(<span class="st">&quot;neoantigens.unfiltered.txt&quot;</span>,files)]</span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">25</span>)</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">25</span>){</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> <span class="fu">read_table2</span>(<span class="fu">paste</span>(<span class="st">&quot;immunetherapy/&quot;</span>,files[i],<span class="at">sep =</span> <span class="st">&quot;&quot;</span>),<span class="at">col_names =</span> <span class="fu">paste0</span>(<span class="fu">rep</span>(<span class="st">&quot;V&quot;</span>,<span class="dv">27</span>),<span class="fu">c</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">27</span>)))</span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">3</span><span class="sc">:</span><span class="dv">10</span>,<span class="dv">20</span><span class="sc">:</span><span class="dv">27</span>))</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">colnames</span>(mut) <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">&quot;sample&quot;</span>,<span class="st">&quot;chr&quot;</span>,<span class="st">&quot;position&quot;</span>,</span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;ref&quot;</span>,<span class="st">&quot;alt&quot;</span>,<span class="st">&quot;gene&quot;</span>,<span class="st">&quot;exp&quot;</span>,<span class="st">&quot;res_pos&quot;</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;hla&quot;</span>,<span class="st">&quot;score_el&quot;</span>,<span class="st">&quot;rank_el&quot;</span>,<span class="st">&quot;score_ba&quot;</span>,</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;rank_ba&quot;</span>,<span class="st">&quot;IC50&quot;</span>,<span class="st">&quot;candidate&quot;</span>,<span class="st">&quot;bindlevel&quot;</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>                     <span class="st">&quot;novelty&quot;</span>)</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">novelty=</span><span class="fu">ifelse</span>(<span class="fu">is.na</span>(novelty),<span class="dv">0</span>,novelty)) <span class="sc">%&gt;%</span></span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(sample,chr,position,ref,alt,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>))</span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  neo <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(IC50<span class="sc">&lt;</span><span class="dv">50</span> <span class="sc">&amp;</span> bindlevel<span class="sc">==</span><span class="st">&quot;SB&quot;</span> <span class="sc">&amp;</span> novelty<span class="sc">==</span><span class="dv">1</span> <span class="sc">&amp;</span> exp<span class="sc">&gt;</span><span class="dv">1</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(index,<span class="at">.keep_all =</span> T)</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span></span>
<span id="cb11-26"><a href="#cb11-26" aria-hidden="true" tabindex="-1"></a>    <span class="fu">distinct</span>(index,<span class="at">.keep_all =</span> T) <span class="sc">%&gt;%</span></span>
<span id="cb11-27"><a href="#cb11-27" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">neo =</span> <span class="fu">ifelse</span>(index <span class="sc">%in%</span> neo<span class="sc">$</span>index , <span class="st">&quot;neo&quot;</span>,<span class="st">&quot;not_neo&quot;</span>))</span>
<span id="cb11-28"><a href="#cb11-28" aria-hidden="true" tabindex="-1"></a>  mut <span class="ot">&lt;-</span> mut <span class="sc">%&gt;%</span> </span>
<span id="cb11-29"><a href="#cb11-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample,chr,position,gene,exp,neo) <span class="sc">%&gt;%</span> </span>
<span id="cb11-30"><a href="#cb11-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">gene=</span><span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">:.+&quot;</span>,<span class="st">&quot;&quot;</span>,gene))</span>
<span id="cb11-31"><a href="#cb11-31" aria-hidden="true" tabindex="-1"></a>  mut<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="fu">gsub</span>(<span class="st">&quot;_.+&quot;</span>,<span class="st">&quot;&quot;</span>,files[i]),mut<span class="sc">$</span>sample,<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb11-32"><a href="#cb11-32" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> mut</span>
<span id="cb11-33"><a href="#cb11-33" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-34"><a href="#cb11-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-35"><a href="#cb11-35" aria-hidden="true" tabindex="-1"></a>all_mut <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb11-36"><a href="#cb11-36" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut,<span class="at">file =</span> <span class="st">&quot;../data/Immunotherapy/all_mut_ici.rds&quot;</span>)</span>
<span id="cb11-37"><a href="#cb11-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-38"><a href="#cb11-38" aria-hidden="true" tabindex="-1"></a><span class="do">###ccf in maf files of ABSOLUTE output</span></span>
<span id="cb11-39"><a href="#cb11-39" aria-hidden="true" tabindex="-1"></a><span class="do">##nadeem</span></span>
<span id="cb11-40"><a href="#cb11-40" aria-hidden="true" tabindex="-1"></a>sample_run <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;immunetherapy/nadeem_absolute/Nad_sample_run&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb11-41"><a href="#cb11-41" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;immunetherapy/nadeem_absolute/&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb11-42"><a href="#cb11-42" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">57</span>)</span>
<span id="cb11-43"><a href="#cb11-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-44"><a href="#cb11-44" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">57</span>){</span>
<span id="cb11-45"><a href="#cb11-45" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;immunetherapy/nadeem_absolute/&quot;</span>,files[i]),<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb11-46"><a href="#cb11-46" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample,Hugo_Symbol,Chromosome,Start_position,cancer_cell_frac,purity)</span>
<span id="cb11-47"><a href="#cb11-47" aria-hidden="true" tabindex="-1"></a>  test<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;nadeem&quot;</span>,<span class="fu">as.character</span>(sample_run[<span class="fu">which</span>(sample_run<span class="sc">$</span>V2<span class="sc">==</span><span class="fu">unique</span>(test<span class="sc">$</span>sample)),<span class="st">&quot;V1&quot;</span>]),</span>
<span id="cb11-48"><a href="#cb11-48" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb11-49"><a href="#cb11-49" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> test</span>
<span id="cb11-50"><a href="#cb11-50" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-51"><a href="#cb11-51" aria-hidden="true" tabindex="-1"></a>nadeem_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb11-52"><a href="#cb11-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-53"><a href="#cb11-53" aria-hidden="true" tabindex="-1"></a><span class="do">##willy</span></span>
<span id="cb11-54"><a href="#cb11-54" aria-hidden="true" tabindex="-1"></a>sample_run <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;immunetherapy/willy_absolute/clinical.txt&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb11-55"><a href="#cb11-55" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;immunetherapy/willy_absolute/&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb11-56"><a href="#cb11-56" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">38</span>)</span>
<span id="cb11-57"><a href="#cb11-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-58"><a href="#cb11-58" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">38</span>){</span>
<span id="cb11-59"><a href="#cb11-59" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;immunetherapy/willy_absolute/&quot;</span>,files[i]),<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb11-60"><a href="#cb11-60" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample,Hugo_Symbol,Chromosome,Start_position,cancer_cell_frac,purity)</span>
<span id="cb11-61"><a href="#cb11-61" aria-hidden="true" tabindex="-1"></a>  test<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;willy&quot;</span>,<span class="fu">as.character</span>(sample_run[<span class="fu">which</span>(sample_run<span class="sc">$</span>V2<span class="sc">==</span><span class="fu">unique</span>(test<span class="sc">$</span>sample)),<span class="st">&quot;V1&quot;</span>]),</span>
<span id="cb11-62"><a href="#cb11-62" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb11-63"><a href="#cb11-63" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> test</span>
<span id="cb11-64"><a href="#cb11-64" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-65"><a href="#cb11-65" aria-hidden="true" tabindex="-1"></a>willy_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb11-66"><a href="#cb11-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-67"><a href="#cb11-67" aria-hidden="true" tabindex="-1"></a><span class="do">###liu</span></span>
<span id="cb11-68"><a href="#cb11-68" aria-hidden="true" tabindex="-1"></a>sample_run <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;immunetherapy/liu_absolute/liu_sample_run&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb11-69"><a href="#cb11-69" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;immunetherapy/liu_absolute/&quot;</span>)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb11-70"><a href="#cb11-70" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">121</span>)</span>
<span id="cb11-71"><a href="#cb11-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-72"><a href="#cb11-72" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">121</span>){</span>
<span id="cb11-73"><a href="#cb11-73" aria-hidden="true" tabindex="-1"></a>  test <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="fu">paste0</span>(<span class="st">&quot;immunetherapy/liu_absolute/&quot;</span>,files[i]),<span class="at">sep =</span> <span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb11-74"><a href="#cb11-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(sample,Hugo_Symbol,Chromosome,Start_position,cancer_cell_frac,purity)</span>
<span id="cb11-75"><a href="#cb11-75" aria-hidden="true" tabindex="-1"></a>  test<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">paste</span>(<span class="st">&quot;liu&quot;</span>,<span class="fu">as.character</span>(sample_run[<span class="fu">which</span>(sample_run<span class="sc">$</span>V2<span class="sc">==</span><span class="fu">unique</span>(test<span class="sc">$</span>sample)),<span class="st">&quot;V1&quot;</span>]),</span>
<span id="cb11-76"><a href="#cb11-76" aria-hidden="true" tabindex="-1"></a>                       <span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb11-77"><a href="#cb11-77" aria-hidden="true" tabindex="-1"></a>  re[[i]] <span class="ot">&lt;-</span> test</span>
<span id="cb11-78"><a href="#cb11-78" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb11-79"><a href="#cb11-79" aria-hidden="true" tabindex="-1"></a>liu_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(re)</span>
<span id="cb11-80"><a href="#cb11-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-81"><a href="#cb11-81" aria-hidden="true" tabindex="-1"></a>all_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(liu_ccf,nadeem_ccf,willy_ccf)</span>
<span id="cb11-82"><a href="#cb11-82" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_ccf,<span class="at">file =</span> <span class="st">&quot;../data/Immunotherapy/all_ccf.rds&quot;</span>)</span>
<span id="cb11-83"><a href="#cb11-83" aria-hidden="true" tabindex="-1"></a><span class="do">###merge ccf and neoantigen data</span></span>
<span id="cb11-84"><a href="#cb11-84" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb11-85"><a href="#cb11-85" aria-hidden="true" tabindex="-1"></a>  all_mut_ici <span class="sc">%&gt;%</span> </span>
<span id="cb11-86"><a href="#cb11-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(sample,chr,position,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)),</span>
<span id="cb11-87"><a href="#cb11-87" aria-hidden="true" tabindex="-1"></a>  all_ccf <span class="sc">%&gt;%</span> </span>
<span id="cb11-88"><a href="#cb11-88" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">Chromosome=</span><span class="fu">paste0</span>(<span class="st">&quot;chr&quot;</span>,Chromosome)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-89"><a href="#cb11-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">index=</span><span class="fu">paste</span>(sample,Chromosome,Start_position,<span class="at">sep =</span> <span class="st">&quot;:&quot;</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb11-90"><a href="#cb11-90" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(index,cancer_cell_frac,purity),</span>
<span id="cb11-91"><a href="#cb11-91" aria-hidden="true" tabindex="-1"></a>  <span class="at">by=</span><span class="st">&quot;index&quot;</span></span>
<span id="cb11-92"><a href="#cb11-92" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb11-93"><a href="#cb11-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-94"><a href="#cb11-94" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(cancer_cell_frac))</span>
<span id="cb11-95"><a href="#cb11-95" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(all_mut_ccf,<span class="at">file =</span> <span class="st">&quot;../data/Immunotherapy/all_mut_ccf_ici.rds&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="tcga-pancancer-analysis" class="section level1">
<h1>TCGA pancancer analysis</h1>
<div id="the-existence-of-significant-immunoediting-signal" class="section level2">
<h2>The existence of significant immunoediting signal</h2>
<p>We investigate the status of this immunoediting signal with the new method developed in this study using TCGA pan-cancer dataset. The method detail showed in Method part of manuscript, and implemented in the package <code>NeoEnrichment</code> which can be found in <a href="https://github.com/wt12318/NeoEnrichment">Github</a></p>
<p>We filterd samples with at least 1 neoantigenic and 1 subclonal mutation (CCF&lt;0.6) and calculated CCF-NES value:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/all_mut_ccf_tpm.rds&quot;</span>)</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_hat) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>samples_has_subclonal <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ccf<span class="sc">&lt;</span><span class="fl">0.6</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample)</span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>all_mut_ccf  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_has_subclonal<span class="sc">$</span>sample) <span class="sc">%&gt;%</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">c_n=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>),<span class="at">c_m=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(c_n<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> c_m <span class="sc">&gt;=</span><span class="dv">1</span>) <span class="ot">-&gt;</span> summ</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> summ<span class="sc">$</span>sample)</span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a>cal_nes_warp <span class="ot">&lt;-</span> <span class="cf">function</span>(dt){</span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(dt<span class="sc">$</span>sample)))</span>
<span id="cb12-13"><a href="#cb12-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(dt<span class="sc">$</span>sample)</span>
<span id="cb12-14"><a href="#cb12-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-15"><a href="#cb12-15" aria-hidden="true" tabindex="-1"></a>  cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">16</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb12-16"><a href="#cb12-16" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb12-17"><a href="#cb12-17" aria-hidden="true" tabindex="-1"></a>                           <span class="cf">function</span>(x){</span>
<span id="cb12-18"><a href="#cb12-18" aria-hidden="true" tabindex="-1"></a>                             data <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">==</span> x)</span>
<span id="cb12-19"><a href="#cb12-19" aria-hidden="true" tabindex="-1"></a>                             a <span class="ot">&lt;-</span> NeoEnrichment<span class="sc">::</span><span class="fu">cal_nes_new_test</span>(<span class="at">dt =</span> data,</span>
<span id="cb12-20"><a href="#cb12-20" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">sample_counts =</span> <span class="dv">1000</span>,</span>
<span id="cb12-21"><a href="#cb12-21" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">need_p =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-22"><a href="#cb12-22" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(a)</span>
<span id="cb12-23"><a href="#cb12-23" aria-hidden="true" tabindex="-1"></a>                           },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb12-24"><a href="#cb12-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopCluster</span>(cl)</span>
<span id="cb12-25"><a href="#cb12-25" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">Filter</span>(<span class="cf">function</span>(x){<span class="fu">length</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>},results_ccf)</span>
<span id="cb12-26"><a href="#cb12-26" aria-hidden="true" tabindex="-1"></a>  pancancer_nes_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb12-27"><a href="#cb12-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pancancer_nes_ccf)</span>
<span id="cb12-28"><a href="#cb12-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb12-29"><a href="#cb12-29" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> neo_missense <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,neo,ccf) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ccf))</span>
<span id="cb12-30"><a href="#cb12-30" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(neo_missense)</span>
<span id="cb12-31"><a href="#cb12-31" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(neo_nes,<span class="at">file =</span> <span class="st">&quot;../data/neo_nes_ccf06_1.rds&quot;</span>)</span></code></pre></div>
<p>For each sample, we permutate the neoantigen labeling (ie. randomly select the same number of mutations as the actual neoantigen number in the selected sample and label them as neoantigenic mutations) and calculate ES value. For pan-cancer or cancer type dataset, we can obtain the same number of simulated samples and corresponding ES values, then calculate the median ES of these simulated samples. This process was repeated many times (usually 2000 times) to get the simulated distribution of median ES. The actual pan-cancer or cancer type median ES values are compared with this simulated ES distribution, and p values are then reported:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/public/slst/home/wutao2/cal_nes/ccf/neo_nes_ccf06_1.rds&quot;</span>)</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;/public/slst/home/wutao2/cal_nes/ccf/all_mut_ccf_tpm.rds&quot;</span>)</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_hat) <span class="sc">%&gt;%</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> neo_nes<span class="sc">$</span>sample)</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> neo_missense <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,neo,ccf) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ccf))</span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>cal_nes_warp <span class="ot">&lt;-</span> <span class="cf">function</span>(dt){</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(dt<span class="sc">$</span>sample)))</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(dt<span class="sc">$</span>sample)</span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a>  cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">23</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>                           <span class="cf">function</span>(x){</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>                             data <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">==</span> x)</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>                             a <span class="ot">&lt;-</span> NeoEnrichment<span class="sc">::</span><span class="fu">cal_nes_new_test</span>(<span class="at">dt =</span> data,</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">sample_counts =</span> <span class="dv">1000</span>,</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">need_p =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-20"><a href="#cb13-20" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(a)</span>
<span id="cb13-21"><a href="#cb13-21" aria-hidden="true" tabindex="-1"></a>                           },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb13-22"><a href="#cb13-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopCluster</span>(cl)</span>
<span id="cb13-23"><a href="#cb13-23" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">Filter</span>(<span class="cf">function</span>(x){<span class="fu">length</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>},results_ccf)</span>
<span id="cb13-24"><a href="#cb13-24" aria-hidden="true" tabindex="-1"></a>  pancancer_nes_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb13-25"><a href="#cb13-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pancancer_nes_ccf)</span>
<span id="cb13-26"><a href="#cb13-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-27"><a href="#cb13-27" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">2000</span>)</span>
<span id="cb13-28"><a href="#cb13-28" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">2000</span>){</span>
<span id="cb13-29"><a href="#cb13-29" aria-hidden="true" tabindex="-1"></a>  neo_missense_sim <span class="ot">&lt;-</span> neo_missense <span class="sc">%&gt;%</span></span>
<span id="cb13-30"><a href="#cb13-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb13-31"><a href="#cb13-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">neo_sim=</span><span class="fu">sample</span>(neo,<span class="fu">length</span>(neo)))</span>
<span id="cb13-32"><a href="#cb13-32" aria-hidden="true" tabindex="-1"></a>  neo_missense_sim <span class="ot">&lt;-</span> neo_missense_sim <span class="sc">%&gt;%</span></span>
<span id="cb13-33"><a href="#cb13-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(<span class="sc">-</span>neo) <span class="sc">%&gt;%</span></span>
<span id="cb13-34"><a href="#cb13-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">neo=</span>neo_sim)</span>
<span id="cb13-35"><a href="#cb13-35" aria-hidden="true" tabindex="-1"></a>  neo_nes_sim <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(neo_missense_sim)</span>
<span id="cb13-36"><a href="#cb13-36" aria-hidden="true" tabindex="-1"></a>  neo_nes_sim<span class="sc">$</span>sim_num <span class="ot">&lt;-</span> i</span>
<span id="cb13-37"><a href="#cb13-37" aria-hidden="true" tabindex="-1"></a>  res[[i]] <span class="ot">&lt;-</span> neo_nes_sim</span>
<span id="cb13-38"><a href="#cb13-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;Complete &quot;</span>,i,<span class="st">&quot; sim. &quot;</span>))</span>
<span id="cb13-39"><a href="#cb13-39" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb13-40"><a href="#cb13-40" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res,<span class="at">file =</span> <span class="st">&quot;/public/slst/home/wutao2/cal_nes/ccf/sim_2000_not_filter_driver.rds&quot;</span>)</span>
<span id="cb13-41"><a href="#cb13-41" aria-hidden="true" tabindex="-1"></a>sim_2000_not_filter <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;~/test/data/2021_09_26/sim_2000_not_filter.rds&quot;</span>)</span>
<span id="cb13-42"><a href="#cb13-42" aria-hidden="true" tabindex="-1"></a>sim_2000_not_filter <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(sim_2000_not_filter)</span>
<span id="cb13-43"><a href="#cb13-43" aria-hidden="true" tabindex="-1"></a>sim_2000_not_filter<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(sim_2000_not_filter<span class="sc">$</span>sample,</span>
<span id="cb13-44"><a href="#cb13-44" aria-hidden="true" tabindex="-1"></a>                                              <span class="at">parallel =</span> T,<span class="at">cores =</span> <span class="dv">20</span>)</span>
<span id="cb13-45"><a href="#cb13-45" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(sim_2000_not_filter,<span class="at">file =</span> <span class="st">&quot;../data/sim_2000_not_filter_all_ccf.rds&quot;</span>)<span class="do">##this file is too large to push to Github</span></span></code></pre></div>
<p>We can visualize the simulation using <code>WVPlots</code> package :</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1.rds&quot;</span>)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>sim_all <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../../tmp/sim_2000_not_filter_all_ccf.rds&quot;</span>)</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer,sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ2</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer&#39;. You can override using the `.groups` argument.</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>neo_nes<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(neo_nes<span class="sc">$</span>sample)</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span> </span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es))</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a><span class="co">#The following code can be used to plot cancer type simulation which showd in FigS3</span></span>
<span id="cb14-12"><a href="#cb14-12" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- vector(&quot;list&quot;,30)</span></span>
<span id="cb14-13"><a href="#cb14-13" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 1:30){</span></span>
<span id="cb14-14"><a href="#cb14-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt &lt;- sim_all %&gt;% filter(cancer==neo_nes_summ$cancer[i])</span></span>
<span id="cb14-15"><a href="#cb14-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt_summ &lt;- dt %&gt;% </span></span>
<span id="cb14-16"><a href="#cb14-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     group_by(sim_num) %&gt;%</span></span>
<span id="cb14-17"><a href="#cb14-17" aria-hidden="true" tabindex="-1"></a><span class="co">#     summarise(median_es=median(es))</span></span>
<span id="cb14-18"><a href="#cb14-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   p &lt;- WVPlots::ShadedDensity(frame = dt_summ, </span></span>
<span id="cb14-19"><a href="#cb14-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                               xvar = &quot;median_es&quot;,</span></span>
<span id="cb14-20"><a href="#cb14-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                               threshold = neo_nes_summ$median_es[i],</span></span>
<span id="cb14-21"><a href="#cb14-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                               title = neo_nes_summ$cancer[i],</span></span>
<span id="cb14-22"><a href="#cb14-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                               tail = &quot;left&quot;)</span></span>
<span id="cb14-23"><a href="#cb14-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$colour &lt;- &quot;red&quot;</span></span>
<span id="cb14-24"><a href="#cb14-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$size &lt;- 1</span></span>
<span id="cb14-25"><a href="#cb14-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[2]]$aes_params$fill &lt;- &quot;blue&quot;     #geom_ribbon</span></span>
<span id="cb14-26"><a href="#cb14-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$colour &lt;- &quot;black&quot; </span></span>
<span id="cb14-27"><a href="#cb14-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$size &lt;- 1</span></span>
<span id="cb14-28"><a href="#cb14-28" aria-hidden="true" tabindex="-1"></a><span class="co">#   p1 &lt;- p + labs(x=&quot;Simulation median es&quot;)+</span></span>
<span id="cb14-29"><a href="#cb14-29" aria-hidden="true" tabindex="-1"></a><span class="co">#     theme_prism()</span></span>
<span id="cb14-30"><a href="#cb14-30" aria-hidden="true" tabindex="-1"></a><span class="co">#   res[[i]] &lt;- p1</span></span>
<span id="cb14-31"><a href="#cb14-31" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb14-32"><a href="#cb14-32" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb14-33"><a href="#cb14-33" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_grid(plotlist = res)</span></span>
<span id="cb14-34"><a href="#cb14-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-35"><a href="#cb14-35" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb14-36"><a href="#cb14-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb14-37"><a href="#cb14-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ</span>
<span id="cb14-38"><a href="#cb14-38" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> WVPlots<span class="sc">::</span><span class="fu">ShadedDensity</span>(<span class="at">frame =</span> summ, </span>
<span id="cb14-39"><a href="#cb14-39" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xvar =</span> <span class="st">&quot;median_es&quot;</span>,</span>
<span id="cb14-40"><a href="#cb14-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">threshold =</span> <span class="fu">median</span>(neo_nes<span class="sc">$</span>es),</span>
<span id="cb14-41"><a href="#cb14-41" aria-hidden="true" tabindex="-1"></a>                            <span class="at">title =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb14-42"><a href="#cb14-42" aria-hidden="true" tabindex="-1"></a>                            <span class="at">tail =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb14-43"><a href="#cb14-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-44"><a href="#cb14-44" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb14-45"><a href="#cb14-45" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb14-46"><a href="#cb14-46" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">2</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>fill <span class="ot">&lt;-</span> <span class="st">&quot;blue&quot;</span>     <span class="co">#geom_ribbon</span></span>
<span id="cb14-47"><a href="#cb14-47" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;black&quot;</span> </span>
<span id="cb14-48"><a href="#cb14-48" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb14-49"><a href="#cb14-49" aria-hidden="true" tabindex="-1"></a><span class="co">#p$layers[[4]]$aes_params$label &lt;- &quot;Actual median ES&quot; #geom_text</span></span>
<span id="cb14-50"><a href="#cb14-50" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Simulation median es&quot;</span>)<span class="sc">+</span></span>
<span id="cb14-51"><a href="#cb14-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()</span>
<span id="cb14-52"><a href="#cb14-52" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/via1-1.png" width="2400" /></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="co"># neo_nes_summ &lt;- neo_nes_summ %&gt;% </span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   rowwise() %&gt;% </span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(p=mean(summ2$median_es[summ2$cancer==cancer] &lt;= median_es))</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(neo_nes_summ,file = &quot;neo_nes_summ_ccf_061_not_filter.rds&quot;)</span></span></code></pre></div>
<p>Then we can compare the median actual CCF-ES with median simulation CCF-ES in pan-cancer and cancer-type:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>get_f1 <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,pancancer_p,dt2,median_es){</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>dt,<span class="fu">aes</span>(<span class="at">x=</span><span class="dv">1</span>,<span class="at">y=</span>es))<span class="sc">+</span></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_violin</span>(<span class="at">alpha=</span><span class="fl">0.7</span>,<span class="at">width=</span><span class="fl">0.5</span>)<span class="sc">+</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>(<span class="at">width=</span><span class="fl">0.2</span>)<span class="sc">+</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a>    <span class="fu">labs</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;median es = &quot;</span>,<span class="fu">round</span>(median_es,<span class="at">digits =</span> <span class="dv">3</span>),<span class="st">&quot;</span><span class="sc">\n</span><span class="st"> n = &quot;</span>,<span class="fu">nrow</span>(dt)),<span class="at">size=</span><span class="dv">1</span>)<span class="sc">+</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.ticks.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.text.x =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">7</span>))<span class="sc">+</span></span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">annotate</span>(<span class="at">geom=</span><span class="st">&quot;text&quot;</span>, <span class="at">x=</span><span class="dv">1</span>, <span class="at">y=</span><span class="fl">1.1</span>, <span class="at">label=</span><span class="fu">paste0</span>(<span class="st">&quot;p = &quot;</span>,pancancer_p),</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>             <span class="at">color=</span><span class="st">&quot;red&quot;</span>,<span class="at">size=</span><span class="dv">4</span>)</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(dt<span class="sc">$</span>sample)</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>  dt <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">median_est=</span><span class="fu">median</span>(es),<span class="at">c=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(median_est) <span class="sc">%&gt;%</span></span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">label=</span><span class="fu">paste0</span>(cancer,<span class="st">&quot;</span><span class="sc">\n</span><span class="st">(n=&quot;</span>,c,<span class="st">&quot;)&quot;</span>))<span class="ot">-&gt;</span> summ1</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  summ1 <span class="ot">&lt;-</span> <span class="fu">left_join</span>(summ1,dt2 <span class="sc">%&gt;%</span> <span class="fu">select</span>(cancer,p))</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  summ1 <span class="ot">&lt;-</span> summ1 <span class="sc">%&gt;%</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">sig=</span><span class="fu">case_when</span>(</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>      p <span class="sc">&lt;</span> <span class="fl">0.05</span> <span class="sc">&amp;</span> p <span class="sc">&gt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">&quot;*&quot;</span>,</span>
<span id="cb16-21"><a href="#cb16-21" aria-hidden="true" tabindex="-1"></a>      p <span class="sc">&lt;</span> <span class="fl">0.01</span> <span class="sc">~</span> <span class="st">&quot;**&quot;</span>,</span>
<span id="cb16-22"><a href="#cb16-22" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;ns&quot;</span></span>
<span id="cb16-23"><a href="#cb16-23" aria-hidden="true" tabindex="-1"></a>    ))</span>
<span id="cb16-24"><a href="#cb16-24" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dt,summ1)</span>
<span id="cb16-25"><a href="#cb16-25" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>label <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>label,<span class="at">levels =</span> summ1<span class="sc">$</span>label)</span>
<span id="cb16-26"><a href="#cb16-26" aria-hidden="true" tabindex="-1"></a>  df2 <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(summ1), <span class="at">y =</span> <span class="fl">1.1</span>, <span class="at">family =</span> summ1<span class="sc">$</span>sig)</span>
<span id="cb16-27"><a href="#cb16-27" aria-hidden="true" tabindex="-1"></a>  p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>dt,<span class="fu">aes</span>(<span class="at">x=</span>label,<span class="at">y=</span>es))<span class="sc">+</span></span>
<span id="cb16-28"><a href="#cb16-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb16-29"><a href="#cb16-29" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb16-30"><a href="#cb16-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb16-31"><a href="#cb16-31" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb16-32"><a href="#cb16-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">6</span>))<span class="sc">+</span></span>
<span id="cb16-33"><a href="#cb16-33" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="at">data=</span>df2,<span class="fu">aes</span>(<span class="at">x=</span>x,<span class="at">y=</span>y,<span class="at">label =</span> family))<span class="sc">+</span></span>
<span id="cb16-34"><a href="#cb16-34" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_hline</span>(<span class="at">yintercept=</span><span class="dv">0</span>,</span>
<span id="cb16-35"><a href="#cb16-35" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">&quot;red&quot;</span>, <span class="at">size=</span><span class="dv">1</span>)</span>
<span id="cb16-36"><a href="#cb16-36" aria-hidden="true" tabindex="-1"></a>  p3 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> p2 <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">widths =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">8</span>))</span>
<span id="cb16-37"><a href="#cb16-37" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb16-38"><a href="#cb16-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-39"><a href="#cb16-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-40"><a href="#cb16-40" aria-hidden="true" tabindex="-1"></a>neo_nes4 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1.rds&quot;</span>)</span>
<span id="cb16-41"><a href="#cb16-41" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_summ_ccf_061_not_filter.rds&quot;</span>)</span>
<span id="cb16-42"><a href="#cb16-42" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">get_f1</span>(neo_nes4,<span class="at">pancancer_p =</span> <span class="fl">0.051</span>,<span class="at">dt2 =</span> neo_nes_summ,<span class="at">median_es =</span> <span class="fu">median</span>(neo_nes4<span class="sc">$</span>es))</span>
<span id="cb16-43"><a href="#cb16-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb16-44"><a href="#cb16-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb16-45"><a href="#cb16-45" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-6-1.png" width="2400" /></p>
<p>The observed ESCCF is -0.017 (p=0.051). In PAAD and LUAD, the observed ESCCF values are significant lower compared with the random simulations, suggesting the existence of immunoediting-elimination signal (Fig. 3a and Extended Data Fig.3). Since some neoantigenic mutations can be cancer drivers, which are known to undergo positive selection during the evolution of cancer. Neoantigens that happens to be cancer drivers are not undergoing immune based negative selection, probably because the driving force could override the negative selection, so we compared ES between the samples which neoantigenic and driver lie in the same gene with other samples:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_protein_change=</span><span class="fu">paste</span>(Hugo_Symbol,Protein_Change,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>driver_mutations <span class="ot">&lt;-</span> driver_mutations <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">gene_protein_change=</span><span class="fu">paste</span>(gene,protein_change,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">is_driver=</span><span class="fu">ifelse</span>(gene_protein_change <span class="sc">%in%</span> driver_mutations<span class="sc">$</span>gene_protein_change,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a><span class="fu">sum</span>(all_mut_ccf<span class="sc">$</span>is_driver<span class="sc">==</span><span class="st">&quot;yes&quot;</span> <span class="sc">&amp;</span> all_mut_ccf<span class="sc">$</span>neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>) <span class="sc">/</span> <span class="fu">sum</span>(all_mut_ccf<span class="sc">$</span>neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">inter_gene=</span><span class="fu">intersect</span>(Hugo_Symbol[neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>],</span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>                                 Hugo_Symbol[is_driver<span class="sc">==</span><span class="st">&quot;yes&quot;</span>])) <span class="ot">-&gt;</span> aaa</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sample_neo_index=</span><span class="fu">paste</span>(sample,neo,Hugo_Symbol,<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>))</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a>aaa <span class="ot">&lt;-</span> aaa <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sample_neo_index=</span><span class="fu">paste</span>(sample,<span class="st">&quot;yes&quot;</span>,inter_gene,<span class="at">sep =</span> <span class="st">&quot;,&quot;</span>))</span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">in_aaa =</span> <span class="fu">ifelse</span>(sample_neo_index <span class="sc">%in%</span> aaa<span class="sc">$</span>sample_neo_index,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">need_sample=</span><span class="fu">ifelse</span>(<span class="fu">any</span>(in_aaa<span class="sc">==</span><span class="st">&quot;yes&quot;</span>),<span class="st">&quot;no&quot;</span>,<span class="st">&quot;yes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(need_sample<span class="sc">==</span><span class="st">&quot;yes&quot;</span>) <span class="ot">-&gt;</span> summ2</span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>need_samples <span class="ot">&lt;-</span> <span class="fu">intersect</span>(samples_has_subclonal<span class="sc">$</span>sample,summ2<span class="sc">$</span>sample)</span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>all_mut_ccf  <span class="sc">%&gt;%</span></span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> need_samples) <span class="sc">%&gt;%</span></span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">c_n=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>),<span class="at">c_m=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(c_n<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> c_m <span class="sc">&gt;=</span><span class="dv">1</span>) <span class="ot">-&gt;</span> summ</span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> summ<span class="sc">$</span>sample)</span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> neo_missense <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,neo,ccf) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ccf))</span>
<span id="cb17-28"><a href="#cb17-28" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(neo_missense)</span>
<span id="cb17-29"><a href="#cb17-29" aria-hidden="true" tabindex="-1"></a><span class="fu">median</span>(neo_nes<span class="sc">$</span>es)</span>
<span id="cb17-30"><a href="#cb17-30" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(neo_nes,<span class="at">file =</span> <span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1.rds&quot;</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>neo_nes_remove_drivers <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples.rds&quot;</span>)</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>neo_nes2 <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(sample <span class="sc">%in%</span> neo_nes_remove_drivers<span class="sc">$</span>sample))</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">es=</span><span class="fu">c</span>(neo_nes_remove_drivers<span class="sc">$</span>es,neo_nes2<span class="sc">$</span>es),</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">type=</span><span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&quot;neo not in driver&quot;</span>,<span class="fu">nrow</span>(neo_nes_remove_drivers)),</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>                        <span class="fu">rep</span>(<span class="st">&quot;neo in driver&quot;</span>,<span class="fu">nrow</span>(neo_nes2))))</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>dt<span class="sc">$</span>type <span class="ot">&lt;-</span> <span class="fu">factor</span>(dt<span class="sc">$</span>type,<span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&quot;neo not in driver&quot;</span>,<span class="st">&quot;neo in driver&quot;</span>))</span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> dt,<span class="fu">aes</span>(<span class="at">x=</span>type,<span class="at">y=</span>es))<span class="sc">+</span></span>
<span id="cb18-10"><a href="#cb18-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb18-11"><a href="#cb18-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb18-12"><a href="#cb18-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb18-13"><a href="#cb18-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.x =</span> <span class="fu">element_blank</span>())</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-8-1.png" width="2400" /></p>
<p>We can see that samples which neoantigenic and driver mutation lie in the same gene have the higher ES than other samples. So we filtered these sample to remove the effect that driver mutation cound couse.</p>
<p>When samples with neoantigenic and driver mutations lied on same genes are not included, the observed median ESCCF is -0.023 (n=5420, P=0.006) (the simulation procedure is the same as above):</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>sim_all <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../../tmp/sim_2000_all_ccf.rds&quot;</span>)<span class="do">##this file is too large to push to Github</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer,sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ2</span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer&#39;. You can override using the `.groups` argument.</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>neo_nes_remove_drivers<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(neo_nes_remove_drivers<span class="sc">$</span>sample)</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> neo_nes_remove_drivers <span class="sc">%&gt;%</span> </span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es))</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co">#The following code can be used to plot cancer type simulation which showd in FigS5 </span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- vector(&quot;list&quot;,30)</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 1:30){</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt &lt;- sim_all %&gt;% filter(cancer==neo_nes_summ$cancer[i])</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt_summ &lt;- dt %&gt;% </span></span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a><span class="co">#     group_by(sim_num) %&gt;%</span></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     summarise(median_es=median(es))</span></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co">#   p &lt;- WVPlots::ShadedDensity(frame = dt_summ, </span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="co">#                               xvar = &quot;median_es&quot;,</span></span>
<span id="cb19-19"><a href="#cb19-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                               threshold = neo_nes_summ$median_es[i],</span></span>
<span id="cb19-20"><a href="#cb19-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                               title = neo_nes_summ$cancer[i],</span></span>
<span id="cb19-21"><a href="#cb19-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                               tail = &quot;left&quot;)</span></span>
<span id="cb19-22"><a href="#cb19-22" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$colour &lt;- &quot;red&quot;</span></span>
<span id="cb19-23"><a href="#cb19-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$size &lt;- 1</span></span>
<span id="cb19-24"><a href="#cb19-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[2]]$aes_params$fill &lt;- &quot;blue&quot;     #geom_ribbon</span></span>
<span id="cb19-25"><a href="#cb19-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$colour &lt;- &quot;black&quot; </span></span>
<span id="cb19-26"><a href="#cb19-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$size &lt;- 1</span></span>
<span id="cb19-27"><a href="#cb19-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   p1 &lt;- p + labs(x=&quot;Simulation median es&quot;)+</span></span>
<span id="cb19-28"><a href="#cb19-28" aria-hidden="true" tabindex="-1"></a><span class="co">#     theme_prism()</span></span>
<span id="cb19-29"><a href="#cb19-29" aria-hidden="true" tabindex="-1"></a><span class="co">#   res[[i]] &lt;- p1</span></span>
<span id="cb19-30"><a href="#cb19-30" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb19-31"><a href="#cb19-31" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb19-32"><a href="#cb19-32" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_grid(plotlist = res)</span></span>
<span id="cb19-33"><a href="#cb19-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-34"><a href="#cb19-34" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb19-35"><a href="#cb19-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb19-36"><a href="#cb19-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ</span>
<span id="cb19-37"><a href="#cb19-37" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> WVPlots<span class="sc">::</span><span class="fu">ShadedDensity</span>(<span class="at">frame =</span> summ, </span>
<span id="cb19-38"><a href="#cb19-38" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xvar =</span> <span class="st">&quot;median_es&quot;</span>,</span>
<span id="cb19-39"><a href="#cb19-39" aria-hidden="true" tabindex="-1"></a>                            <span class="at">threshold =</span> <span class="fu">median</span>(neo_nes_remove_drivers<span class="sc">$</span>es),</span>
<span id="cb19-40"><a href="#cb19-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">title =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb19-41"><a href="#cb19-41" aria-hidden="true" tabindex="-1"></a>                            <span class="at">tail =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb19-42"><a href="#cb19-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-43"><a href="#cb19-43" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb19-44"><a href="#cb19-44" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb19-45"><a href="#cb19-45" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">2</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>fill <span class="ot">&lt;-</span> <span class="st">&quot;blue&quot;</span>     <span class="co">#geom_ribbon</span></span>
<span id="cb19-46"><a href="#cb19-46" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;black&quot;</span> </span>
<span id="cb19-47"><a href="#cb19-47" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb19-48"><a href="#cb19-48" aria-hidden="true" tabindex="-1"></a><span class="co">#p$layers[[4]]$aes_params$label &lt;- &quot;Actual median ES&quot; #geom_text</span></span>
<span id="cb19-49"><a href="#cb19-49" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Simulation median es&quot;</span>)<span class="sc">+</span></span>
<span id="cb19-50"><a href="#cb19-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()</span>
<span id="cb19-51"><a href="#cb19-51" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-9-1.png" width="2400" /></p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a><span class="co"># neo_nes_summ &lt;- neo_nes_summ %&gt;% </span></span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   rowwise() %&gt;% </span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(p=mean(summ2$median_es[summ2$cancer==cancer] &lt;= median_es))</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(neo_nes_summ,file = &quot;neo_nes_summ_ccf_061.rds&quot;)</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>neo_nes3 <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples.rds&quot;</span>)</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_summ_ccf_061.rds&quot;</span>)</span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">get_f1</span>(neo_nes3,<span class="at">pancancer_p =</span> <span class="fl">0.006</span>,<span class="at">dt2 =</span> neo_nes_summ,<span class="at">median_es =</span> <span class="fu">median</span>(neo_nes3<span class="sc">$</span>es))</span>
<span id="cb20-10"><a href="#cb20-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb20-11"><a href="#cb20-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb20-12"><a href="#cb20-12" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-9-2.png" width="2400" /></p>
<p>Several cancer types including ACC, CHOL, UCEC, LUAD show significant low ESCCF values. This data demonstrates the existence of immunoediting-elimination signal in TCGA dataset.</p>
<p>We then calculated the EXP-ES and corresponding simulation median ES distribution:</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>all_mut_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/all_mut_tpm_not_filter.rds&quot;</span>)</span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>all_mut_exp <span class="ot">&lt;-</span> all_mut_exp <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,tpm_exp,neo) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">exp=</span>tpm_exp)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>all_mut_exp <span class="sc">%&gt;%</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_a=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;not_neo&quot;</span>),<span class="at">n_c=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(n_a<span class="sc">&gt;=</span><span class="dv">1</span>,n_c<span class="sc">&gt;=</span><span class="dv">1</span>) <span class="ot">-&gt;</span> summ</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>neo_exp <span class="ot">&lt;-</span> all_mut_exp <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> summ<span class="sc">$</span>sample) <span class="sc">%&gt;%</span></span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span>(sample <span class="sc">%in%</span> summ2<span class="sc">$</span>sample))</span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>cal_nes_warp <span class="ot">&lt;-</span> <span class="cf">function</span>(dt){</span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(dt<span class="sc">$</span>sample)))</span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(dt<span class="sc">$</span>sample)</span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">18</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a>                           <span class="cf">function</span>(x){</span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a>                             a <span class="ot">&lt;-</span> NeoEnrichment<span class="sc">::</span><span class="fu">cales_t</span>(<span class="at">data =</span> dt,<span class="at">barcode =</span> x,<span class="at">type =</span> <span class="st">&quot;II&quot;</span>,</span>
<span id="cb21-17"><a href="#cb21-17" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">calp =</span> <span class="cn">TRUE</span>,<span class="at">sample_counts =</span> <span class="dv">1000</span>,</span>
<span id="cb21-18"><a href="#cb21-18" aria-hidden="true" tabindex="-1"></a>                                                         <span class="at">cal_type =</span> <span class="st">&quot;exp&quot;</span>)</span>
<span id="cb21-19"><a href="#cb21-19" aria-hidden="true" tabindex="-1"></a>                             <span class="co">#df &lt;- data.frame(sample=x,es=a)</span></span>
<span id="cb21-20"><a href="#cb21-20" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(a)</span>
<span id="cb21-21"><a href="#cb21-21" aria-hidden="true" tabindex="-1"></a>                           },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb21-22"><a href="#cb21-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopCluster</span>(cl)</span>
<span id="cb21-23"><a href="#cb21-23" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">Filter</span>(<span class="cf">function</span>(x){<span class="fu">length</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>},results_ccf)</span>
<span id="cb21-24"><a href="#cb21-24" aria-hidden="true" tabindex="-1"></a>  pancancer_nes_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb21-25"><a href="#cb21-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pancancer_nes_ccf)</span>
<span id="cb21-26"><a href="#cb21-26" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb21-27"><a href="#cb21-27" aria-hidden="true" tabindex="-1"></a>nes_exp <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(neo_exp)</span>
<span id="cb21-28"><a href="#cb21-28" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(nes_exp,<span class="at">file =</span> <span class="st">&quot;../data/nes_exp.rds&quot;</span>)</span></code></pre></div>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>sim_all <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../../tmp/sim_2000_all_exp.rds&quot;</span>)<span class="do">##this file is too large to push to Github</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>nes_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/nes_exp.rds&quot;</span>)</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer,sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ2</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;cancer&#39;. You can override using the `.groups` argument.</span></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>nes_exp<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(nes_exp<span class="sc">$</span>sample)</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> nes_exp <span class="sc">%&gt;%</span> </span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es))</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a><span class="co">#The following code can be used to plot cancer type simulation which showd in FigS6 </span></span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a><span class="co">#res &lt;- vector(&quot;list&quot;,30)</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a><span class="co"># for (i in 1:30){</span></span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt &lt;- sim_all %&gt;% filter(cancer==neo_nes_summ$cancer[i])</span></span>
<span id="cb22-15"><a href="#cb22-15" aria-hidden="true" tabindex="-1"></a><span class="co">#   dt_summ &lt;- dt %&gt;% </span></span>
<span id="cb22-16"><a href="#cb22-16" aria-hidden="true" tabindex="-1"></a><span class="co">#     group_by(sim_num) %&gt;%</span></span>
<span id="cb22-17"><a href="#cb22-17" aria-hidden="true" tabindex="-1"></a><span class="co">#     summarise(median_es=median(es))</span></span>
<span id="cb22-18"><a href="#cb22-18" aria-hidden="true" tabindex="-1"></a><span class="co">#   p &lt;- WVPlots::ShadedDensity(frame = dt_summ, </span></span>
<span id="cb22-19"><a href="#cb22-19" aria-hidden="true" tabindex="-1"></a><span class="co">#                               xvar = &quot;median_es&quot;,</span></span>
<span id="cb22-20"><a href="#cb22-20" aria-hidden="true" tabindex="-1"></a><span class="co">#                               threshold = neo_nes_summ$median_es[i],</span></span>
<span id="cb22-21"><a href="#cb22-21" aria-hidden="true" tabindex="-1"></a><span class="co">#                               title = neo_nes_summ$cancer[i],</span></span>
<span id="cb22-22"><a href="#cb22-22" aria-hidden="true" tabindex="-1"></a><span class="co">#                               tail = &quot;left&quot;)</span></span>
<span id="cb22-23"><a href="#cb22-23" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$colour &lt;- &quot;red&quot;</span></span>
<span id="cb22-24"><a href="#cb22-24" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[1]]$aes_params$size &lt;- 1</span></span>
<span id="cb22-25"><a href="#cb22-25" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[2]]$aes_params$fill &lt;- &quot;blue&quot;     #geom_ribbon</span></span>
<span id="cb22-26"><a href="#cb22-26" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$colour &lt;- &quot;black&quot; </span></span>
<span id="cb22-27"><a href="#cb22-27" aria-hidden="true" tabindex="-1"></a><span class="co">#   p$layers[[3]]$aes_params$size &lt;- 1</span></span>
<span id="cb22-28"><a href="#cb22-28" aria-hidden="true" tabindex="-1"></a><span class="co">#   p1 &lt;- p + labs(x=&quot;Simulation median es&quot;)+</span></span>
<span id="cb22-29"><a href="#cb22-29" aria-hidden="true" tabindex="-1"></a><span class="co">#     theme_prism()</span></span>
<span id="cb22-30"><a href="#cb22-30" aria-hidden="true" tabindex="-1"></a><span class="co">#   res[[i]] &lt;- p1</span></span>
<span id="cb22-31"><a href="#cb22-31" aria-hidden="true" tabindex="-1"></a><span class="co"># }</span></span>
<span id="cb22-32"><a href="#cb22-32" aria-hidden="true" tabindex="-1"></a><span class="co"># </span></span>
<span id="cb22-33"><a href="#cb22-33" aria-hidden="true" tabindex="-1"></a><span class="co"># plot_grid(plotlist = res)</span></span>
<span id="cb22-34"><a href="#cb22-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-35"><a href="#cb22-35" aria-hidden="true" tabindex="-1"></a>sim_all <span class="sc">%&gt;%</span></span>
<span id="cb22-36"><a href="#cb22-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sim_num) <span class="sc">%&gt;%</span></span>
<span id="cb22-37"><a href="#cb22-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_es=</span><span class="fu">median</span>(es)) <span class="ot">-&gt;</span> summ</span>
<span id="cb22-38"><a href="#cb22-38" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> WVPlots<span class="sc">::</span><span class="fu">ShadedDensity</span>(<span class="at">frame =</span> summ, </span>
<span id="cb22-39"><a href="#cb22-39" aria-hidden="true" tabindex="-1"></a>                            <span class="at">xvar =</span> <span class="st">&quot;median_es&quot;</span>,</span>
<span id="cb22-40"><a href="#cb22-40" aria-hidden="true" tabindex="-1"></a>                            <span class="at">threshold =</span> <span class="fu">median</span>(nes_exp<span class="sc">$</span>es),</span>
<span id="cb22-41"><a href="#cb22-41" aria-hidden="true" tabindex="-1"></a>                            <span class="at">title =</span> <span class="st">&quot;&quot;</span>,</span>
<span id="cb22-42"><a href="#cb22-42" aria-hidden="true" tabindex="-1"></a>                            <span class="at">tail =</span> <span class="st">&quot;left&quot;</span>)</span>
<span id="cb22-43"><a href="#cb22-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-44"><a href="#cb22-44" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;red&quot;</span></span>
<span id="cb22-45"><a href="#cb22-45" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">1</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb22-46"><a href="#cb22-46" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">2</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>fill <span class="ot">&lt;-</span> <span class="st">&quot;blue&quot;</span>     <span class="co">#geom_ribbon</span></span>
<span id="cb22-47"><a href="#cb22-47" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>colour <span class="ot">&lt;-</span> <span class="st">&quot;black&quot;</span> </span>
<span id="cb22-48"><a href="#cb22-48" aria-hidden="true" tabindex="-1"></a>p<span class="sc">$</span>layers[[<span class="dv">3</span>]]<span class="sc">$</span>aes_params<span class="sc">$</span>size <span class="ot">&lt;-</span> <span class="dv">1</span></span>
<span id="cb22-49"><a href="#cb22-49" aria-hidden="true" tabindex="-1"></a><span class="co">#p$layers[[4]]$aes_params$label &lt;- &quot;Actual median ES&quot; #geom_text</span></span>
<span id="cb22-50"><a href="#cb22-50" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> p <span class="sc">+</span> <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Simulation median es&quot;</span>)<span class="sc">+</span></span>
<span id="cb22-51"><a href="#cb22-51" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()</span>
<span id="cb22-52"><a href="#cb22-52" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-11-1.png" width="2400" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="do">##save</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a><span class="co"># neo_nes_summ &lt;- neo_nes_summ %&gt;% </span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a><span class="co">#   rowwise() %&gt;% </span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co">#   mutate(p=mean(summ2$median_es[summ2$cancer==cancer] &lt;= median_es))</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># saveRDS(neo_nes_summ,file = &quot;neo_nes_summ_exp.rds&quot;)</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>neo_nes_summ <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_summ_exp.rds&quot;</span>)</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">get_f1</span>(nes_exp,<span class="at">pancancer_p =</span> <span class="st">&quot;&lt; 0.0005&quot;</span>,<span class="at">dt2 =</span> neo_nes_summ,<span class="at">median_es =</span> <span class="fu">median</span>(nes_exp<span class="sc">$</span>es))</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;cancer&quot;</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-11-2.png" width="2400" /></p>
<p>In majority of cancer types (including KICH, DLBC, CHOL, SARC, LUAD, PAAD, UCS, STAD, LGG, LUSC, HNSC, OV, UCEC, BRCA, LIHC, COAD), a significant low ESRNA values are observed . This study demonstrates that the immunoediting escape through down-regulating the expression of neoantigenic alteration is prevalent in human cancer.</p>
<p>When we compared the median cancer type CCF-ES and EXP-ES, we found the significant negative corrlation between this two type ES value:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="do">##these add p data can be calculated by change the parameter of need_p to TRUE</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>nes_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples_addp.rds&quot;</span>)</span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>nes_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/nes_exp_addp.rds&quot;</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>nes_ccf_exp <span class="ot">&lt;-</span> <span class="fu">inner_join</span>(</span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>  nes_ccf <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ccf_es=</span>es),</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  nes_exp <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">exp_es=</span>es)</span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;sample&quot;</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>nes_ccf_exp<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(nes_ccf_exp<span class="sc">$</span>sample)</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>nes_ccf_exp <span class="sc">%&gt;%</span></span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">median_ccf_es=</span><span class="fu">median</span>(ccf_es),</span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>            <span class="at">median_exp_es=</span><span class="fu">median</span>(exp_es)) <span class="ot">-&gt;</span> summ_ccf_exp</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(summ_ccf_exp<span class="sc">$</span>median_ccf_es,summ_ccf_exp<span class="sc">$</span>median_exp_es)</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Pearson&#39;s product-moment correlation</span></span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb24-19"><a href="#cb24-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  summ_ccf_exp$median_ccf_es and summ_ccf_exp$median_exp_es</span></span>
<span id="cb24-20"><a href="#cb24-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; t = -3.3231, df = 28, p-value = 0.002489</span></span>
<span id="cb24-21"><a href="#cb24-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true correlation is not equal to 0</span></span>
<span id="cb24-22"><a href="#cb24-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb24-23"><a href="#cb24-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  -0.7486576 -0.2122281</span></span>
<span id="cb24-24"><a href="#cb24-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb24-25"><a href="#cb24-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;        cor </span></span>
<span id="cb24-26"><a href="#cb24-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -0.5318337</span></span>
<span id="cb24-27"><a href="#cb24-27" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> summ_ccf_exp,<span class="fu">aes</span>(<span class="at">x=</span>median_ccf_es,<span class="at">y=</span>median_exp_es))<span class="sc">+</span></span>
<span id="cb24-28"><a href="#cb24-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">formula =</span> y <span class="sc">~</span> x) <span class="sc">+</span></span>
<span id="cb24-29"><a href="#cb24-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb24-30"><a href="#cb24-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">label.y =</span> <span class="fl">0.3</span>,</span>
<span id="cb24-31"><a href="#cb24-31" aria-hidden="true" tabindex="-1"></a>           <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste</span>(..r.label.., ..p.label.., <span class="at">sep =</span> <span class="st">&quot;~`,`~&quot;</span>)),</span>
<span id="cb24-32"><a href="#cb24-32" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb24-33"><a href="#cb24-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb24-34"><a href="#cb24-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Median CCF ES&quot;</span>,<span class="at">y=</span><span class="st">&quot;Median EXP ES&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-12-1.png" width="2400" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>nes_ccf_exp <span class="sc">%&gt;%</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">per_escape=</span><span class="fu">mean</span>(exp_es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p_value<span class="sc">&lt;</span><span class="fl">0.05</span>),</span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">median_es=</span><span class="fu">median</span>(ccf_es)) <span class="ot">-&gt;</span> summ</span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a><span class="fu">cor.test</span>(summ<span class="sc">$</span>per_escape,summ<span class="sc">$</span>median_es)</span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  Pearson&#39;s product-moment correlation</span></span>
<span id="cb25-10"><a href="#cb25-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb25-11"><a href="#cb25-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; data:  summ$per_escape and summ$median_es</span></span>
<span id="cb25-12"><a href="#cb25-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; t = 3.6094, df = 28, p-value = 0.001185</span></span>
<span id="cb25-13"><a href="#cb25-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; alternative hypothesis: true correlation is not equal to 0</span></span>
<span id="cb25-14"><a href="#cb25-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 95 percent confidence interval:</span></span>
<span id="cb25-15"><a href="#cb25-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;  0.2550042 0.7678830</span></span>
<span id="cb25-16"><a href="#cb25-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; sample estimates:</span></span>
<span id="cb25-17"><a href="#cb25-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       cor </span></span>
<span id="cb25-18"><a href="#cb25-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; 0.5635042</span></span>
<span id="cb25-19"><a href="#cb25-19" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> summ,<span class="fu">aes</span>(<span class="at">x=</span>median_es,<span class="at">y=</span>per_escape))<span class="sc">+</span></span>
<span id="cb25-20"><a href="#cb25-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> <span class="st">&quot;lm&quot;</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">color=</span><span class="st">&quot;black&quot;</span>, <span class="at">formula =</span> y <span class="sc">~</span> x) <span class="sc">+</span></span>
<span id="cb25-21"><a href="#cb25-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>()<span class="sc">+</span></span>
<span id="cb25-22"><a href="#cb25-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_cor</span>(<span class="at">label.y =</span> <span class="fl">0.2</span>,</span>
<span id="cb25-23"><a href="#cb25-23" aria-hidden="true" tabindex="-1"></a>           <span class="fu">aes</span>(<span class="at">label =</span> <span class="fu">paste</span>(..r.label.., ..p.label.., <span class="at">sep =</span> <span class="st">&quot;~`,`~&quot;</span>)),</span>
<span id="cb25-24"><a href="#cb25-24" aria-hidden="true" tabindex="-1"></a>           <span class="at">size=</span><span class="dv">4</span>)<span class="sc">+</span></span>
<span id="cb25-25"><a href="#cb25-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb25-26"><a href="#cb25-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Median CCF ES&quot;</span>,<span class="at">y=</span><span class="st">&quot;Escape sample percent&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-12-2.png" width="2400" /></p>
</div>
<div id="neoantigen-enrichment-score-and-immune-negative-selection-strength-quantification" class="section level2">
<h2>Neoantigen enrichment score and immune negative selection strength quantification</h2>
<p>Here we investigate the connections between neoantigen enrichment score (ESCCF) and immune negative selection strength s using a stochastic branching cancer evolution model as previously described (Lakatos et al., 2020). The detail of simulation process was described in the Method part and code can be found in <code>code/julia/simulation.jl</code>.</p>
<p>Mutations harbored in at least 5 cells out of 105 were collected at the end of each simulation and the CCF was calculated. To account for imperfect sequencing measurements, CCF values were computed via a simulated sequencing step introducing noise to these frequencies with the indicated read depth (see method):</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>cal_ccf <span class="ot">&lt;-</span> <span class="cf">function</span>(depth){</span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(ccf){</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map_dbl</span>(ccf,<span class="cf">function</span>(x,y){<span class="fu">rbinom</span>(<span class="dv">1</span>,y,x)<span class="sc">/</span>y},depth)</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>dep_ccf <span class="ot">&lt;-</span> <span class="fu">cal_ccf</span>(<span class="at">depth =</span> <span class="dv">200</span>)</span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;data/merge/&quot;</span>)</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="dv">50</span>)</span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>){</span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/test/data/merge/&quot;</span>,files[i]))</span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span></span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mut_neo=</span><span class="fu">ifelse</span>(mut_neo<span class="sc">==</span><span class="st">&quot;false&quot;</span>,<span class="st">&quot;no&quot;</span>,<span class="st">&quot;yes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">neo=</span>mut_neo)</span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a>  s<span class="sc">$</span>ccf_noise <span class="ot">&lt;-</span> <span class="fu">dep_ccf</span>(s<span class="sc">$</span>ccf)</span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ccf_noise<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>ccf) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_noise)</span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>  tt <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(s)<span class="do">##calculate ES</span></span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a>  res[[i]] <span class="ot">&lt;-</span> tt</span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;complete &quot;</span>,i))</span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a>nes_sim <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a>nes_sim <span class="ot">&lt;-</span> nes_sim <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">selection =</span> <span class="fu">substr</span>(sample,<span class="dv">3</span>,<span class="dv">6</span>))</span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(nes_sim,<span class="at">file =</span> <span class="st">&quot;../data/nes_sim_growth_200_addp.rds&quot;</span>)</span></code></pre></div>
<p>We can plot the distribution of ES of 100 simulated tumors (read depth 200×) across different selection strength and the precent of significant tumors (FDR adjust p value &lt;0.1):</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>nes_sim <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/nes_sim_growth_200_addp.rds&quot;</span>)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>nes_sim<span class="sc">$</span>s <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(nes_sim<span class="sc">$</span>selection)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>need_s <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">0.24</span>,<span class="fl">0.02</span>)</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>nes_sim_s <span class="ot">&lt;-</span> nes_sim <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s <span class="sc">%in%</span> need_s)</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>nes_sim_s,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;-&quot;</span>,selection),<span class="at">y=</span>es,<span class="at">fill=</span>selection))<span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()<span class="sc">+</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb27-9"><a href="#cb27-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept=</span>(<span class="sc">-</span><span class="fl">0.023</span>),</span>
<span id="cb27-10"><a href="#cb27-10" aria-hidden="true" tabindex="-1"></a>             <span class="at">color =</span> <span class="st">&quot;red&quot;</span>, <span class="at">size=</span><span class="dv">1</span>,<span class="at">linetype =</span> <span class="st">&quot;dashed&quot;</span>)<span class="sc">+</span></span>
<span id="cb27-11"><a href="#cb27-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Selection&quot;</span>,<span class="at">y=</span><span class="st">&quot;ES&quot;</span>)<span class="sc">+</span></span>
<span id="cb27-12"><a href="#cb27-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span>F)<span class="sc">+</span></span>
<span id="cb27-13"><a href="#cb27-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun=</span><span class="st">&quot;median&quot;</span>, <span class="at">geom=</span><span class="st">&quot;point&quot;</span>,<span class="at">color=</span><span class="st">&quot;yellow&quot;</span>,<span class="at">size=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-14-1.png" width="2400" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>nes_sim_s <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(selection) <span class="sc">%&gt;%</span></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">adj_p=</span><span class="fu">p.adjust</span>(p,<span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>),<span class="at">sig_per=</span><span class="fu">mean</span>(adj_p<span class="sc">&lt;</span><span class="fl">0.1</span>)) <span class="ot">-&gt;</span> summ</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;selection&#39;. You can override using the `.groups` argument.</span></span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>summ <span class="ot">&lt;-</span> summ <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(selection) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">per=</span><span class="fu">unique</span>(sig_per))</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>summ,<span class="fu">aes</span>(<span class="at">x=</span>selection,<span class="at">y=</span>per<span class="sc">*</span><span class="dv">100</span>))<span class="sc">+</span></span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Selection&quot;</span>,<span class="at">y=</span><span class="st">&quot;Percent of significant samples (%)&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-14-2.png" width="2400" /></p>
<p>ES-CCF show near linear correlation with s values. This analysis suggests that the quantified ES-CCF can be used to infer the immune selection strength in patient. The median ES-CCF in TCGA datasets is -0.023, suggesting a median immune negative selection strength s=-0.08.</p>
<p>Using this simulated data, we can also compare our method with other method. To compare the power of our method with method used in this study (Lakatos et al., 2020), we used two side K-S test to detect the difference between the VAF distribution of all mutations and neoantigen-associated mutations and reported K-S D statistic and corresponding p value:</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="st">&quot;data/merge/&quot;</span>)</span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="dv">50</span>)</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>){</span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/test/data/merge/&quot;</span>,files[i]))</span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mut_neo=</span><span class="fu">ifelse</span>(mut_neo<span class="sc">==</span><span class="st">&quot;false&quot;</span>,<span class="st">&quot;no&quot;</span>,<span class="st">&quot;yes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rename</span>(<span class="at">neo=</span>mut_neo)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a>  s<span class="sc">$</span>ccf_noise <span class="ot">&lt;-</span> <span class="fu">dep_ccf</span>(s<span class="sc">$</span>ccf)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(ccf_noise<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>ccf) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_noise)</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(s<span class="sc">$</span>sample)))</span>
<span id="cb29-15"><a href="#cb29-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(s<span class="sc">$</span>sample)</span>
<span id="cb29-16"><a href="#cb29-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-17"><a href="#cb29-17" aria-hidden="true" tabindex="-1"></a>  cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">20</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb29-18"><a href="#cb29-18" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb29-19"><a href="#cb29-19" aria-hidden="true" tabindex="-1"></a>                           <span class="cf">function</span>(x){</span>
<span id="cb29-20"><a href="#cb29-20" aria-hidden="true" tabindex="-1"></a>                             data <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">==</span> x)</span>
<span id="cb29-21"><a href="#cb29-21" aria-hidden="true" tabindex="-1"></a>                             neo <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>)</span>
<span id="cb29-22"><a href="#cb29-22" aria-hidden="true" tabindex="-1"></a>                             ks_t <span class="ot">&lt;-</span> <span class="fu">ks.test</span>(neo<span class="sc">$</span>ccf,data<span class="sc">$</span>ccf)</span>
<span id="cb29-23"><a href="#cb29-23" aria-hidden="true" tabindex="-1"></a>                             ttt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">D=</span>ks_t<span class="sc">$</span>statistic,<span class="at">p=</span>ks_t<span class="sc">$</span>p.value,<span class="at">sample=</span>x)</span>
<span id="cb29-24"><a href="#cb29-24" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(ttt)</span>
<span id="cb29-25"><a href="#cb29-25" aria-hidden="true" tabindex="-1"></a>                           },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb29-26"><a href="#cb29-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopCluster</span>(cl)</span>
<span id="cb29-27"><a href="#cb29-27" aria-hidden="true" tabindex="-1"></a>  res_ones <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb29-28"><a href="#cb29-28" aria-hidden="true" tabindex="-1"></a>  res[[i]] <span class="ot">&lt;-</span> res_ones</span>
<span id="cb29-29"><a href="#cb29-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="st">&quot;complete &quot;</span>,i))</span>
<span id="cb29-30"><a href="#cb29-30" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb29-31"><a href="#cb29-31" aria-hidden="true" tabindex="-1"></a>res_ks <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(res)</span>
<span id="cb29-32"><a href="#cb29-32" aria-hidden="true" tabindex="-1"></a>res_ks<span class="sc">$</span>selection <span class="ot">&lt;-</span> <span class="fu">substr</span>(res_ks<span class="sc">$</span>sample,<span class="dv">3</span>,<span class="dv">6</span>)</span>
<span id="cb29-33"><a href="#cb29-33" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(res_ks,<span class="at">file =</span> <span class="st">&quot;../data/ks_sim_depth200.rds&quot;</span>)</span></code></pre></div>
<p>Plot the distribution of KS D statics of 100 simulated tumors (read depth 200×) across different selection strength and the precent of significant tumors (FDR adjust p value &lt;0.1)</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>res_ks <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/ks_sim_depth200.rds&quot;</span>)</span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>res_ks<span class="sc">$</span>s <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(res_ks<span class="sc">$</span>selection)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>need_s <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">0.24</span>,<span class="fl">0.02</span>)</span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>res_ks_s <span class="ot">&lt;-</span> res_ks <span class="sc">%&gt;%</span> <span class="fu">filter</span>(s <span class="sc">%in%</span> need_s)</span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>res_ks_s,<span class="fu">aes</span>(<span class="at">x=</span><span class="fu">paste0</span>(<span class="st">&quot;-&quot;</span>,selection),<span class="at">y=</span>D,<span class="at">fill=</span>selection))<span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_violin</span>()<span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb30-9"><a href="#cb30-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Selection&quot;</span>,<span class="at">y=</span><span class="st">&quot;K-S statistic&quot;</span>)<span class="sc">+</span></span>
<span id="cb30-10"><a href="#cb30-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">fill=</span>F)<span class="sc">+</span></span>
<span id="cb30-11"><a href="#cb30-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(<span class="at">fun=</span><span class="st">&quot;median&quot;</span>, <span class="at">geom=</span><span class="st">&quot;point&quot;</span>,<span class="at">color=</span><span class="st">&quot;yellow&quot;</span>,<span class="at">size=</span><span class="dv">1</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-16-1.png" width="2400" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a>res_ks_s <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(selection) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">adj_p=</span><span class="fu">p.adjust</span>(p,<span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>),<span class="at">sig_per=</span><span class="fu">mean</span>(adj_p<span class="sc">&lt;</span><span class="fl">0.1</span>)) <span class="ot">-&gt;</span> summ</span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `summarise()` has grouped output by &#39;selection&#39;. You can override using the `.groups` argument.</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data=</span>summ,<span class="fu">aes</span>(<span class="at">x=</span>selection,<span class="at">y=</span>sig_per))<span class="sc">+</span></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>,<span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Selection&quot;</span>,<span class="at">y=</span><span class="st">&quot;Percent of significant samples&quot;</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-16-2.png" width="2400" /> Of note, no tumors show significant signal under the same simulated conditions.</p>
<p>To futher compare the power of methods, in addition to sequencing limitations, we also added different proportions of false positive neoantigen: we randomly sampled nonantigenic mutations of simulated tumors (varied between 5 and 500% of the number of true neoantigen) that were falsely labeled as neoantigen. Then we calculated the number of significant samples in each read depth and proportions of false positive neoantigen combination as the power of method to detect selection strength:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a><span class="do">###power analysis</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a><span class="do">##es</span></span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>cal_ccf <span class="ot">&lt;-</span> <span class="cf">function</span>(depth){</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="cf">function</span>(ccf){</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">map_dbl</span>(ccf,<span class="cf">function</span>(x,y){<span class="fu">rbinom</span>(<span class="dv">1</span>,y,x)<span class="sc">/</span>y},depth)</span>
<span id="cb32-6"><a href="#cb32-6" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-7"><a href="#cb32-7" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-8"><a href="#cb32-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-9"><a href="#cb32-9" aria-hidden="true" tabindex="-1"></a>neo_per <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">5</span>,<span class="dv">20</span>,<span class="dv">50</span>,<span class="dv">100</span>,<span class="dv">200</span>,<span class="dv">500</span>)<span class="sc">*</span><span class="fl">0.01</span></span>
<span id="cb32-10"><a href="#cb32-10" aria-hidden="true" tabindex="-1"></a>depth <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="dv">50</span>,<span class="dv">100</span>,<span class="dv">200</span>,<span class="dv">500</span>,<span class="dv">800</span>,<span class="dv">1000</span>,<span class="dv">2000</span>,<span class="dv">5000</span>,<span class="dv">10000</span>)</span>
<span id="cb32-11"><a href="#cb32-11" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="dv">63</span>)</span>
<span id="cb32-12"><a href="#cb32-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-13"><a href="#cb32-13" aria-hidden="true" tabindex="-1"></a>com <span class="ot">&lt;-</span> <span class="fu">expand.grid</span>(neo_per,depth)</span>
<span id="cb32-14"><a href="#cb32-14" aria-hidden="true" tabindex="-1"></a>com<span class="sc">$</span>lable <span class="ot">&lt;-</span> <span class="fu">paste</span>(com<span class="sc">$</span>Var1,com<span class="sc">$</span>Var2,<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb32-15"><a href="#cb32-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-16"><a href="#cb32-16" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(re) <span class="ot">&lt;-</span> com<span class="sc">$</span>lable</span>
<span id="cb32-17"><a href="#cb32-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-18"><a href="#cb32-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-19"><a href="#cb32-19" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">paste0</span>(<span class="st">&quot;~/test/data/merge/0.24.rds&quot;</span>))</span>
<span id="cb32-20"><a href="#cb32-20" aria-hidden="true" tabindex="-1"></a>s <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span></span>
<span id="cb32-21"><a href="#cb32-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mut_neo=</span><span class="fu">ifelse</span>(mut_neo<span class="sc">==</span><span class="st">&quot;false&quot;</span>,<span class="st">&quot;no&quot;</span>,<span class="st">&quot;yes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb32-22"><a href="#cb32-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">neo=</span>mut_neo) <span class="sc">%&gt;%</span></span>
<span id="cb32-23"><a href="#cb32-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mut_sample_id=</span><span class="fu">paste</span>(sample,mut,<span class="at">sep =</span> <span class="st">&quot;-&quot;</span>))</span>
<span id="cb32-24"><a href="#cb32-24" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">20211024</span>)</span>
<span id="cb32-25"><a href="#cb32-25" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> neo_per){</span>
<span id="cb32-26"><a href="#cb32-26" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> depth){</span>
<span id="cb32-27"><a href="#cb32-27" aria-hidden="true" tabindex="-1"></a>    dep_ccf <span class="ot">&lt;-</span> <span class="fu">cal_ccf</span>(<span class="at">depth =</span> j)</span>
<span id="cb32-28"><a href="#cb32-28" aria-hidden="true" tabindex="-1"></a>    neo <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo <span class="sc">==</span><span class="st">&quot;yes&quot;</span>)</span>
<span id="cb32-29"><a href="#cb32-29" aria-hidden="true" tabindex="-1"></a>    neo_summ <span class="ot">&lt;-</span> neo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">neo_c=</span><span class="fu">n</span>())</span>
<span id="cb32-30"><a href="#cb32-30" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo <span class="sc">!=</span> <span class="st">&quot;yes&quot;</span>)</span>
<span id="cb32-31"><a href="#cb32-31" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> <span class="fu">left_join</span>(not_neo,neo_summ)</span>
<span id="cb32-32"><a href="#cb32-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-33"><a href="#cb32-33" aria-hidden="true" tabindex="-1"></a>    false_neo <span class="ot">&lt;-</span> not_neo <span class="sc">%&gt;%</span></span>
<span id="cb32-34"><a href="#cb32-34" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb32-35"><a href="#cb32-35" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">sample_id=</span><span class="fu">sample</span>(mut_sample_id,<span class="fu">unique</span>(neo_c)<span class="sc">*</span>i,<span class="at">replace =</span> F))</span>
<span id="cb32-36"><a href="#cb32-36" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> not_neo <span class="sc">%&gt;%</span></span>
<span id="cb32-37"><a href="#cb32-37" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">neo =</span> <span class="fu">ifelse</span>(mut_sample_id <span class="sc">%in%</span> false_neo<span class="sc">$</span>sample_id,<span class="st">&quot;yes&quot;</span>,neo))</span>
<span id="cb32-38"><a href="#cb32-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-39"><a href="#cb32-39" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> <span class="fu">rbind</span>(neo,not_neo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>neo_c))</span>
<span id="cb32-40"><a href="#cb32-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-41"><a href="#cb32-41" aria-hidden="true" tabindex="-1"></a>    s_new<span class="sc">$</span>ccf_noise <span class="ot">&lt;-</span> <span class="fu">dep_ccf</span>(s_new<span class="sc">$</span>ccf)</span>
<span id="cb32-42"><a href="#cb32-42" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> s_new <span class="sc">%&gt;%</span></span>
<span id="cb32-43"><a href="#cb32-43" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(ccf_noise<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb32-44"><a href="#cb32-44" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> s_new <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>ccf) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_noise)</span>
<span id="cb32-45"><a href="#cb32-45" aria-hidden="true" tabindex="-1"></a>    a <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(s_new)</span>
<span id="cb32-46"><a href="#cb32-46" aria-hidden="true" tabindex="-1"></a>    re[[<span class="fu">paste</span>(i,j,<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)]] <span class="ot">&lt;-</span> a</span>
<span id="cb32-47"><a href="#cb32-47" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;FP &quot;</span>,i,<span class="st">&quot; &amp; depth &quot;</span>,j,<span class="st">&quot; complted </span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb32-48"><a href="#cb32-48" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-49"><a href="#cb32-49" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-50"><a href="#cb32-50" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(re,<span class="at">file =</span> <span class="st">&quot;../data/sim_growth_power_alt.rds&quot;</span>)</span>
<span id="cb32-51"><a href="#cb32-51" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-52"><a href="#cb32-52" aria-hidden="true" tabindex="-1"></a><span class="do">##KS</span></span>
<span id="cb32-53"><a href="#cb32-53" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> neo_per){</span>
<span id="cb32-54"><a href="#cb32-54" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> depth){</span>
<span id="cb32-55"><a href="#cb32-55" aria-hidden="true" tabindex="-1"></a>    dep_ccf <span class="ot">&lt;-</span> <span class="fu">cal_ccf</span>(<span class="at">depth =</span> j)</span>
<span id="cb32-56"><a href="#cb32-56" aria-hidden="true" tabindex="-1"></a>    neo <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo <span class="sc">==</span><span class="st">&quot;yes&quot;</span>)</span>
<span id="cb32-57"><a href="#cb32-57" aria-hidden="true" tabindex="-1"></a>    neo_summ <span class="ot">&lt;-</span> neo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">neo_c=</span><span class="fu">n</span>())</span>
<span id="cb32-58"><a href="#cb32-58" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> s <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo <span class="sc">!=</span> <span class="st">&quot;yes&quot;</span>)</span>
<span id="cb32-59"><a href="#cb32-59" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> <span class="fu">left_join</span>(not_neo,neo_summ)</span>
<span id="cb32-60"><a href="#cb32-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-61"><a href="#cb32-61" aria-hidden="true" tabindex="-1"></a>    <span class="co">#false_neo &lt;- sample(not_neo$mut,nrow(neo)*i,replace = F)</span></span>
<span id="cb32-62"><a href="#cb32-62" aria-hidden="true" tabindex="-1"></a>    false_neo <span class="ot">&lt;-</span> not_neo <span class="sc">%&gt;%</span></span>
<span id="cb32-63"><a href="#cb32-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb32-64"><a href="#cb32-64" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">sample_id=</span><span class="fu">sample</span>(mut_sample_id,<span class="fu">unique</span>(neo_c)<span class="sc">*</span>i,<span class="at">replace =</span> F))</span>
<span id="cb32-65"><a href="#cb32-65" aria-hidden="true" tabindex="-1"></a>    not_neo <span class="ot">&lt;-</span> not_neo <span class="sc">%&gt;%</span></span>
<span id="cb32-66"><a href="#cb32-66" aria-hidden="true" tabindex="-1"></a>      <span class="fu">mutate</span>(<span class="at">neo =</span> <span class="fu">ifelse</span>(mut_sample_id <span class="sc">%in%</span> false_neo<span class="sc">$</span>sample_id,<span class="st">&quot;yes&quot;</span>,neo))</span>
<span id="cb32-67"><a href="#cb32-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-68"><a href="#cb32-68" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> <span class="fu">rbind</span>(neo,not_neo <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>neo_c))</span>
<span id="cb32-69"><a href="#cb32-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-70"><a href="#cb32-70" aria-hidden="true" tabindex="-1"></a>    s_new<span class="sc">$</span>ccf_noise <span class="ot">&lt;-</span> <span class="fu">dep_ccf</span>(s_new<span class="sc">$</span>ccf)</span>
<span id="cb32-71"><a href="#cb32-71" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> s_new <span class="sc">%&gt;%</span></span>
<span id="cb32-72"><a href="#cb32-72" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(ccf_noise<span class="sc">&gt;</span><span class="dv">0</span>)</span>
<span id="cb32-73"><a href="#cb32-73" aria-hidden="true" tabindex="-1"></a>    s_new <span class="ot">&lt;-</span> s_new <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>ccf) <span class="sc">%&gt;%</span> <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_noise)</span>
<span id="cb32-74"><a href="#cb32-74" aria-hidden="true" tabindex="-1"></a>    results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(s_new<span class="sc">$</span>sample)))</span>
<span id="cb32-75"><a href="#cb32-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(s_new<span class="sc">$</span>sample)</span>
<span id="cb32-76"><a href="#cb32-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb32-77"><a href="#cb32-77" aria-hidden="true" tabindex="-1"></a>    cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">20</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb32-78"><a href="#cb32-78" aria-hidden="true" tabindex="-1"></a>    results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb32-79"><a href="#cb32-79" aria-hidden="true" tabindex="-1"></a>                             <span class="cf">function</span>(x){</span>
<span id="cb32-80"><a href="#cb32-80" aria-hidden="true" tabindex="-1"></a>                               data <span class="ot">&lt;-</span> s_new <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">==</span> x)</span>
<span id="cb32-81"><a href="#cb32-81" aria-hidden="true" tabindex="-1"></a>                               neo <span class="ot">&lt;-</span> data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>)</span>
<span id="cb32-82"><a href="#cb32-82" aria-hidden="true" tabindex="-1"></a>                               ks_t <span class="ot">&lt;-</span> <span class="fu">ks.test</span>(neo<span class="sc">$</span>ccf,data<span class="sc">$</span>ccf)</span>
<span id="cb32-83"><a href="#cb32-83" aria-hidden="true" tabindex="-1"></a>                               ttt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">D=</span>ks_t<span class="sc">$</span>statistic,<span class="at">p=</span>ks_t<span class="sc">$</span>p.value,<span class="at">sample=</span>x)</span>
<span id="cb32-84"><a href="#cb32-84" aria-hidden="true" tabindex="-1"></a>                               <span class="fu">return</span>(ttt)</span>
<span id="cb32-85"><a href="#cb32-85" aria-hidden="true" tabindex="-1"></a>                             },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb32-86"><a href="#cb32-86" aria-hidden="true" tabindex="-1"></a>    <span class="fu">stopCluster</span>(cl)</span>
<span id="cb32-87"><a href="#cb32-87" aria-hidden="true" tabindex="-1"></a>    res_ones <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb32-88"><a href="#cb32-88" aria-hidden="true" tabindex="-1"></a>    re[[<span class="fu">paste</span>(i,j,<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)]] <span class="ot">&lt;-</span> res_ones</span>
<span id="cb32-89"><a href="#cb32-89" aria-hidden="true" tabindex="-1"></a>    <span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="st">&quot;FP &quot;</span>,i,<span class="st">&quot; &amp; depth &quot;</span>,j,<span class="st">&quot; complted </span><span class="sc">\n</span><span class="st">&quot;</span>))</span>
<span id="cb32-90"><a href="#cb32-90" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb32-91"><a href="#cb32-91" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb32-92"><a href="#cb32-92" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(re,<span class="at">file =</span> <span class="st">&quot;../data/sim_growth_power_alt_ks.rds&quot;</span>)</span></code></pre></div>
<p>We used heatmap to display the power:</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="do">##es</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/sim_growth_power_alt.rds&quot;</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">neo_depth=</span><span class="fu">names</span>(re),<span class="at">fre=</span><span class="cn">NA</span>)</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(results)){</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> re[[i]]</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dt<span class="sc">$</span>p,<span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(re)[i],<span class="st">&quot; &quot;</span>,<span class="fu">sum</span>(dt<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)))</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>fre[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(dt<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_50 38&quot;</span></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_50 33&quot;</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_50 21&quot;</span></span>
<span id="cb33-13"><a href="#cb33-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_50 0&quot;</span></span>
<span id="cb33-14"><a href="#cb33-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_50 7&quot;</span></span>
<span id="cb33-15"><a href="#cb33-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_50 0&quot;</span></span>
<span id="cb33-16"><a href="#cb33-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_50 0&quot;</span></span>
<span id="cb33-17"><a href="#cb33-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_100 83&quot;</span></span>
<span id="cb33-18"><a href="#cb33-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_100 73&quot;</span></span>
<span id="cb33-19"><a href="#cb33-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_100 63&quot;</span></span>
<span id="cb33-20"><a href="#cb33-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_100 44&quot;</span></span>
<span id="cb33-21"><a href="#cb33-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_100 13&quot;</span></span>
<span id="cb33-22"><a href="#cb33-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_100 0&quot;</span></span>
<span id="cb33-23"><a href="#cb33-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_100 4&quot;</span></span>
<span id="cb33-24"><a href="#cb33-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_200 90&quot;</span></span>
<span id="cb33-25"><a href="#cb33-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_200 89&quot;</span></span>
<span id="cb33-26"><a href="#cb33-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_200 70&quot;</span></span>
<span id="cb33-27"><a href="#cb33-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_200 53&quot;</span></span>
<span id="cb33-28"><a href="#cb33-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_200 28&quot;</span></span>
<span id="cb33-29"><a href="#cb33-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_200 12&quot;</span></span>
<span id="cb33-30"><a href="#cb33-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_200 0&quot;</span></span>
<span id="cb33-31"><a href="#cb33-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_500 99&quot;</span></span>
<span id="cb33-32"><a href="#cb33-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_500 99&quot;</span></span>
<span id="cb33-33"><a href="#cb33-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_500 83&quot;</span></span>
<span id="cb33-34"><a href="#cb33-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_500 71&quot;</span></span>
<span id="cb33-35"><a href="#cb33-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_500 51&quot;</span></span>
<span id="cb33-36"><a href="#cb33-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_500 11&quot;</span></span>
<span id="cb33-37"><a href="#cb33-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_500 1&quot;</span></span>
<span id="cb33-38"><a href="#cb33-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_800 98&quot;</span></span>
<span id="cb33-39"><a href="#cb33-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_800 97&quot;</span></span>
<span id="cb33-40"><a href="#cb33-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_800 92&quot;</span></span>
<span id="cb33-41"><a href="#cb33-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_800 79&quot;</span></span>
<span id="cb33-42"><a href="#cb33-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_800 44&quot;</span></span>
<span id="cb33-43"><a href="#cb33-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_800 19&quot;</span></span>
<span id="cb33-44"><a href="#cb33-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_800 1&quot;</span></span>
<span id="cb33-45"><a href="#cb33-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_1000 100&quot;</span></span>
<span id="cb33-46"><a href="#cb33-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_1000 99&quot;</span></span>
<span id="cb33-47"><a href="#cb33-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_1000 92&quot;</span></span>
<span id="cb33-48"><a href="#cb33-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_1000 75&quot;</span></span>
<span id="cb33-49"><a href="#cb33-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_1000 46&quot;</span></span>
<span id="cb33-50"><a href="#cb33-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_1000 34&quot;</span></span>
<span id="cb33-51"><a href="#cb33-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_1000 7&quot;</span></span>
<span id="cb33-52"><a href="#cb33-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_2000 100&quot;</span></span>
<span id="cb33-53"><a href="#cb33-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_2000 100&quot;</span></span>
<span id="cb33-54"><a href="#cb33-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_2000 97&quot;</span></span>
<span id="cb33-55"><a href="#cb33-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_2000 88&quot;</span></span>
<span id="cb33-56"><a href="#cb33-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_2000 70&quot;</span></span>
<span id="cb33-57"><a href="#cb33-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_2000 16&quot;</span></span>
<span id="cb33-58"><a href="#cb33-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_2000 0&quot;</span></span>
<span id="cb33-59"><a href="#cb33-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_5000 100&quot;</span></span>
<span id="cb33-60"><a href="#cb33-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_5000 100&quot;</span></span>
<span id="cb33-61"><a href="#cb33-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_5000 98&quot;</span></span>
<span id="cb33-62"><a href="#cb33-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_5000 95&quot;</span></span>
<span id="cb33-63"><a href="#cb33-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_5000 72&quot;</span></span>
<span id="cb33-64"><a href="#cb33-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_5000 31&quot;</span></span>
<span id="cb33-65"><a href="#cb33-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_5000 7&quot;</span></span>
<span id="cb33-66"><a href="#cb33-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_10000 100&quot;</span></span>
<span id="cb33-67"><a href="#cb33-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_10000 100&quot;</span></span>
<span id="cb33-68"><a href="#cb33-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_10000 98&quot;</span></span>
<span id="cb33-69"><a href="#cb33-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_10000 95&quot;</span></span>
<span id="cb33-70"><a href="#cb33-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_10000 79&quot;</span></span>
<span id="cb33-71"><a href="#cb33-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_10000 48&quot;</span></span>
<span id="cb33-72"><a href="#cb33-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_10000 9&quot;</span></span>
<span id="cb33-73"><a href="#cb33-73" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results <span class="sc">%&gt;%</span></span>
<span id="cb33-74"><a href="#cb33-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> neo_depth,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;neo_per&quot;</span>,<span class="st">&quot;depth&quot;</span>),<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb33-75"><a href="#cb33-75" aria-hidden="true" tabindex="-1"></a>results <span class="sc">%&gt;%</span></span>
<span id="cb33-76"><a href="#cb33-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">depth=</span><span class="fu">paste0</span>(<span class="st">&quot;depth&quot;</span>,depth)) <span class="sc">%&gt;%</span></span>
<span id="cb33-77"><a href="#cb33-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> depth,<span class="at">values_from =</span> fre) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="ot">-&gt;</span> mat</span>
<span id="cb33-78"><a href="#cb33-78" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> mat<span class="sc">$</span>neo_per</span>
<span id="cb33-79"><a href="#cb33-79" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> mat <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>neo_per)</span>
<span id="cb33-80"><a href="#cb33-80" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;#551A8B&quot;</span>, <span class="st">&quot;#8B3A62&quot;</span>, <span class="st">&quot;#FFFF00&quot;</span>)</span>
<span id="cb33-81"><a href="#cb33-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-82"><a href="#cb33-82" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;depth&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(mat))</span>
<span id="cb33-83"><a href="#cb33-83" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="dv">100</span><span class="sc">*</span><span class="fu">as.numeric</span>(<span class="fu">rownames</span>(mat)))</span>
<span id="cb33-84"><a href="#cb33-84" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(mat,<span class="at">cluster_columns =</span> F,<span class="at">cluster_rows =</span> F,<span class="at">row_names_side =</span> <span class="st">&quot;left&quot;</span>,</span>
<span id="cb33-85"><a href="#cb33-85" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&quot;Power(%)&quot;</span>,<span class="at">col =</span> cols,</span>
<span id="cb33-86"><a href="#cb33-86" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_title =</span> <span class="st">&quot;Ratio of false positive </span><span class="sc">\n</span><span class="st"> neoantigens added (%)&quot;</span>,</span>
<span id="cb33-87"><a href="#cb33-87" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_title =</span> <span class="st">&quot;Read depth&quot;</span>,<span class="at">column_names_rot =</span> <span class="dv">0</span>,<span class="at">column_title_side =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb33-88"><a href="#cb33-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: The input is a data frame, convert it to the matrix.</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-18-1.png" width="2400" /></p>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a><span class="do">##ks</span></span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>re <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/sim_growth_power_alt_ks.rds&quot;</span>)</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">neo_depth=</span><span class="fu">names</span>(re),<span class="at">fre=</span><span class="cn">NA</span>)</span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(results)){</span>
<span id="cb34-6"><a href="#cb34-6" aria-hidden="true" tabindex="-1"></a>  dt <span class="ot">&lt;-</span> re[[i]]</span>
<span id="cb34-7"><a href="#cb34-7" aria-hidden="true" tabindex="-1"></a>  dt<span class="sc">$</span>padj <span class="ot">&lt;-</span> <span class="fu">p.adjust</span>(dt<span class="sc">$</span>p,<span class="at">method =</span> <span class="st">&quot;fdr&quot;</span>)</span>
<span id="cb34-8"><a href="#cb34-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">print</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(re)[i],<span class="st">&quot; &quot;</span>,<span class="fu">sum</span>(dt<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)))</span>
<span id="cb34-9"><a href="#cb34-9" aria-hidden="true" tabindex="-1"></a>  results<span class="sc">$</span>fre[i] <span class="ot">&lt;-</span> <span class="fu">sum</span>(dt<span class="sc">$</span>padj <span class="sc">&lt;</span> <span class="fl">0.1</span>)</span>
<span id="cb34-10"><a href="#cb34-10" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb34-11"><a href="#cb34-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_50 0&quot;</span></span>
<span id="cb34-12"><a href="#cb34-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_50 0&quot;</span></span>
<span id="cb34-13"><a href="#cb34-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_50 0&quot;</span></span>
<span id="cb34-14"><a href="#cb34-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_50 0&quot;</span></span>
<span id="cb34-15"><a href="#cb34-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_50 0&quot;</span></span>
<span id="cb34-16"><a href="#cb34-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_50 0&quot;</span></span>
<span id="cb34-17"><a href="#cb34-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_50 0&quot;</span></span>
<span id="cb34-18"><a href="#cb34-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_100 0&quot;</span></span>
<span id="cb34-19"><a href="#cb34-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_100 0&quot;</span></span>
<span id="cb34-20"><a href="#cb34-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_100 0&quot;</span></span>
<span id="cb34-21"><a href="#cb34-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_100 0&quot;</span></span>
<span id="cb34-22"><a href="#cb34-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_100 0&quot;</span></span>
<span id="cb34-23"><a href="#cb34-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_100 0&quot;</span></span>
<span id="cb34-24"><a href="#cb34-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_100 0&quot;</span></span>
<span id="cb34-25"><a href="#cb34-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_200 0&quot;</span></span>
<span id="cb34-26"><a href="#cb34-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_200 0&quot;</span></span>
<span id="cb34-27"><a href="#cb34-27" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_200 0&quot;</span></span>
<span id="cb34-28"><a href="#cb34-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_200 0&quot;</span></span>
<span id="cb34-29"><a href="#cb34-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_200 0&quot;</span></span>
<span id="cb34-30"><a href="#cb34-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_200 0&quot;</span></span>
<span id="cb34-31"><a href="#cb34-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_200 0&quot;</span></span>
<span id="cb34-32"><a href="#cb34-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_500 66&quot;</span></span>
<span id="cb34-33"><a href="#cb34-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_500 62&quot;</span></span>
<span id="cb34-34"><a href="#cb34-34" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_500 30&quot;</span></span>
<span id="cb34-35"><a href="#cb34-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_500 16&quot;</span></span>
<span id="cb34-36"><a href="#cb34-36" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_500 1&quot;</span></span>
<span id="cb34-37"><a href="#cb34-37" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_500 0&quot;</span></span>
<span id="cb34-38"><a href="#cb34-38" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_500 0&quot;</span></span>
<span id="cb34-39"><a href="#cb34-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_800 95&quot;</span></span>
<span id="cb34-40"><a href="#cb34-40" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_800 85&quot;</span></span>
<span id="cb34-41"><a href="#cb34-41" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_800 68&quot;</span></span>
<span id="cb34-42"><a href="#cb34-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_800 47&quot;</span></span>
<span id="cb34-43"><a href="#cb34-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_800 13&quot;</span></span>
<span id="cb34-44"><a href="#cb34-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_800 0&quot;</span></span>
<span id="cb34-45"><a href="#cb34-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_800 0&quot;</span></span>
<span id="cb34-46"><a href="#cb34-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_1000 97&quot;</span></span>
<span id="cb34-47"><a href="#cb34-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_1000 97&quot;</span></span>
<span id="cb34-48"><a href="#cb34-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_1000 95&quot;</span></span>
<span id="cb34-49"><a href="#cb34-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_1000 55&quot;</span></span>
<span id="cb34-50"><a href="#cb34-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_1000 19&quot;</span></span>
<span id="cb34-51"><a href="#cb34-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_1000 0&quot;</span></span>
<span id="cb34-52"><a href="#cb34-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_1000 0&quot;</span></span>
<span id="cb34-53"><a href="#cb34-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_2000 100&quot;</span></span>
<span id="cb34-54"><a href="#cb34-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_2000 100&quot;</span></span>
<span id="cb34-55"><a href="#cb34-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_2000 99&quot;</span></span>
<span id="cb34-56"><a href="#cb34-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_2000 96&quot;</span></span>
<span id="cb34-57"><a href="#cb34-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_2000 73&quot;</span></span>
<span id="cb34-58"><a href="#cb34-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_2000 0&quot;</span></span>
<span id="cb34-59"><a href="#cb34-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_2000 0&quot;</span></span>
<span id="cb34-60"><a href="#cb34-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_5000 100&quot;</span></span>
<span id="cb34-61"><a href="#cb34-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_5000 100&quot;</span></span>
<span id="cb34-62"><a href="#cb34-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_5000 100&quot;</span></span>
<span id="cb34-63"><a href="#cb34-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_5000 100&quot;</span></span>
<span id="cb34-64"><a href="#cb34-64" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_5000 98&quot;</span></span>
<span id="cb34-65"><a href="#cb34-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_5000 65&quot;</span></span>
<span id="cb34-66"><a href="#cb34-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_5000 0&quot;</span></span>
<span id="cb34-67"><a href="#cb34-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0_10000 100&quot;</span></span>
<span id="cb34-68"><a href="#cb34-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.05_10000 100&quot;</span></span>
<span id="cb34-69"><a href="#cb34-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.2_10000 100&quot;</span></span>
<span id="cb34-70"><a href="#cb34-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;0.5_10000 100&quot;</span></span>
<span id="cb34-71"><a href="#cb34-71" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;1_10000 100&quot;</span></span>
<span id="cb34-72"><a href="#cb34-72" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;2_10000 96&quot;</span></span>
<span id="cb34-73"><a href="#cb34-73" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; [1] &quot;5_10000 0&quot;</span></span>
<span id="cb34-74"><a href="#cb34-74" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> results <span class="sc">%&gt;%</span></span>
<span id="cb34-75"><a href="#cb34-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(<span class="at">col =</span> neo_depth,<span class="at">into =</span> <span class="fu">c</span>(<span class="st">&quot;neo_per&quot;</span>,<span class="st">&quot;depth&quot;</span>),<span class="at">sep =</span> <span class="st">&quot;_&quot;</span>)</span>
<span id="cb34-76"><a href="#cb34-76" aria-hidden="true" tabindex="-1"></a>results <span class="sc">%&gt;%</span></span>
<span id="cb34-77"><a href="#cb34-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">depth=</span><span class="fu">paste0</span>(<span class="st">&quot;depth&quot;</span>,depth)) <span class="sc">%&gt;%</span></span>
<span id="cb34-78"><a href="#cb34-78" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> depth,<span class="at">values_from =</span> fre) <span class="sc">%&gt;%</span> <span class="fu">as.data.frame</span>() <span class="ot">-&gt;</span> mat</span>
<span id="cb34-79"><a href="#cb34-79" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> mat<span class="sc">$</span>neo_per</span>
<span id="cb34-80"><a href="#cb34-80" aria-hidden="true" tabindex="-1"></a>mat <span class="ot">&lt;-</span> mat <span class="sc">%&gt;%</span> <span class="fu">select</span>(<span class="sc">-</span>neo_per)</span>
<span id="cb34-81"><a href="#cb34-81" aria-hidden="true" tabindex="-1"></a>cols <span class="ot">&lt;-</span><span class="fu">c</span>(<span class="st">&quot;#551A8B&quot;</span>, <span class="st">&quot;#8B3A62&quot;</span>, <span class="st">&quot;#FFFF00&quot;</span>)</span>
<span id="cb34-82"><a href="#cb34-82" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb34-83"><a href="#cb34-83" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot;depth&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">colnames</span>(mat))</span>
<span id="cb34-84"><a href="#cb34-84" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(mat) <span class="ot">&lt;-</span> <span class="fu">as.character</span>(<span class="dv">100</span><span class="sc">*</span><span class="fu">as.numeric</span>(<span class="fu">rownames</span>(mat)))</span>
<span id="cb34-85"><a href="#cb34-85" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(mat,<span class="at">cluster_columns =</span> F,<span class="at">cluster_rows =</span> F,<span class="at">row_names_side =</span> <span class="st">&quot;left&quot;</span>,</span>
<span id="cb34-86"><a href="#cb34-86" aria-hidden="true" tabindex="-1"></a>        <span class="at">name =</span> <span class="st">&quot;Power(%)&quot;</span>,<span class="at">col =</span> cols,</span>
<span id="cb34-87"><a href="#cb34-87" aria-hidden="true" tabindex="-1"></a>        <span class="at">row_title =</span> <span class="st">&quot;Ratio of false positive </span><span class="sc">\n</span><span class="st"> neoantigens added (%)&quot;</span>,</span>
<span id="cb34-88"><a href="#cb34-88" aria-hidden="true" tabindex="-1"></a>        <span class="at">column_title =</span> <span class="st">&quot;Read depth&quot;</span>,<span class="at">column_names_rot =</span> <span class="dv">0</span>,<span class="at">column_title_side =</span> <span class="st">&quot;bottom&quot;</span>)</span>
<span id="cb34-89"><a href="#cb34-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: The input is a data frame, convert it to the matrix.</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-18-2.png" width="2400" /></p>
</div>
<div id="pan-cancer-features-and-correlations-of-immunoediting-signal" class="section level2">
<h2>Pan-cancer features and correlations of immunoediting signal</h2>
<p>Then we compared the immune cell infiltration level between tumors with detectable immunoediting-elimination signal (ESCCF&lt;0, P&lt;0.05) or immunoediting-escape signal (ESRNA&lt;0, P&lt;0.05) and other samples:</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>pancancer_subtcells <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/pancancer_subtcells.rds&quot;</span>)</span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>nes_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples_addp.rds&quot;</span>)</span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>nes_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/nes_exp_addp.rds&quot;</span>)</span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a>ccf_immune <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>  nes_ccf,</span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  pancancer_subtcells</span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p <span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;sample&quot;</span></span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>exp_immune <span class="ot">&lt;-</span> <span class="fu">left_join</span>(</span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  nes_exp ,</span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  pancancer_subtcells</span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p_value <span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;sample&quot;</span></span>
<span id="cb35-18"><a href="#cb35-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-19"><a href="#cb35-19" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>ccf_immune,<span class="fu">aes</span>(<span class="at">x=</span>es_type,<span class="at">y=</span>CD8_T<span class="sc">+</span>NK))<span class="sc">+</span></span>
<span id="cb35-20"><a href="#cb35-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb35-21"><a href="#cb35-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb35-22"><a href="#cb35-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb35-23"><a href="#cb35-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Elimination&quot;</span>,<span class="at">y=</span><span class="st">&quot;CD8 T + NK&quot;</span>)</span>
<span id="cb35-24"><a href="#cb35-24" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>ccf_immune,<span class="fu">aes</span>(<span class="at">x=</span>es_type,<span class="at">y=</span>iTreg<span class="sc">+</span>nTreg))<span class="sc">+</span></span>
<span id="cb35-25"><a href="#cb35-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb35-26"><a href="#cb35-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb35-27"><a href="#cb35-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb35-28"><a href="#cb35-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Elimination&quot;</span>,<span class="at">y=</span><span class="st">&quot;iTreg + nTreg&quot;</span>)</span>
<span id="cb35-29"><a href="#cb35-29" aria-hidden="true" tabindex="-1"></a>p1 <span class="sc">+</span> p2</span>
<span id="cb35-30"><a href="#cb35-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 70 rows containing non-finite values (stat_boxplot).</span></span>
<span id="cb35-31"><a href="#cb35-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 70 rows containing non-finite values (stat_compare_means).</span></span>
<span id="cb35-32"><a href="#cb35-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 70 rows containing non-finite values (stat_boxplot).</span></span>
<span id="cb35-33"><a href="#cb35-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 70 rows containing non-finite values (stat_compare_means).</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-19-1.png" width="2400" /></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>exp_immune,<span class="fu">aes</span>(<span class="at">x=</span>es_type,<span class="at">y=</span>CD8_T<span class="sc">+</span>NK))<span class="sc">+</span></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Escape&quot;</span>,<span class="at">y=</span><span class="st">&quot;CD8 T + NK&quot;</span>)</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>exp_immune,<span class="fu">aes</span>(<span class="at">x=</span>es_type,<span class="at">y=</span>iTreg<span class="sc">+</span>nTreg))<span class="sc">+</span></span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>()<span class="sc">+</span></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_compare_means</span>()<span class="sc">+</span></span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_prism</span>()<span class="sc">+</span></span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Escape&quot;</span>,<span class="at">y=</span><span class="st">&quot;iTreg + nTreg&quot;</span>)</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>p3 <span class="sc">+</span> p4</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 93 rows containing non-finite values (stat_boxplot).</span></span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 93 rows containing non-finite values (stat_compare_means).</span></span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 93 rows containing non-finite values (stat_boxplot).</span></span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Removed 93 rows containing non-finite values (stat_compare_means).</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-19-2.png" width="2400" /></p>
<p>In tumors with detectable immunoediting-elimination signal, a slightly increased CD8+ T cell infiltration status compared with the remaining samples were observed, but the difference does not reach statistical significance (P=0.2). This data suggests that historically happened immunoediting-elimination process may not be reflected in the current immune cell infiltration status.</p>
<p>The immunoediting-escape signal quantified as ES-RNA also do not show statistically significant difference between samples with detectable immunoediting-escape signal and the remaining samples in CD8+ T cell infiltration. However we observe a significant up-regulated regulatory T cell (Treg) percentage in samples with detectable immunoediting-escape signal, and up-regulated Treg has been reported to stimulate tumor immune escape.</p>
</div>
</div>
<div id="immunotherapy-dataset-analysis" class="section level1">
<h1>Immunotherapy dataset analysis</h1>
<div id="quantified-immunoediting-elimination-signal-predicts-the-clinical-response-of-cancer-immunotherapy" class="section level2">
<h2>Quantified immunoediting-elimination signal predicts the clinical response of cancer immunotherapy</h2>
<p>To investigate the predictive performance of the quantified immunoediting-elimination signal (ESCCF) in ICI response prediction for individual patient, we searched for public ICI datasets with raw WES data and RNA-seq data available, and three melanoma ICI datasets have been identified. Then we calculated the immunoediting-elimination signal (ES-CCF) for each patient:</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/all_mut_ccf_ici.rds&quot;</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici <span class="ot">&lt;-</span> all_mut_ccf_ici <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ccf=</span>cancer_cell_frac) <span class="sc">%&gt;%</span></span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>samples_has_subclonal <span class="ot">&lt;-</span> all_mut_ccf_ici <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ccf<span class="sc">&lt;</span><span class="fl">0.6</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb37-6"><a href="#cb37-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample)</span>
<span id="cb37-7"><a href="#cb37-7" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici  <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> samples_has_subclonal<span class="sc">$</span>sample) <span class="sc">%&gt;%</span></span>
<span id="cb37-8"><a href="#cb37-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb37-9"><a href="#cb37-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">c_n=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>),<span class="at">c_m=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(c_n<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> c_m <span class="sc">&gt;=</span><span class="dv">1</span>) <span class="ot">-&gt;</span> summ</span>
<span id="cb37-10"><a href="#cb37-10" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> all_mut_ccf_ici <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">%in%</span> summ<span class="sc">$</span>sample)</span>
<span id="cb37-11"><a href="#cb37-11" aria-hidden="true" tabindex="-1"></a>cal_nes_warp <span class="ot">&lt;-</span> <span class="cf">function</span>(dt){</span>
<span id="cb37-12"><a href="#cb37-12" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">vector</span>(<span class="st">&quot;list&quot;</span>,<span class="at">length =</span> <span class="fu">length</span>(<span class="fu">unique</span>(dt<span class="sc">$</span>sample)))</span>
<span id="cb37-13"><a href="#cb37-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">names</span>(results_ccf) <span class="ot">&lt;-</span> <span class="fu">unique</span>(dt<span class="sc">$</span>sample)</span>
<span id="cb37-14"><a href="#cb37-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb37-15"><a href="#cb37-15" aria-hidden="true" tabindex="-1"></a>  cl <span class="ot">&lt;-</span> <span class="fu">makeCluster</span>(<span class="fu">getOption</span>(<span class="st">&quot;cl.cores&quot;</span>, <span class="dv">18</span>),<span class="at">type=</span><span class="st">&quot;FORK&quot;</span>)</span>
<span id="cb37-16"><a href="#cb37-16" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">parSapply</span>(<span class="at">cl=</span>cl,<span class="fu">names</span>(results_ccf),</span>
<span id="cb37-17"><a href="#cb37-17" aria-hidden="true" tabindex="-1"></a>                           <span class="cf">function</span>(x){</span>
<span id="cb37-18"><a href="#cb37-18" aria-hidden="true" tabindex="-1"></a>                             data <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">filter</span>(sample <span class="sc">==</span> x)</span>
<span id="cb37-19"><a href="#cb37-19" aria-hidden="true" tabindex="-1"></a>                             a <span class="ot">&lt;-</span> NeoEnrichment<span class="sc">::</span><span class="fu">cal_nes_new_test</span>(<span class="at">dt =</span> data,</span>
<span id="cb37-20"><a href="#cb37-20" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">sample_counts =</span> <span class="dv">1000</span>,</span>
<span id="cb37-21"><a href="#cb37-21" aria-hidden="true" tabindex="-1"></a>                                                                  <span class="at">need_p =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-22"><a href="#cb37-22" aria-hidden="true" tabindex="-1"></a>                             <span class="fu">return</span>(a)</span>
<span id="cb37-23"><a href="#cb37-23" aria-hidden="true" tabindex="-1"></a>                           },<span class="at">simplify =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-24"><a href="#cb37-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stopCluster</span>(cl)</span>
<span id="cb37-25"><a href="#cb37-25" aria-hidden="true" tabindex="-1"></a>  results_ccf <span class="ot">&lt;-</span> <span class="fu">Filter</span>(<span class="cf">function</span>(x){<span class="fu">length</span>(x)<span class="sc">&gt;</span><span class="dv">1</span>},results_ccf)</span>
<span id="cb37-26"><a href="#cb37-26" aria-hidden="true" tabindex="-1"></a>  pancancer_nes_ccf <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(results_ccf)</span>
<span id="cb37-27"><a href="#cb37-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(pancancer_nes_ccf)</span>
<span id="cb37-28"><a href="#cb37-28" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb37-29"><a href="#cb37-29" aria-hidden="true" tabindex="-1"></a>neo_missense <span class="ot">&lt;-</span> neo_missense <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample,neo,ccf) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(ccf))</span>
<span id="cb37-30"><a href="#cb37-30" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">cal_nes_warp</span>(neo_missense)</span>
<span id="cb37-31"><a href="#cb37-31" aria-hidden="true" tabindex="-1"></a>neo_nes<span class="sc">$</span>es <span class="ot">&lt;-</span> <span class="fu">round</span>(neo_nes<span class="sc">$</span>es,<span class="at">digits =</span> <span class="dv">3</span>)</span>
<span id="cb37-32"><a href="#cb37-32" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(neo_nes,<span class="at">file =</span> <span class="st">&quot;../data/Immunotherapy/nes_immunetherapy.rds&quot;</span>)</span></code></pre></div>
<p>In univariate Cox proportional hazards regression analysis, quantified ESCCF value is significantly associated with cancer patients’ survival (p=0.03), and low ESCCF value (suggest the presence of high immunoediting-elimination signal) is associated with improved ICI clinical response (Hazard ratio (HR)=3.74, 95%CI=1.11-12.6) :</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a>all_clinical <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/all_clinical.rds&quot;</span>)</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/nes_immunetherapy.rds&quot;</span>)</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">left_join</span>(neo_nes,all_clinical,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(response2))</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a><span class="do">##cox analysis</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">show_forest</span>(neo_nes,<span class="at">covariates =</span> <span class="st">&quot;es&quot;</span>,<span class="at">time =</span> <span class="st">&quot;OS.time&quot;</span>,<span class="at">status =</span> <span class="st">&quot;OS&quot;</span>)</span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; =&gt; Processing variable es</span></span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Surv object...</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Building Cox model...</span></span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ==&gt; Done.</span></span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Resized limits to included dashed line in forest panel</span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>p1</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-21-1.png" width="2400" /></p>
<p>Patients are divided into three groups based on ES-CCF value cutoff determined by <code>surv_cutpoint</code> function, patients with lowest ES-CCF values (indicate the presence of immunoediting-elimination signal) show the best survival after ICI:</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/all_mut_ccf_ici.rds&quot;</span>)</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici <span class="ot">&lt;-</span> all_mut_ccf_ici <span class="sc">%&gt;%</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ccf=</span>cancer_cell_frac) <span class="sc">%&gt;%</span></span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a><span class="fu">surv_cutpoint</span>(neo_nes,<span class="st">&quot;OS.time&quot;</span>,<span class="st">&quot;OS&quot;</span>,<span class="st">&quot;es&quot;</span>)</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;    cutpoint statistic</span></span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; es   -0.222  2.443914</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span></span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type =</span> <span class="fu">case_when</span>(</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a>    es <span class="sc">&lt;=</span> (<span class="sc">-</span><span class="fl">0.222</span>) <span class="sc">~</span> <span class="st">&quot;low&quot;</span>,</span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>    es <span class="sc">&gt;</span> (<span class="sc">-</span><span class="fl">0.222</span>) <span class="sc">~</span> <span class="st">&quot;high&quot;</span></span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>low <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(es_type<span class="sc">==</span><span class="st">&quot;low&quot;</span>)</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>high <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span> <span class="fu">filter</span>(es_type<span class="sc">==</span><span class="st">&quot;high&quot;</span>)</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>all_clinical <span class="ot">&lt;-</span> all_clinical <span class="sc">%&gt;%</span></span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(sample <span class="sc">%in%</span> all_mut_ccf_ici<span class="sc">$</span>sample) <span class="sc">%&gt;%</span></span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="fu">case_when</span>(</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>    sample <span class="sc">%in%</span> low<span class="sc">$</span>sample <span class="sc">~</span> <span class="st">&quot;low&quot;</span>,</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>    sample <span class="sc">%in%</span> high<span class="sc">$</span>sample <span class="sc">~</span> <span class="st">&quot;high&quot;</span>,</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>    <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">&quot;others&quot;</span></span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>  ))</span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> <span class="fu">show_km</span>(all_clinical,<span class="st">&quot;type&quot;</span>)</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: Vectorized input to `element_text()` is not officially supported.</span></span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Results may be unexpected or may change in future versions of ggplot2.</span></span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>p3</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-22-1.png" width="2400" /></p>
<p>we use logistic regression to compare the efficiency of ESCCF, TMB and neoantigenic mutation count in predicting immunotherapy clinical response. Relationship between prognosis (patients with clinical response or without clinical response) and ESCCF, TMB and neoantigenic mutation count was analyzed. The goodness of fit was performed by Hosmer–Lemeshow test (H-L test):</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> neo_nes <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">obj=</span><span class="fu">ifelse</span>(response2<span class="sc">==</span><span class="st">&quot;response&quot;</span>,<span class="dv">1</span>,<span class="dv">0</span>))</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>model <span class="ot">&lt;-</span> <span class="fu">glm</span>(obj <span class="sc">~</span> es, <span class="at">data =</span> neo_nes, <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model)</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; glm(formula = obj ~ es, family = &quot;binomial&quot;, data = neo_nes)</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Deviance Residuals: </span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Min       1Q   Median       3Q      Max  </span></span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -1.1562  -0.8402  -0.7737   1.4243   1.7095  </span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error z value Pr(&gt;|z|)    </span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Intercept)  -0.8676     0.2247  -3.862 0.000113 ***</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; es           -1.1293     0.9067  -1.246 0.212942    </span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Dispersion parameter for binomial family taken to be 1)</span></span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Null deviance: 118.34  on 96  degrees of freedom</span></span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Residual deviance: 116.79  on 95  degrees of freedom</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AIC: 120.79</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Fisher Scoring iterations: 4</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>performance<span class="sc">::</span><span class="fu">performance_hosmer</span>(model)</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Hosmer-Lemeshow Goodness-of-Fit Test</span></span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Chi-squared: 4.874</span></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            df: 8    </span></span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       p-value: 0.771</span></span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Summary: model seems to fit well.</span></span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>all_mut_ccf_ici <span class="sc">%&gt;%</span></span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">tmb=</span><span class="fu">n</span>()<span class="sc">/</span><span class="dv">38</span>,<span class="at">nb=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>)<span class="sc">/</span><span class="dv">38</span>) <span class="ot">-&gt;</span> tmb_nb</span>
<span id="cb40-38"><a href="#cb40-38" aria-hidden="true" tabindex="-1"></a>neo_nes <span class="ot">&lt;-</span> <span class="fu">left_join</span>(neo_nes,tmb_nb)</span>
<span id="cb40-39"><a href="#cb40-39" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Joining, by = &quot;sample&quot;</span></span>
<span id="cb40-40"><a href="#cb40-40" aria-hidden="true" tabindex="-1"></a>model2 <span class="ot">&lt;-</span> <span class="fu">glm</span>(obj <span class="sc">~</span> tmb, <span class="at">data =</span> neo_nes, <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>)</span>
<span id="cb40-41"><a href="#cb40-41" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model2)</span>
<span id="cb40-42"><a href="#cb40-42" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-43"><a href="#cb40-43" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb40-44"><a href="#cb40-44" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; glm(formula = obj ~ tmb, family = &quot;binomial&quot;, data = neo_nes)</span></span>
<span id="cb40-45"><a href="#cb40-45" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-46"><a href="#cb40-46" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Deviance Residuals: </span></span>
<span id="cb40-47"><a href="#cb40-47" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Min       1Q   Median       3Q      Max  </span></span>
<span id="cb40-48"><a href="#cb40-48" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -1.5311  -0.7640  -0.7319   1.3947   1.7305  </span></span>
<span id="cb40-49"><a href="#cb40-49" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-50"><a href="#cb40-50" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb40-51"><a href="#cb40-51" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error z value Pr(&gt;|z|)    </span></span>
<span id="cb40-52"><a href="#cb40-52" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Intercept) -1.30753    0.29590  -4.419 9.92e-06 ***</span></span>
<span id="cb40-53"><a href="#cb40-53" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; tmb          0.06035    0.02584   2.335   0.0195 *  </span></span>
<span id="cb40-54"><a href="#cb40-54" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb40-55"><a href="#cb40-55" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb40-56"><a href="#cb40-56" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-57"><a href="#cb40-57" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Dispersion parameter for binomial family taken to be 1)</span></span>
<span id="cb40-58"><a href="#cb40-58" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-59"><a href="#cb40-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Null deviance: 118.34  on 96  degrees of freedom</span></span>
<span id="cb40-60"><a href="#cb40-60" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Residual deviance: 110.98  on 95  degrees of freedom</span></span>
<span id="cb40-61"><a href="#cb40-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AIC: 114.98</span></span>
<span id="cb40-62"><a href="#cb40-62" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-63"><a href="#cb40-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Fisher Scoring iterations: 4</span></span>
<span id="cb40-64"><a href="#cb40-64" aria-hidden="true" tabindex="-1"></a>performance<span class="sc">::</span><span class="fu">performance_hosmer</span>(model2)</span>
<span id="cb40-65"><a href="#cb40-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Hosmer-Lemeshow Goodness-of-Fit Test</span></span>
<span id="cb40-66"><a href="#cb40-66" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-67"><a href="#cb40-67" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Chi-squared: 15.447</span></span>
<span id="cb40-68"><a href="#cb40-68" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            df:  8    </span></span>
<span id="cb40-69"><a href="#cb40-69" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       p-value:  0.051</span></span>
<span id="cb40-70"><a href="#cb40-70" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Summary: model seems to fit well.</span></span>
<span id="cb40-71"><a href="#cb40-71" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-72"><a href="#cb40-72" aria-hidden="true" tabindex="-1"></a>model3 <span class="ot">&lt;-</span> <span class="fu">glm</span>(obj <span class="sc">~</span> nb, <span class="at">data =</span> neo_nes, <span class="at">family =</span> <span class="st">&quot;binomial&quot;</span>)</span>
<span id="cb40-73"><a href="#cb40-73" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(model3)</span>
<span id="cb40-74"><a href="#cb40-74" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-75"><a href="#cb40-75" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Call:</span></span>
<span id="cb40-76"><a href="#cb40-76" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; glm(formula = obj ~ nb, family = &quot;binomial&quot;, data = neo_nes)</span></span>
<span id="cb40-77"><a href="#cb40-77" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-78"><a href="#cb40-78" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Deviance Residuals: </span></span>
<span id="cb40-79"><a href="#cb40-79" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Min       1Q   Median       3Q      Max  </span></span>
<span id="cb40-80"><a href="#cb40-80" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -1.5523  -0.7890  -0.7279   1.1717   1.7284  </span></span>
<span id="cb40-81"><a href="#cb40-81" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-82"><a href="#cb40-82" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Coefficients:</span></span>
<span id="cb40-83"><a href="#cb40-83" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;             Estimate Std. Error z value Pr(&gt;|z|)    </span></span>
<span id="cb40-84"><a href="#cb40-84" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Intercept)  -1.2858     0.3006  -4.278 1.89e-05 ***</span></span>
<span id="cb40-85"><a href="#cb40-85" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; nb            0.8816     0.4001   2.203   0.0276 *  </span></span>
<span id="cb40-86"><a href="#cb40-86" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; ---</span></span>
<span id="cb40-87"><a href="#cb40-87" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1</span></span>
<span id="cb40-88"><a href="#cb40-88" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-89"><a href="#cb40-89" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; (Dispersion parameter for binomial family taken to be 1)</span></span>
<span id="cb40-90"><a href="#cb40-90" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-91"><a href="#cb40-91" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;     Null deviance: 118.34  on 96  degrees of freedom</span></span>
<span id="cb40-92"><a href="#cb40-92" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Residual deviance: 112.47  on 95  degrees of freedom</span></span>
<span id="cb40-93"><a href="#cb40-93" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; AIC: 116.47</span></span>
<span id="cb40-94"><a href="#cb40-94" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-95"><a href="#cb40-95" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Number of Fisher Scoring iterations: 4</span></span>
<span id="cb40-96"><a href="#cb40-96" aria-hidden="true" tabindex="-1"></a>performance<span class="sc">::</span><span class="fu">performance_hosmer</span>(model3)</span>
<span id="cb40-97"><a href="#cb40-97" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; # Hosmer-Lemeshow Goodness-of-Fit Test</span></span>
<span id="cb40-98"><a href="#cb40-98" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb40-99"><a href="#cb40-99" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   Chi-squared: 9.389</span></span>
<span id="cb40-100"><a href="#cb40-100" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;            df: 8    </span></span>
<span id="cb40-101"><a href="#cb40-101" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;       p-value: 0.311</span></span>
<span id="cb40-102"><a href="#cb40-102" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Summary: model seems to fit well.</span></span>
<span id="cb40-103"><a href="#cb40-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-104"><a href="#cb40-104" aria-hidden="true" tabindex="-1"></a>p4 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(neo_nes, <span class="fu">aes</span>(<span class="at">x=</span>es, <span class="at">y=</span>obj)) <span class="sc">+</span></span>
<span id="cb40-105"><a href="#cb40-105" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span>.<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb40-106"><a href="#cb40-106" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">&quot;glm&quot;</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">fullrange=</span><span class="cn">TRUE</span>,</span>
<span id="cb40-107"><a href="#cb40-107" aria-hidden="true" tabindex="-1"></a>              <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span>binomial),<span class="at">color=</span><span class="st">&quot;red&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-108"><a href="#cb40-108" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb40-109"><a href="#cb40-109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb40-110"><a href="#cb40-110" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;ES&quot;</span>,<span class="at">title =</span> <span class="st">&quot;H-L test P value = 0.771&quot;</span>)</span>
<span id="cb40-111"><a href="#cb40-111" aria-hidden="true" tabindex="-1"></a>p5 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(neo_nes, <span class="fu">aes</span>(<span class="at">x=</span>tmb, <span class="at">y=</span>obj)) <span class="sc">+</span></span>
<span id="cb40-112"><a href="#cb40-112" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span>.<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb40-113"><a href="#cb40-113" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">&quot;glm&quot;</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">fullrange=</span><span class="cn">TRUE</span>,</span>
<span id="cb40-114"><a href="#cb40-114" aria-hidden="true" tabindex="-1"></a>              <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span>binomial),<span class="at">color=</span><span class="st">&quot;red&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-115"><a href="#cb40-115" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb40-116"><a href="#cb40-116" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb40-117"><a href="#cb40-117" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;TMB&quot;</span>,<span class="at">title =</span> <span class="st">&quot;H-L test P value = 0.051&quot;</span>)</span>
<span id="cb40-118"><a href="#cb40-118" aria-hidden="true" tabindex="-1"></a>p6 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(neo_nes, <span class="fu">aes</span>(<span class="at">x=</span>nb, <span class="at">y=</span>obj)) <span class="sc">+</span></span>
<span id="cb40-119"><a href="#cb40-119" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha=</span>.<span class="dv">5</span>) <span class="sc">+</span></span>
<span id="cb40-120"><a href="#cb40-120" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_smooth</span>(<span class="at">method=</span><span class="st">&quot;glm&quot;</span>, <span class="at">se=</span><span class="cn">FALSE</span>, <span class="at">fullrange=</span><span class="cn">TRUE</span>,</span>
<span id="cb40-121"><a href="#cb40-121" aria-hidden="true" tabindex="-1"></a>              <span class="at">method.args =</span> <span class="fu">list</span>(<span class="at">family=</span>binomial),<span class="at">color=</span><span class="st">&quot;red&quot;</span>)<span class="sc">+</span></span>
<span id="cb40-122"><a href="#cb40-122" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_bw</span>()<span class="sc">+</span></span>
<span id="cb40-123"><a href="#cb40-123" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.title.y =</span> <span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb40-124"><a href="#cb40-124" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x=</span><span class="st">&quot;Neoantigen burden&quot;</span>,<span class="at">title =</span> <span class="st">&quot;H-L test P value = 0.311&quot;</span>)</span>
<span id="cb40-125"><a href="#cb40-125" aria-hidden="true" tabindex="-1"></a>p7 <span class="ot">&lt;-</span> p4 <span class="sc">+</span> p5 <span class="sc">+</span> p6</span>
<span id="cb40-126"><a href="#cb40-126" aria-hidden="true" tabindex="-1"></a>p7</span>
<span id="cb40-127"><a href="#cb40-127" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `geom_smooth()` using formula &#39;y ~ x&#39;</span></span>
<span id="cb40-128"><a href="#cb40-128" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `geom_smooth()` using formula &#39;y ~ x&#39;</span></span>
<span id="cb40-129"><a href="#cb40-129" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; `geom_smooth()` using formula &#39;y ~ x&#39;</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-23-1.png" width="2400" /></p>
<p>The H-L test P-value of TMB is 0.051, close to 0.05, implicate the difference between prediction and expectation is close to significant. The H-L test P-value of ESCCF is 0.771 , higher than the H-L test P-value of TMB and neoantigen count, suggesting ESCCF is more suitable for predicting prognosis of patients than TMB and neoantigen count.</p>
</div>
</div>
<div id="supplementary-analyses" class="section level1">
<h1>Supplementary analyses</h1>
<p>Number of the patient with CCF information and number of patient with at least one neoantigenic mutation and subclonal mutation (CCF&lt;0.6) are shown for each cancer type and number of the patient with mRNA expression information and number of patient with at least one neoantigenic mutation and accompanied mRNA expression information are shown for each cancer type:</p>
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="do">###sample statistics</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/all_mut_ccf_tpm.rds&quot;</span>)</span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">ccf=</span>ccf_hat) <span class="sc">%&gt;%</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">neo=</span><span class="fu">ifelse</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>))</span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a>samples_has_subclonal <span class="ot">&lt;-</span> all_mut_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(ccf<span class="sc">&lt;</span><span class="fl">0.6</span>) <span class="sc">%&gt;%</span> <span class="fu">select</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(sample)</span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a>all_mut_ccf <span class="sc">%&gt;%</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">c_n=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;yes&quot;</span>),<span class="at">c_m=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;no&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">type=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> samples_has_subclonal<span class="sc">$</span>sample,<span class="st">&quot;yes&quot;</span>,<span class="st">&quot;no&quot;</span>)) <span class="ot">-&gt;</span> sample_summ</span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a>sample_summ<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(sample_summ<span class="sc">$</span>sample)</span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a>sample_summ <span class="sc">%&gt;%</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sample_counts=</span><span class="fu">n</span>(),</span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a>            <span class="at">sample_with_neoantigen_counts=</span><span class="fu">sum</span>(c_n<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> c_m <span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> type<span class="sc">==</span><span class="st">&quot;yes&quot;</span>)) <span class="ot">-&gt;</span> cancer_summ</span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> cancer_summ <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample_counts)</span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a>cancer_summ<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">factor</span>(cancer_summ<span class="sc">$</span>cancer,<span class="at">levels =</span> cancer_summ<span class="sc">$</span>cancer)</span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="sc">%&gt;%</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">`</span><span class="at">All samples</span><span class="st">`</span><span class="ot">=</span>sample_counts,</span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a>                <span class="st">`</span><span class="at">Samples with &gt;= 1 neoantigen and have subclonal mutations</span><span class="st">`</span><span class="ot">=</span>sample_with_neoantigen_counts) <span class="sc">%&gt;%</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;All samples&quot;</span>,<span class="st">&quot;Samples with &gt;= 1 neoantigen and have subclonal mutations&quot;</span>),</span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">&quot;type&quot;</span>,</span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">&quot;counts&quot;</span>) <span class="ot">-&gt;</span> cancer_summ_longer</span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">pyramid_chart</span>(<span class="at">data =</span> cancer_summ_longer, <span class="at">x =</span> cancer, <span class="at">y =</span> counts, <span class="at">group =</span> type)</span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a>all_mut_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/all_mut_tpm_not_filter.rds&quot;</span>)</span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a>all_mut_exp <span class="sc">%&gt;%</span></span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">n_a=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;not_neo&quot;</span>),<span class="at">n_c=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>)) <span class="ot">-&gt;</span> sample_summ</span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a>sample_summ<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(sample_summ<span class="sc">$</span>sample)</span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a>sample_summ <span class="sc">%&gt;%</span></span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">sample_counts=</span><span class="fu">n</span>(),</span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a>            <span class="at">sample_with_neoantigen_counts=</span><span class="fu">sum</span>(n_a<span class="sc">&gt;=</span><span class="dv">1</span> <span class="sc">&amp;</span> n_c <span class="sc">&gt;=</span><span class="dv">1</span>)) <span class="ot">-&gt;</span> cancer_summ</span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="ot">&lt;-</span> cancer_summ <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample_counts)</span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a>cancer_summ<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">factor</span>(cancer_summ<span class="sc">$</span>cancer,<span class="at">levels =</span> cancer_summ<span class="sc">$</span>cancer)</span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a>cancer_summ <span class="sc">%&gt;%</span></span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">rename</span>(<span class="st">`</span><span class="at">All samples</span><span class="st">`</span><span class="ot">=</span>sample_counts,</span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a>                <span class="st">`</span><span class="at">Samples with &gt;= 1 neoantigen</span><span class="st">`</span><span class="ot">=</span>sample_with_neoantigen_counts) <span class="sc">%&gt;%</span></span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(<span class="st">&quot;All samples&quot;</span>,<span class="st">&quot;Samples with &gt;= 1 neoantigen&quot;</span>),</span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to=</span><span class="st">&quot;type&quot;</span>,</span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to=</span><span class="st">&quot;counts&quot;</span>) <span class="ot">-&gt;</span> cancer_summ_longer</span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fu">pyramid_chart</span>(<span class="at">data =</span> cancer_summ_longer, <span class="at">x =</span> cancer, <span class="at">y =</span> counts, <span class="at">group =</span> type)</span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-64"><a href="#cb41-64" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-65"><a href="#cb41-65" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning: `expand_scale()` is deprecated; use `expansion()` instead.</span></span>
<span id="cb41-66"><a href="#cb41-66" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-67"><a href="#cb41-67" aria-hidden="true" tabindex="-1"></a>p1 </span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-24-1.png" width="2400" /></p>
<div class="sourceCode" id="cb42"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a>p2</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-24-2.png" width="2400" /></p>
<p>We also calculated the significant sample in each cancer type:</p>
<div class="sourceCode" id="cb43"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>pancancer_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/neo_nes_ccf06_1_remove_driver_samples_addp.rds&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p <span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;sig_neg&quot;</span>,<span class="st">&quot;others&quot;</span>))</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>pancancer_exp <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/nes_exp_addp.rds&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p_value <span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;sig_neg&quot;</span>,<span class="st">&quot;others&quot;</span>))</span>
<span id="cb43-5"><a href="#cb43-5" aria-hidden="true" tabindex="-1"></a>pancancer_ccf<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(pancancer_ccf<span class="sc">$</span>sample)</span>
<span id="cb43-6"><a href="#cb43-6" aria-hidden="true" tabindex="-1"></a>pancancer_exp<span class="sc">$</span>cancer <span class="ot">&lt;-</span> <span class="fu">get_cancer_type</span>(pancancer_exp<span class="sc">$</span>sample)</span>
<span id="cb43-7"><a href="#cb43-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-8"><a href="#cb43-8" aria-hidden="true" tabindex="-1"></a>get_plot <span class="ot">&lt;-</span> <span class="cf">function</span>(dt,total_sig_sample){</span>
<span id="cb43-9"><a href="#cb43-9" aria-hidden="true" tabindex="-1"></a>  sig <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span></span>
<span id="cb43-10"><a href="#cb43-10" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(es_type<span class="sc">==</span><span class="st">&quot;sig_neg&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb43-11"><a href="#cb43-11" aria-hidden="true" tabindex="-1"></a>    dplyr<span class="sc">::</span><span class="fu">group_by</span>(cancer) <span class="sc">%&gt;%</span></span>
<span id="cb43-12"><a href="#cb43-12" aria-hidden="true" tabindex="-1"></a>    <span class="fu">summarise</span>(<span class="at">counts=</span><span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb43-13"><a href="#cb43-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">arrange</span>(counts) <span class="sc">%&gt;%</span></span>
<span id="cb43-14"><a href="#cb43-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">cancer=</span><span class="fu">factor</span>(cancer,<span class="at">levels =</span> cancer)) <span class="sc">%&gt;%</span></span>
<span id="cb43-15"><a href="#cb43-15" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Sample proportion</span><span class="st">`</span><span class="ot">=</span> counts<span class="sc">/</span>total_sig_sample) <span class="sc">%&gt;%</span></span>
<span id="cb43-16"><a href="#cb43-16" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">label=</span><span class="fu">paste</span>(<span class="st">&quot;frac(&quot;</span>,counts,<span class="st">&quot;,&quot;</span>,total_sig_sample,<span class="st">&quot;)&quot;</span>,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>))</span>
<span id="cb43-17"><a href="#cb43-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-18"><a href="#cb43-18" aria-hidden="true" tabindex="-1"></a>  p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="at">data=</span>sig,<span class="fu">aes</span>(<span class="at">x=</span>cancer,<span class="at">y=</span><span class="st">`</span><span class="at">Sample proportion</span><span class="st">`</span>))<span class="sc">+</span></span>
<span id="cb43-19"><a href="#cb43-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_bar</span>(<span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>cancer,<span class="at">y=</span><span class="st">`</span><span class="at">Sample proportion</span><span class="st">`</span>),<span class="at">stat =</span> <span class="st">&quot;identity&quot;</span>)<span class="sc">+</span></span>
<span id="cb43-20"><a href="#cb43-20" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme_classic</span>()<span class="sc">+</span></span>
<span id="cb43-21"><a href="#cb43-21" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust =</span> <span class="dv">1</span>))<span class="sc">+</span></span>
<span id="cb43-22"><a href="#cb43-22" aria-hidden="true" tabindex="-1"></a>    <span class="fu">scale_y_continuous</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fu">c</span>(<span class="dv">0</span>, .<span class="dv">1</span>)))<span class="sc">+</span><span class="do">##remove blank in the bottom</span></span>
<span id="cb43-23"><a href="#cb43-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">axis.title.x=</span><span class="fu">element_blank</span>())<span class="sc">+</span></span>
<span id="cb43-24"><a href="#cb43-24" aria-hidden="true" tabindex="-1"></a>    <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">label=</span>label), <span class="at">position=</span><span class="fu">position_dodge</span>(<span class="at">width=</span><span class="fl">0.9</span>), <span class="at">vjust=</span><span class="sc">-</span><span class="fl">0.25</span>,<span class="at">size=</span><span class="dv">2</span>,<span class="at">parse=</span><span class="cn">TRUE</span>)</span>
<span id="cb43-25"><a href="#cb43-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(p1)</span>
<span id="cb43-26"><a href="#cb43-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-27"><a href="#cb43-27" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb43-28"><a href="#cb43-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb43-29"><a href="#cb43-29" aria-hidden="true" tabindex="-1"></a><span class="fu">get_plot</span>(pancancer_ccf,<span class="fu">sum</span>(pancancer_ccf<span class="sc">$</span>es_type<span class="sc">==</span><span class="st">&quot;sig_neg&quot;</span>))</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-25-1.png" width="2400" /></p>
<div class="sourceCode" id="cb44"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="fu">get_plot</span>(pancancer_exp,<span class="fu">sum</span>(pancancer_exp<span class="sc">$</span>es_type<span class="sc">==</span><span class="st">&quot;sig_neg&quot;</span>))</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-25-2.png" width="2400" /></p>
<p>Pan-cancer sample distribution in pie plot:</p>
<div class="sourceCode" id="cb45"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a>pancancer_ccf <span class="ot">&lt;-</span> pancancer_ccf <span class="sc">%&gt;%</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type2=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;ES &lt; 0 &amp; P &lt; 0.05&quot;</span>,<span class="st">&quot;Others&quot;</span>))</span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a>pancancer_exp <span class="ot">&lt;-</span> pancancer_exp <span class="sc">%&gt;%</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">es_type2=</span><span class="fu">ifelse</span>(es<span class="sc">&lt;</span><span class="dv">0</span> <span class="sc">&amp;</span> p_value<span class="sc">&lt;</span><span class="fl">0.05</span>,<span class="st">&quot;ES &lt; 0 &amp; P &lt; 0.05&quot;</span>,<span class="st">&quot;Others&quot;</span>))</span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a>ccf_es <span class="ot">&lt;-</span> pancancer_ccf<span class="sc">$</span>es</span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a>ccf_es1 <span class="ot">&lt;-</span> pancancer_ccf<span class="sc">$</span>es_type2</span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow=</span><span class="fu">c</span>(<span class="dv">4</span>,<span class="dv">1</span>))</span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a>op <span class="ot">&lt;-</span> <span class="fu">par</span>(<span class="at">mar =</span> <span class="fu">rep</span>(<span class="dv">0</span>, <span class="dv">4</span>))</span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pie</span>(ccf_es,<span class="at">expression =</span> <span class="st">&quot;ccf_es&lt;0&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;ES (CCF) &lt; 0&quot;</span>,<span class="st">&quot;ES (CCF) &gt;= 0&quot;</span>))</span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pie</span>(ccf_es1,<span class="at">expression =</span> <span class="st">&quot;ccf_es1==&#39;ES &lt; 0 &amp; P &lt; 0.05&#39;&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;ES (CCF) &lt; 0 &amp; P &lt; 0.05&quot;</span>,<span class="st">&quot;Others&quot;</span>))</span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a>exp_es <span class="ot">&lt;-</span> pancancer_exp<span class="sc">$</span>es</span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a>exp_es1 <span class="ot">&lt;-</span> pancancer_exp<span class="sc">$</span>es_type2</span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pie</span>(exp_es,<span class="at">expression =</span> <span class="st">&quot;exp_es&lt;0&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;ES (EXP) &lt; 0&quot;</span>,<span class="st">&quot;ES (EXP) &gt;= 0&quot;</span>))</span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; NULL</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pie</span>(exp_es1,<span class="at">expression =</span> <span class="st">&quot;exp_es1==&#39;ES &lt; 0 &amp; P &lt; 0.05&#39;&quot;</span>,<span class="fu">c</span>(<span class="st">&quot;ES (EXP) &lt; 0 &amp; P &lt; 0.05&quot;</span>,<span class="st">&quot;Others&quot;</span>))</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-26-1.png" width="2400" /></p>
<pre><code>#&gt; NULL</code></pre>
<p>The immune-thearpy dataset sample distribution and clinical parameters:</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a>all_sample_exp_ccf <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/all_mut_ccf_ici.rds&quot;</span>)</span>
<span id="cb47-2"><a href="#cb47-2" aria-hidden="true" tabindex="-1"></a>ccf <span class="ot">&lt;-</span> all_sample_exp_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(cancer_cell_frac))</span>
<span id="cb47-3"><a href="#cb47-3" aria-hidden="true" tabindex="-1"></a>exp <span class="ot">&lt;-</span> all_sample_exp_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(exp))</span>
<span id="cb47-4"><a href="#cb47-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-5"><a href="#cb47-5" aria-hidden="true" tabindex="-1"></a>all_sample <span class="ot">&lt;-</span> <span class="fu">union</span>(<span class="fu">names</span>(<span class="fu">table</span>(ccf<span class="sc">$</span>sample)),<span class="fu">names</span>(<span class="fu">table</span>(exp<span class="sc">$</span>sample)))</span>
<span id="cb47-6"><a href="#cb47-6" aria-hidden="true" tabindex="-1"></a>both <span class="ot">&lt;-</span> <span class="fu">intersect</span>(<span class="fu">names</span>(<span class="fu">table</span>(ccf<span class="sc">$</span>sample)),<span class="fu">names</span>(<span class="fu">table</span>(exp<span class="sc">$</span>sample)))</span>
<span id="cb47-7"><a href="#cb47-7" aria-hidden="true" tabindex="-1"></a>WES <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(<span class="fu">table</span>(ccf<span class="sc">$</span>sample)),<span class="fu">names</span>(<span class="fu">table</span>(exp<span class="sc">$</span>sample)))</span>
<span id="cb47-8"><a href="#cb47-8" aria-hidden="true" tabindex="-1"></a>RNA <span class="ot">&lt;-</span> <span class="fu">setdiff</span>(<span class="fu">names</span>(<span class="fu">table</span>(exp<span class="sc">$</span>sample)),<span class="fu">names</span>(<span class="fu">table</span>(ccf<span class="sc">$</span>sample)))</span>
<span id="cb47-9"><a href="#cb47-9" aria-hidden="true" tabindex="-1"></a>all_clinical <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/all_clinical.rds&quot;</span>)</span>
<span id="cb47-10"><a href="#cb47-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-11"><a href="#cb47-11" aria-hidden="true" tabindex="-1"></a>response <span class="ot">&lt;-</span> all_clinical <span class="sc">%&gt;%</span> <span class="fu">filter</span>(response2<span class="sc">==</span><span class="st">&quot;response&quot;</span>)</span>
<span id="cb47-12"><a href="#cb47-12" aria-hidden="true" tabindex="-1"></a>no_response <span class="ot">&lt;-</span> all_clinical <span class="sc">%&gt;%</span> <span class="fu">filter</span>(response2<span class="sc">!=</span><span class="st">&quot;response&quot;</span>)</span>
<span id="cb47-13"><a href="#cb47-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-14"><a href="#cb47-14" aria-hidden="true" tabindex="-1"></a>willy_tre <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(<span class="st">&quot;../data/Immunotherapy/willy_clinical.csv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-15"><a href="#cb47-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">Patient ID</span><span class="st">`</span>,Treatment,Gender,Age) <span class="sc">%&gt;%</span></span>
<span id="cb47-16"><a href="#cb47-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample=</span><span class="st">`</span><span class="at">Patient ID</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">paste</span>(<span class="st">&quot;willy_&quot;</span>,sample,<span class="at">sep =</span> <span class="st">&quot;&quot;</span>)) <span class="sc">%&gt;%</span> <span class="fu">na.omit</span>()</span>
<span id="cb47-17"><a href="#cb47-17" aria-hidden="true" tabindex="-1"></a><span class="do">##nadeem Nivolumab</span></span>
<span id="cb47-18"><a href="#cb47-18" aria-hidden="true" tabindex="-1"></a>nadeem_tre <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample=</span>all_sample[<span class="fu">grepl</span>(<span class="st">&quot;nadeem_&quot;</span>,all_sample)],<span class="at">Treatment=</span><span class="st">&quot;Nivolumab&quot;</span>,</span>
<span id="cb47-19"><a href="#cb47-19" aria-hidden="true" tabindex="-1"></a>                         <span class="at">Gender=</span><span class="cn">NA</span>,<span class="at">Age=</span><span class="cn">NA</span>,<span class="at">stringsAsFactors =</span> F)</span>
<span id="cb47-20"><a href="#cb47-20" aria-hidden="true" tabindex="-1"></a>liu_clinical <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/liu_clinical.rds&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-21"><a href="#cb47-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X,Tx,<span class="st">`</span><span class="at">gender..Male.1..Female.0.</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-22"><a href="#cb47-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Gender=</span><span class="fu">ifelse</span>(<span class="st">`</span><span class="at">gender..Male.1..Female.0.</span><span class="st">`</span><span class="sc">==</span><span class="dv">1</span>,<span class="st">&quot;M&quot;</span>,<span class="st">&quot;F&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-23"><a href="#cb47-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="st">`</span><span class="at">gender..Male.1..Female.0.</span><span class="st">`</span>) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">Age=</span><span class="cn">NA</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-24"><a href="#cb47-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">sample=</span>X,<span class="at">Treatment=</span>Tx) <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">sample=</span><span class="fu">paste0</span>(<span class="st">&quot;liu_&quot;</span>,sample)) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;_Patient&quot;</span>,sample))</span>
<span id="cb47-25"><a href="#cb47-25" aria-hidden="true" tabindex="-1"></a>all_tre <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">list</span>(willy_tre,nadeem_tre,liu_clinical))</span>
<span id="cb47-26"><a href="#cb47-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-27"><a href="#cb47-27" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">sample=</span>all_sample,</span>
<span id="cb47-28"><a href="#cb47-28" aria-hidden="true" tabindex="-1"></a>                 <span class="at">stringsAsFactors =</span> F) <span class="sc">%&gt;%</span></span>
<span id="cb47-29"><a href="#cb47-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">Data type</span><span class="st">`</span><span class="ot">=</span> <span class="fu">case_when</span>(</span>
<span id="cb47-30"><a href="#cb47-30" aria-hidden="true" tabindex="-1"></a>    sample <span class="sc">%in%</span> both <span class="sc">~</span> <span class="st">&quot;Both&quot;</span>,</span>
<span id="cb47-31"><a href="#cb47-31" aria-hidden="true" tabindex="-1"></a>    sample <span class="sc">%in%</span> WES <span class="sc">~</span> <span class="st">&quot;WES&quot;</span>,</span>
<span id="cb47-32"><a href="#cb47-32" aria-hidden="true" tabindex="-1"></a>    sample <span class="sc">%in%</span> RNA <span class="sc">~</span> <span class="st">&quot;RNA-Seq&quot;</span></span>
<span id="cb47-33"><a href="#cb47-33" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb47-34"><a href="#cb47-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Response=</span><span class="fu">ifelse</span>(sample <span class="sc">%in%</span> response<span class="sc">$</span>sample,<span class="st">&quot;Responder&quot;</span>,<span class="st">&quot;Non-Responder&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb47-35"><a href="#cb47-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort=</span><span class="fu">gsub</span>(<span class="st">&quot;_.*&quot;</span>,<span class="st">&quot;&quot;</span>,sample)) <span class="sc">%&gt;%</span></span>
<span id="cb47-36"><a href="#cb47-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,all_tre,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span>
<span id="cb47-37"><a href="#cb47-37" aria-hidden="true" tabindex="-1"></a>dt <span class="ot">&lt;-</span> dt <span class="sc">%&gt;%</span> <span class="fu">arrange</span>(sample)</span>
<span id="cb47-38"><a href="#cb47-38" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-39"><a href="#cb47-39" aria-hidden="true" tabindex="-1"></a>dt_cli <span class="ot">&lt;-</span> <span class="fu">left_join</span>(dt,all_clinical,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span>
<span id="cb47-40"><a href="#cb47-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-41"><a href="#cb47-41" aria-hidden="true" tabindex="-1"></a>cohort_col <span class="ot">&lt;-</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">3</span>,<span class="st">&quot;Accent&quot;</span>)</span>
<span id="cb47-42"><a href="#cb47-42" aria-hidden="true" tabindex="-1"></a>data_type_col <span class="ot">&lt;-</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">3</span>,<span class="st">&quot;Dark2&quot;</span>)</span>
<span id="cb47-43"><a href="#cb47-43" aria-hidden="true" tabindex="-1"></a>Response_col <span class="ot">&lt;-</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">3</span>,<span class="st">&quot;Set1&quot;</span>)[<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>]</span>
<span id="cb47-44"><a href="#cb47-44" aria-hidden="true" tabindex="-1"></a>Treatment_col <span class="ot">&lt;-</span> RColorBrewer<span class="sc">::</span><span class="fu">brewer.pal</span>(<span class="dv">3</span>,<span class="st">&quot;Paired&quot;</span>)</span>
<span id="cb47-45"><a href="#cb47-45" aria-hidden="true" tabindex="-1"></a>ha <span class="ot">=</span> <span class="fu">HeatmapAnnotation</span>(</span>
<span id="cb47-46"><a href="#cb47-46" aria-hidden="true" tabindex="-1"></a>  <span class="at">Cohort=</span>dt<span class="sc">$</span>cohort,</span>
<span id="cb47-47"><a href="#cb47-47" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Data type</span><span class="st">`</span> <span class="ot">=</span> dt<span class="sc">$</span><span class="st">`</span><span class="at">Data type</span><span class="st">`</span>,</span>
<span id="cb47-48"><a href="#cb47-48" aria-hidden="true" tabindex="-1"></a>  <span class="at">Response =</span> dt<span class="sc">$</span>Response,</span>
<span id="cb47-49"><a href="#cb47-49" aria-hidden="true" tabindex="-1"></a>  <span class="at">Treatment=</span>dt<span class="sc">$</span>Treatment,</span>
<span id="cb47-50"><a href="#cb47-50" aria-hidden="true" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">list</span>(<span class="at">Cohort=</span><span class="fu">c</span>(<span class="st">&quot;liu&quot;</span><span class="ot">=</span>cohort_col[<span class="dv">1</span>],<span class="st">&quot;nadeem&quot;</span><span class="ot">=</span>cohort_col[<span class="dv">2</span>],<span class="st">&quot;willy&quot;</span><span class="ot">=</span>cohort_col[<span class="dv">3</span>]),</span>
<span id="cb47-51"><a href="#cb47-51" aria-hidden="true" tabindex="-1"></a>             <span class="st">`</span><span class="at">Data type</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">c</span>(<span class="st">&quot;WES&quot;</span> <span class="ot">=</span>data_type_col[<span class="dv">1</span>], <span class="st">&quot;RNA-Seq&quot;</span> <span class="ot">=</span> data_type_col[<span class="dv">2</span>],<span class="st">&quot;Both&quot;</span><span class="ot">=</span>data_type_col[<span class="dv">3</span>]),</span>
<span id="cb47-52"><a href="#cb47-52" aria-hidden="true" tabindex="-1"></a>             <span class="at">Response =</span> <span class="fu">c</span>(<span class="st">&quot;Responder&quot;</span><span class="ot">=</span>Response_col[<span class="dv">1</span>],<span class="st">&quot;Non-Responder&quot;</span><span class="ot">=</span>Response_col[<span class="dv">2</span>]),</span>
<span id="cb47-53"><a href="#cb47-53" aria-hidden="true" tabindex="-1"></a>             <span class="at">Treatment=</span><span class="fu">c</span>(<span class="st">&quot;Nivolumab&quot;</span><span class="ot">=</span>Treatment_col[<span class="dv">1</span>],<span class="st">&quot;Pembrolizumab&quot;</span><span class="ot">=</span>Treatment_col[<span class="dv">2</span>])),</span>
<span id="cb47-54"><a href="#cb47-54" aria-hidden="true" tabindex="-1"></a>  <span class="at">annotation_name_side=</span><span class="st">&quot;left&quot;</span>,<span class="at">height =</span> <span class="fu">unit</span>(<span class="dv">300</span>, <span class="st">&quot;cm&quot;</span>),<span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">20</span>, <span class="st">&quot;cm&quot;</span>)</span>
<span id="cb47-55"><a href="#cb47-55" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb47-56"><a href="#cb47-56" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb47-57"><a href="#cb47-57" aria-hidden="true" tabindex="-1"></a>lgd1 <span class="ot">=</span> <span class="fu">Legend</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;liu&quot;</span>,<span class="st">&quot;nadeem&quot;</span>,<span class="st">&quot;willy&quot;</span>), <span class="at">legend_gp =</span> <span class="fu">gpar</span>(<span class="at">fill=</span><span class="fu">c</span>(cohort_col[<span class="dv">1</span>],cohort_col[<span class="dv">2</span>],cohort_col[<span class="dv">3</span>])),</span>
<span id="cb47-58"><a href="#cb47-58" aria-hidden="true" tabindex="-1"></a>              <span class="at">title =</span> <span class="st">&quot;Cohort&quot;</span>, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb47-59"><a href="#cb47-59" aria-hidden="true" tabindex="-1"></a>lgd2 <span class="ot">=</span> <span class="fu">Legend</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;WES&quot;</span>,<span class="st">&quot;RNA-Seq&quot;</span>,<span class="st">&quot;Both&quot;</span>), <span class="at">legend_gp =</span> <span class="fu">gpar</span>(<span class="at">fill=</span><span class="fu">c</span>(data_type_col[<span class="dv">1</span>],data_type_col[<span class="dv">2</span>],data_type_col[<span class="dv">3</span>])),</span>
<span id="cb47-60"><a href="#cb47-60" aria-hidden="true" tabindex="-1"></a>              <span class="at">title =</span> <span class="st">&quot;Data type&quot;</span>, <span class="at">ncol =</span> <span class="dv">3</span>)</span>
<span id="cb47-61"><a href="#cb47-61" aria-hidden="true" tabindex="-1"></a>lgd3 <span class="ot">=</span> <span class="fu">Legend</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Responder&quot;</span>,<span class="st">&quot;Non-Responder&quot;</span>), <span class="at">legend_gp =</span> <span class="fu">gpar</span>(<span class="at">fill=</span><span class="fu">c</span>(Response_col[<span class="dv">1</span>],Response_col[<span class="dv">2</span>])),</span>
<span id="cb47-62"><a href="#cb47-62" aria-hidden="true" tabindex="-1"></a>              <span class="at">title =</span> <span class="st">&quot;Response&quot;</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb47-63"><a href="#cb47-63" aria-hidden="true" tabindex="-1"></a>lgd4 <span class="ot">=</span> <span class="fu">Legend</span>(<span class="at">labels =</span> <span class="fu">c</span>(<span class="st">&quot;Nivolumab&quot;</span>,<span class="st">&quot;Pembrolizumab&quot;</span>), <span class="at">legend_gp =</span> <span class="fu">gpar</span>(<span class="at">fill=</span><span class="fu">c</span>(Treatment_col[<span class="dv">1</span>],Treatment_col[<span class="dv">2</span>],Treatment_col[<span class="dv">3</span>])),</span>
<span id="cb47-64"><a href="#cb47-64" aria-hidden="true" tabindex="-1"></a>              <span class="at">title =</span> <span class="st">&quot;Treatment&quot;</span>, <span class="at">ncol =</span> <span class="dv">2</span>)</span>
<span id="cb47-65"><a href="#cb47-65" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(ha, <span class="at">test =</span> T)</span>
<span id="cb47-66"><a href="#cb47-66" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">=</span> <span class="fu">packLegend</span>(lgd1, lgd2,lgd3,lgd4)</span>
<span id="cb47-67"><a href="#cb47-67" aria-hidden="true" tabindex="-1"></a><span class="fu">draw</span>(pd,<span class="at">x =</span> <span class="fu">unit</span>(<span class="dv">14</span>, <span class="st">&quot;cm&quot;</span>), <span class="at">y =</span> <span class="fu">unit</span>(<span class="dv">6</span>, <span class="st">&quot;cm&quot;</span>), <span class="at">just =</span> <span class="fu">c</span>(<span class="st">&quot;left&quot;</span>, <span class="st">&quot;bottom&quot;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb48"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;sample_statistics.png&quot;</span>)</span></code></pre></div>
<p><img src="sample_statistics.png" width="565" /></p>
<div class="sourceCode" id="cb49"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb49-1"><a href="#cb49-1" aria-hidden="true" tabindex="-1"></a>all_sample_exp_ccf <span class="ot">&lt;-</span> <span class="fu">left_join</span>(all_sample_exp_ccf,dt_cli,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>)</span>
<span id="cb49-2"><a href="#cb49-2" aria-hidden="true" tabindex="-1"></a>all_sample_exp_ccf <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">TMB=</span><span class="fu">log</span>((<span class="fu">n</span>()<span class="sc">/</span><span class="dv">38</span>)<span class="sc">+</span><span class="dv">1</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-3"><a href="#cb49-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,dt_cli,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>) <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(cohort)) <span class="ot">-&gt;</span> all_tmb</span>
<span id="cb49-4"><a href="#cb49-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-5"><a href="#cb49-5" aria-hidden="true" tabindex="-1"></a>all_tmb <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cohort) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">tmb_median=</span><span class="fu">median</span>(TMB),<span class="at">tmb_min=</span><span class="fu">min</span>(TMB),<span class="at">max_tmb=</span><span class="fu">max</span>(TMB))</span>
<span id="cb49-6"><a href="#cb49-6" aria-hidden="true" tabindex="-1"></a>all_sample_exp_ccf <span class="sc">%&gt;%</span> <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">is.na</span>(cohort)) <span class="sc">%&gt;%</span></span>
<span id="cb49-7"><a href="#cb49-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(sample) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">neo_counts=</span><span class="fu">sum</span>(neo<span class="sc">==</span><span class="st">&quot;neo&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb49-8"><a href="#cb49-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(.,dt_cli,<span class="at">by=</span><span class="st">&quot;sample&quot;</span>)<span class="ot">-&gt;</span> all_neo</span>
<span id="cb49-9"><a href="#cb49-9" aria-hidden="true" tabindex="-1"></a>all_neo <span class="sc">%&gt;%</span> <span class="fu">group_by</span>(cohort) <span class="sc">%&gt;%</span> <span class="fu">summarise</span>(<span class="at">median_neo=</span><span class="fu">median</span>(neo_counts))</span>
<span id="cb49-10"><a href="#cb49-10" aria-hidden="true" tabindex="-1"></a><span class="do">########表格####</span></span>
<span id="cb49-11"><a href="#cb49-11" aria-hidden="true" tabindex="-1"></a>gt_input <span class="ot">&lt;-</span> <span class="fu">tibble</span>(</span>
<span id="cb49-12"><a href="#cb49-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">cohort=</span><span class="fu">c</span>(<span class="st">&quot;Liu et.al </span><span class="sc">\n</span><span class="st"> (2019)&quot;</span>,<span class="st">&quot;Hogo et.al </span><span class="sc">\n</span><span class="st"> (2016)&quot;</span>,<span class="st">&quot;Riaz et.al </span><span class="sc">\n</span><span class="st"> (2017)&quot;</span>),</span>
<span id="cb49-13"><a href="#cb49-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Tumor type</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;melanoma&quot;</span>,<span class="st">&quot;melanoma&quot;</span>,<span class="st">&quot;melanoma&quot;</span>),</span>
<span id="cb49-14"><a href="#cb49-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">Treatment=</span><span class="fu">c</span>(<span class="st">&quot;monotherapy&quot;</span>,<span class="st">&quot;monotherapy&quot;</span>,<span class="st">&quot;monotherapy&quot;</span>),</span>
<span id="cb49-15"><a href="#cb49-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">Strategy=</span><span class="fu">c</span>(<span class="st">&quot;WES RNA-seq&quot;</span>,<span class="st">&quot;WES RNA-seq&quot;</span>,<span class="st">&quot;WES RNA-seq&quot;</span>),</span>
<span id="cb49-16"><a href="#cb49-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">Patients=</span><span class="fu">c</span>(<span class="dv">130</span>,<span class="dv">37</span>,<span class="dv">56</span>),</span>
<span id="cb49-17"><a href="#cb49-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Number of men(%)</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;77(59.2%)&quot;</span>,<span class="st">&quot;26(70.3%)&quot;</span>,<span class="cn">NA</span>),</span>
<span id="cb49-18"><a href="#cb49-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Number of women(%)</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="st">&quot;53(40.8%)&quot;</span>,<span class="st">&quot;11(29.7%)&quot;</span>,<span class="cn">NA</span>),</span>
<span id="cb49-19"><a href="#cb49-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Median age, years</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="cn">NA</span>,<span class="dv">61</span>,<span class="cn">NA</span>),</span>
<span id="cb49-20"><a href="#cb49-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Median survial time(OS,month)</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="fl">19.4</span>,<span class="fl">18.3</span>,<span class="fl">17.4</span>),</span>
<span id="cb49-21"><a href="#cb49-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">TMB median</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="fl">1.42</span>,<span class="fl">2.49</span>,<span class="fl">1.86</span>),</span>
<span id="cb49-22"><a href="#cb49-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">`</span><span class="at">Neoantigen counts median</span><span class="st">`</span><span class="ot">=</span><span class="fu">c</span>(<span class="fl">28.5</span>,<span class="dv">77</span>,<span class="dv">52</span>)</span>
<span id="cb49-23"><a href="#cb49-23" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb49-24"><a href="#cb49-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb49-25"><a href="#cb49-25" aria-hidden="true" tabindex="-1"></a>gt_tbl <span class="ot">&lt;-</span> <span class="fu">gt</span>(<span class="at">data =</span> gt_input)</span>
<span id="cb49-26"><a href="#cb49-26" aria-hidden="true" tabindex="-1"></a>gt_input <span class="sc">%&gt;%</span></span>
<span id="cb49-27"><a href="#cb49-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gt</span>(<span class="at">rowname_col =</span> <span class="st">&quot;cohort&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb49-28"><a href="#cb49-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">cols_align</span>(</span>
<span id="cb49-29"><a href="#cb49-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">&quot;center&quot;</span></span>
<span id="cb49-30"><a href="#cb49-30" aria-hidden="true" tabindex="-1"></a>  ) <span class="ot">-&gt;</span> gt_tbl</span>
<span id="cb49-31"><a href="#cb49-31" aria-hidden="true" tabindex="-1"></a>gridExtra<span class="sc">::</span><span class="fu">grid.table</span>(gt_input)</span></code></pre></div>
<div class="sourceCode" id="cb50"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span><span class="fu">include_graphics</span>(<span class="st">&quot;sample_table.png&quot;</span>)</span></code></pre></div>
<p><img src="sample_table.png" width="641" /></p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="do">#####oncoprint############</span></span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a><span class="do">##liu</span></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a>liu_mutations <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/liu_mutations.rds&quot;</span>)</span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>liu_mutations <span class="ot">&lt;-</span> liu_mutations <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">change=</span><span class="fu">str_extract</span>(cDNA_Change,<span class="st">&quot;[A-Z]&gt;[A-Z]&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Tumor_Seq_Allele2=</span><span class="fu">gsub</span>(<span class="st">&quot;&gt;&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">str_extract</span>(change,<span class="st">&quot;&gt;[A-Z]&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Reference_Allele=</span><span class="fu">gsub</span>(<span class="st">&quot;&gt;&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="fu">str_extract</span>(change,<span class="st">&quot;[A-Z]&gt;&quot;</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Tumor_Sample_Barcode=</span>Patient)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a>liu_clinical <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;../data/Immunotherapy/liu_clinical.rds&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">grepl</span>(<span class="st">&quot;Patient&quot;</span>,X)) <span class="sc">%&gt;%</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(X,OS,dead) <span class="sc">%&gt;%</span></span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Tumor_Sample_Barcode=</span>X,<span class="at">Overall_Survival_time=</span>OS,<span class="at">Overall_Survival_Status=</span>dead)</span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>liu <span class="ot">&lt;-</span> <span class="fu">read.maf</span>(<span class="at">maf =</span> liu_mutations,</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">clinicalData =</span> liu_clinical,</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>                <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoplot</span>(<span class="at">maf =</span> liu,<span class="at">top=</span><span class="dv">10</span>,<span class="at">draw_titv =</span> <span class="cn">TRUE</span>)</span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Warning in titv(maf = maf, useSyn = TRUE, plot = FALSE): Non standard Ti/Tv</span></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; class: 1540TRUE</span></span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-31-1.png" width="2400" /></p>
<div class="sourceCode" id="cb52"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="do">##willy</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">as.list</span>(<span class="fu">paste0</span>(<span class="st">&quot;../data/Immunotherapy/willy/&quot;</span>,<span class="fu">dir</span>(<span class="st">&quot;../data/Immunotherapy/willy/&quot;</span>)))</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>willy_maf <span class="ot">&lt;-</span> <span class="fu">merge_mafs</span>(files)</span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Merging 38 MAF files</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Validating</span></span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Removed 1 duplicated variants</span></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Silent variants: 225812 </span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Summarizing</span></span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Possible FLAGS among top ten genes:</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TTN</span></span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   MUC16</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   FLG</span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Processing clinical data</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Missing clinical data</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Finished in 5.090s elapsed (3.860s cpu)</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoplot</span>(<span class="at">maf =</span> willy_maf,<span class="at">top=</span><span class="dv">10</span>,<span class="at">draw_titv =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-31-2.png" width="2400" /></p>
<div class="sourceCode" id="cb53"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a><span class="do">###nadeem</span></span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>files <span class="ot">&lt;-</span> <span class="fu">as.list</span>(<span class="fu">paste0</span>(<span class="st">&quot;../data/Immunotherapy/nadeem/&quot;</span>,<span class="fu">dir</span>(<span class="st">&quot;../data/Immunotherapy/nadeem/&quot;</span>)))</span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a>nadeem_maf <span class="ot">&lt;-</span> <span class="fu">merge_mafs</span>(files)</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; Merging 66 MAF files</span></span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Validating</span></span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Removed 2 duplicated variants</span></span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Silent variants: 84077 </span></span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Summarizing</span></span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Possible FLAGS among top ten genes:</span></span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   TTN</span></span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   MUC16</span></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt;   OBSCN</span></span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Processing clinical data</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; --Missing clinical data</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a><span class="co">#&gt; -Finished in 3.720s elapsed (2.930s cpu)</span></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a><span class="fu">oncoplot</span>(<span class="at">maf =</span> nadeem_maf,<span class="at">top=</span><span class="dv">10</span>,<span class="at">draw_titv =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="report_main_files/figure-html/unnamed-chunk-31-3.png" width="2400" /></p>
</div>
</div>

   
   
            
      

  <script>
    $(document).ready(function () {

			
 		
	    });
  </script>



    <!-- dynamically load mathjax for compatibility with self-contained -->
  <script>
    (function () {
	var script = document.createElement("script");
	script.type = "text/javascript";
	script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
	document.getElementsByTagName("head")[0].appendChild(script);
    })();
  </script>
  
</body>
</html>
