---
title: "Data download and preprocessing"
author: "Tao Wu"
date: "`r Sys.Date()`"
output:
  rmdformats::readthedown:
    highlight: kate
    lightbox: no
    mathjax: yes
    toc_depth: 3
---

```{r pre-process-setup, include=FALSE}
options(max.print = "75")
knitr::opts_chunk$set(echo = TRUE, comment = "#>", eval = TRUE, collapse = TRUE)
knitr::opts_knit$set(width = 75)
```
## Supplemental Analysis

### HLA Typing
Use [Opitype](https://github.com/FRED-2/OptiType) to do HLA typing for samples without HLA informations.

```{python, fish_bam, eval = FALSE}
rule all:
    input:
        fa = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_fished.fastq"

rule clean_bam:
    input: 
        bam = "/public/home/wangshx/wt/immune_therapy_dataset/willy_hugo_2016_cell/bam/SRR3083837.bam"
    output: 
        cleaned_bam = temp("/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_clean.bam")
    conda:
        "HLA.yaml"
    params:
        samtools = "view -b -F 4"
    shell: 
        "samtools {params.samtools} {input.bam} > {output.cleaned_bam}"

rule bam2fq:
    input:
        bam = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_clean.bam"
    output:
        fa = temp("/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837.fastq")
    conda:
        "HLA.yaml"
    shell:
        "samtools bam2fq {input.bam} > {output.fa}"

rule fished_bam:
    input: 
        fa = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837.fastq",
        ref = "/public/home/liuxs/anaconda3/pkgs/optitype-1.3.2-py27_3/share/optitype-1.3.2-3/data/hla_reference_dna.fasta"
    output: 
        bam = temp("/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_fished.bam")
    shell: 
        "/public/home/liuxs/anaconda3/envs/python3/bin/razers3 -tc 6 -i 95 -m 1 -dr 0 -o {output} {input.ref} {input.fa}"

rule bam2fq_fished:
    input: 
        bam = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_fished.bam"
    output: 
        fa = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/SRR3083837_fished.fastq"
    shell: 
        "samtools bam2fq {input.bam} > {output.fa}"
```
```{python, typing, eval = FALSE}
rule typing:
    input: 
        fa = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/test_fished.fastq",
        config = "/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/config.ini"
    params:
        op = "-d -o /public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/demo/HLA_typing/snakemake/TCGA-AB-2855"
    shell:
        "OptiTypePipeline.py -i {input.fa} {params.op} -c {input.config}" 
```

### Neoantige Prediction

Use [NetMHCpan-4.0](http://www.cbs.dtu.dk/services/NetMHCpan-4.0/) predicted neoantigen with pipeline [pVACseq](https://pvactools.readthedocs.io/en/latest/pvacseq.html)

```{bash, neoantigen_presiction, eval = FALSE}
#PBS -N TCGA-QH-A870-01A
#PBS -l walltime=100:00:00
#PBS -l nodes=1:ppn=8
#PBS -l mem=20gb
#PBS -S /bin/bash
#PBS -q normal_3
#PBS -j oe


if [ -f "/public/home/liuxs/anaconda3/etc/profile.d/conda.sh" ]; then
        . "/public/home/liuxs/anaconda3/etc/profile.d/conda.sh"
else
        export PATH="/public/home/liuxs/anaconda3/bin:$PATH"
fi

conda activate neopip

vcf=/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/vcf_annoated/pancancer/mhc-i/vcf

out_dir=/public/home/wangshx/wx/neoantigens_delption/find_neoantigens_delotion_signal/neo_pre/one_method/TCGA-QH-A870-01A

iedb_dir=/public/home/liuxs/biosoft/iedb

pvacseq run \
$vcf/TCGA-QH-A870-01A.annoated.vcf \
TCGA-QH-A870-01A \
HLA-A*02:01,HLA-A*02:01,HLA-B*13:02,HLA-B*51:01,HLA-C*04:01,HLA-C*14:02 \
NetMHCpan \
$out_dir \
-e 9 \
--iedb-install-directory $iedb_dir \
--fasta-size 50
```

### Calcultion Harmonic Mean
```{r, harmonic_mean, eval = FALSE}
harmonic_mean <- function(x) {
  1/mean(1/x,na.rm=T) 
}
```
```{r, cal_harmonic_mean, eval=FALSE}
source('~/neo_depletion/R/function/harmonic_mean.R')
cal_harmonic <- function(file, sep, header = T, 
                         stringsAsFactors = F, 
                         out = '~/harmonic.rds', ...){
  
  df <- data.table::fread(file = file, sep = sep, 
                          stringsAsFactors = stringsAsFactors, 
                          header = header)
  df <- df[!`Gene Name` == ""]
  
  
  df <- df %>% 
    mutate(
      index = paste(Tumor_Barcode, Chromosome, Stop, sep = ':')
    ) %>% 
    group_by(index, `HLA Allele`) %>% 
    summarise(
      MT_aff = min(`Best MT Score`),
      WT_aff = `Corresponding WT Score`[which(`Best MT Score` == min(`Best MT Score`))][1]
    ) %>% 
    group_by(index) %>% 
    summarise(
      MT_har_mean = harmonic_mean(MT_aff),
      WT_har_mean = harmonic_mean(WT_aff)
    ) %>% 
    ungroup()
  
  
  saveRDS(df, file = out)
}

```



