---
title: "tcga_pancancer_analysis"
author: "Tao Wu"
date: "`r Sys.Date()`"
output:   
  rmdformats::readthedown:
    highlight: kate
    lightbox: false
    toc_depth: 3
    mathjax: true
---

```{r tcga_pancancer_analysis-setup, include=FALSE}
options(max.print = "75")
knitr::opts_chunk$set(echo = TRUE, comment = "#>", eval = TRUE, collapse = TRUE,cache = FALSE)
knitr::opts_knit$set(width = 75)
```

```{r lib1,echo=TRUE,eval=TRUE,include=FALSE}
library(tidyverse)
library(ggpubr)
library(cowplot)
library(viridis)
library(NeoEnrichment)
library(gridGraphics)
library(survminer)
library(survival)
```

##  The existence of significant immunoediting signal

From our pan-cancer data, we can get two distributions: mutation counts of samples and IC50 values of neoantigens. 
We randomly selected 10000 counts from the distribution of mutation counts to construct simulated samples which each sample has its own mutation counts. For each mutations in a simulated sample, we randomly selected a IC50 value from IC50 distribution. Now we have 10000 samples, each sample has its mutation counts and each mutations has its IC50 values. Given that CCF or expression is a criterion for ranking, we just used the natural order to rank the simulated mutations. At last, we calculated NES for these simulated samples to get neutral distribution of NES.

The following code do this simulation step for each cancer type:

```{r simulation,eval=FALSE}
pancancer_neoantigen_exp_ccf <- readRDS("~/Immunoediting/data/pancancer_neoantigen_exp_ccf.rds")
pancancer_neoantigen_exp_ccf <- pancancer_neoantigen_exp_ccf %>%
  rename(sample=Tumor_Barcode,chromosome=chr,position=Stop,MT_mean=MT_har_mean) %>% 
  as.data.frame(stringsAsFactors=FALSE)
cancer_type <- names(table(pancancer_neoantigen_exp_ccf$cancer_type))
set.seed(202011104)
for (i in 1:length(cancer_type)){
  cancer <- pancancer_neoantigen_exp_ccf %>% 
    dplyr::filter(cancer_type==.env$cancer_type[i])
  cancer %>% group_by(sample) %>%
    summarise(mutation_counts=n()) -> mutation_counts
  ##sampling mutation counts
  sample_mt <- sample(mutation_counts$mutation_counts,10000,replace = T)
  
  sample <- vector("list",length = 10000)
  names(sample) <- c(1:10000)
  sample <- lapply(sample_mt,
                    function(x){
                      tmp <- data.frame(mutation_id=c(1:x),
                                        IC50=sample(cancer$MT_mean,x),##sample IC50 for mutations
                                        rank=c(x:1))
                      a <- nrow(tmp)
                      tmp <- tmp %>% 
                        mutate(rank=abs((a/2)-rank)+1)
                      return(tmp)
                    })
  
  result <- vector("list",length = 10000)
  names(result) <- c(1:10000)
  cl <- makeCluster(getOption("cl.cores", 12),type="FORK")
  result <- parLapply(cl=cl,sample,
                          function(x){
                            neoantigen_list <- x %>% 
                              filter(IC50<500) %>% 
                              select(mutation_id)
                            neoantigen_list <- neoantigen_list[[1]]
                            if(length(neoantigen_list)==0){
                              return(NA)
                            }else{
                              es <- NeoEnrichment::cales_simulation(x,neoantigen_list)
                              nes <- NeoEnrichment::cal_nes_simulation(es,neoantigen_list,x)
                              return(nes)
                            }
                          })
  stopCluster(cl)
  
  result <- Filter(function(x){length(x)>1},result)
  result_sim <- bind_rows(result)
  saveRDS(result_sim,
          file = paste0("~/Immunoediting/data/simulation/",cancer_type[i],"_simulation.rds"))
}
```
Then we bind all cancer type simulation data:

```{r bind_sim,eval=FALSE}
bind_all <- lapply(list.files("~/Immunoediting/data/simulation/"),
                   function(x){
                     re <- readRDS(paste0("~/Immunoediting/data/simulation/",x)) %>% 
                       mutate(cancer=gsub("_simulation.rds","",x))
                   })
pancancer_simulation_data <- bind_rows(bind_all)
saveRDS(pancancer_simulation_data,file = "~/Immunoediting/data/simulation/pancancer_simulation.rds")
```
This simulation can considered as neutral (or random) nes distribution, so we can compare our TCGA pancancer real nes distribution to this data. If the two distributions are significantly different, then we can conclude that the observed nes in TCGA real data is not random.

```{r compare_nes,eval=TRUE}
pancancer_simulation <- readRDS("~/Immunoediting/data/simulation/pancancer_simulation.rds")
pancancer_nes_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds")
pancancer_nes_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds")  

dt_ccf <- data.frame(type=c(rep("simulation",nrow(pancancer_simulation)),
                        rep("TCGA pancancer",nrow(pancancer_nes_ccf))),
                     NES=c(pancancer_simulation$nes,pancancer_nes_ccf$nes))
dt_exp <- data.frame(type=c(rep("simulation",nrow(pancancer_simulation)),
                            rep("TCGA pancancer",nrow(pancancer_nes_exp))),
                     NES=c(pancancer_simulation$nes,pancancer_nes_exp$nes))

p1 <- ggplot(dt_ccf, aes(x=type, y=NES, fill=type)) +
  scale_fill_manual(values=c("white", "#FC8D62"))+
  stat_compare_means(aes(x=type, y=NES))+
  geom_violin(width=0.8, alpha=0.2)+
  geom_boxplot(width=0.2) +
  guides(fill=F)+
  labs(title="CCF")+
  theme_bw()+
  theme(axis.title.x=element_blank())

p2 <- ggplot(dt_exp, aes(x=type, y=NES, fill=type)) +
  scale_fill_manual(values=c("white", "#FC8D62"))+
  stat_compare_means(aes(x=type, y=NES))+
  geom_violin(width=0.8, alpha=0.2)+
  geom_boxplot(width=0.2) +
  guides(fill=F)+
  labs(title="EXP")+
  theme_bw()+
  theme(axis.title.x=element_blank())

p1
p2
# plot_res <- plot_grid(p1,p2)
# save_plot("~/Immunoediting/results/compare_nes_pancaner.pdf", plot_res, nrow = 1, ncol = 2,
#           base_asp = 1.3)
```

From the pancancer overview, It has significant difference between neutral simulation and TCGA real data.

Then we can check how many patients have significant neoantigen negative enrichment signal.

```{r p_distribution,eval=TRUE,message=FALSE}
###firstly, we adjust p value to FDR q values:
pancancer_nes_ccf$fdr <- p.adjust(pancancer_nes_ccf$p_value,method = "fdr")
pancancer_nes_exp$fdr <- p.adjust(pancancer_nes_exp$p_value,method = "fdr")

p1 <- plot_pie(pancancer_nes_ccf$fdr,expression = "pancancer_nes_ccf$fdr<0.25 & pancancer_nes_ccf$nes<0",label = c("samples with FDR<25% \n and NES(CCF) < 0","others"))
p2 <- plot_pie(pancancer_nes_exp$fdr,expression = "pancancer_nes_exp$fdr<0.25 & pancancer_nes_exp$nes<0",label = c("samples with FDR<25% \n and NES(EXP) < 0","others")
)
###FDR 5%
p3 <- plot_pie(pancancer_nes_ccf$fdr,expression = "pancancer_nes_ccf$fdr<0.05 & pancancer_nes_ccf$nes<0",label = c("samples with FDR<5% \n and NES(CCF) < 0","others"))
p4 <- plot_pie(pancancer_nes_exp$fdr,expression = "pancancer_nes_exp$fdr<0.05 & pancancer_nes_exp$nes<0",label = c("samples with FDR<5% \n and NES(EXP) < 0","others"))
```

Thus we conclude that the immunoediting signal is undoubtedly exist, however in majority of cancer patient, this signal is undetectable, and only 2.5% (CCF FDR<0.25) of patient show this significant immunoediting signal.

Then we view the cancer type distribution of NES and compare with cancer type simulation:
```{r view_cancer_type_nes,eval=TRUE}
pancancer_simulation <- readRDS("~/Immunoediting/data/simulation/pancancer_simulation.rds")
pancancer_nes_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds")
pancancer_nes_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds") 

##load function
source("~/Immunoediting/code/R/plot_pancancer_nes_distribution.R")
pancancer_simulation$cancer <- gsub("TCGA-","",pancancer_simulation$cancer)
pancancer_nes_ccf$cancer <- gsub("TCGA-","",getcancer_type(pancancer_nes_ccf$sample))
pancancer_nes_exp$cancer <- gsub("TCGA-","",getcancer_type(pancancer_nes_exp$sample))
p1 <- plot_pancancer(pancancer_nes_ccf,pancancer_simulation,text_position = 4,size=3)
p2 <- plot_pancancer(pancancer_nes_exp,pancancer_simulation,text_position = -6,size=3)

# plot_res <- plot_grid(p1,p2,nrow = 2)
# save_plot("~/Immunoediting/results/compare_nes_cancer_type.pdf", plot_res, nrow = 2, ncol = 1,
#           base_height = 15,base_width = 15,limitsize = FALSE)
p1
p2
```



## Neoantigen enrichment score is associated with immune cell infiltration 
NES reflects neoantigen-mediated immune selection, thus we want to check relation between the immune cell infiltration level and our NES score.

```{r immune_cor,eval=TRUE}
pancancer_nes_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds")
pancancer_nes_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds")

###load all immune data
cibersort_immune <- readRDS("~/Immunoediting/data/immue_estimate.rds")
lf <- readRDS("~/Immunoediting/data/immune_landscape.rds") %>% 
  select(`TCGA Participant Barcode`,`Leukocyte Fraction`,
         `Stromal Fraction`,`Intratumor Heterogeneity`) %>% 
  rename(sample=`TCGA Participant Barcode`)
elife <- readRDS("~/Immunoediting/data/wang_2019_elife.rds")
cyt <- readRDS("~/Immunoediting/data/pancancer_cyt.rds")
```

Firstly, we calculate correlation using spearman method by cancer types

```{r cal_cor_cancertype,eval=TRUE,warning = FALSE}
##bind all data
all_immune <- left_join(elife %>% 
                          rename(sample=Tumor_Sample_Barcode),
                        cibersort_immune,
                        by="sample") %>% 
  left_join(.,cyt %>% 
              mutate(sample=substr(sample,1,15)),
            by="sample") %>% 
  mutate(sample=substr(sample,1,12)) %>% 
  left_join(.,lf,by="sample") %>% 
  distinct(sample,.keep_all = T)

pancancer_nes_ccf <- left_join(pancancer_nes_ccf %>% 
                                 mutate(sample=substr(sample,1,12)),
                               all_immune,by="sample")
pancancer_nes_ccf$cancer <- getcancer_type(pancancer_nes_ccf$sample)
pancancer_nes_ccf <- pancancer_nes_ccf %>%
  select(-c(sample, es, p_value, Project)) %>%
  filter(!is.na(nes)) %>%
  select(cancer, nes, everything())


pancancer_nes_exp <- left_join(pancancer_nes_exp %>% 
                                 mutate(sample=substr(sample,1,12)),
                               all_immune,by="sample")
pancancer_nes_exp$cancer <- getcancer_type(pancancer_nes_exp$sample)
pancancer_nes_exp <- pancancer_nes_exp %>%
  select(-c(sample, es, p_value, Project)) %>%
  filter(!is.na(nes)) %>%
  select(cancer, nes, everything())


summ_ccf <- pancancer_nes_ccf %>% 
  group_by(cancer) %>%
  summarise(across(nes:`Intratumor Heterogeneity`,~median(.x, na.rm = TRUE)))
summ_exp <- pancancer_nes_exp %>% 
  select(-c(3:6)) %>% 
  group_by(cancer) %>%
  summarise(across(nes:`Intratumor Heterogeneity`,~median(.x, na.rm = TRUE)))
```

```{r err,eval=FALSE}
cor_mat1 <- cor(summ_ccf %>% select(-cancer), 
                method = "spearman", use = "pairwise.complete.obs")
cor_mat1 <- cor_mat1[,1]  %>% as.data.frame()
colnames(cor_mat1) <- "NES(CCF)"
cor_mat2 <- cor(summ_exp %>% select(-cancer), 
                method = "spearman", use = "pairwise.complete.obs")
cor_mat2 <- cor_mat2[,1]  %>% as.data.frame()
colnames(cor_mat2) <- "NES(EXP)"

ccf_exp_cor <- bind_cols(cor_mat1,cor_mat2)
ccf_exp_cor <- ccf_exp_cor[-1,]

saveRDS(ccf_exp_cor,file = "~/Immunoediting/data/ccf_exp_immune_cor.rds")
```


```{r plot_ccf_exp_immune_cor,eval=TRUE}
ccf_exp_cor <- readRDS("~/Immunoediting/data/ccf_exp_immune_cor.rds")
p1 <- ggcorrplot::ggcorrplot(t(ccf_exp_cor))

##view data
DT::datatable(ccf_exp_cor,
  options = list(scrollX = TRUE, keys = TRUE), rownames = TRUE
)

# plot_res <- plot_grid(p1)
# save_plot("~/Immunoediting/results/al_immune_nes.pdf", plot_res)
p1
```

We can see the negative corrlation between nes and most immune score.
Plot LF:

```{r plot_LF,eval=TRUE,warning = FALSE,message=FALSE}
##this code is modified from wang et.al 2019 elife 
###https://xsliulab.github.io/tumor-immunogenicity-score/#strong-association-between-aps-and-immune-cell-infiltration-level
library(ggpubr)
plot_scatter <- function(data, x, y, xlab = "Median NES", ylab = "Median LF", conf.int = TRUE, method = "spearman", label.x = -0.7, label.y = 0.1, label = "cancer", ...) {
  ggscatter(data,
    x = x, y = y,
    xlab = xlab, ylab = ylab,
    shape = 21, size = 3, color = "black",
    add = "reg.line", add.params = list(color = "blue", fill = "lightgray"),
    conf.int = conf.int,
    cor.coef = TRUE,
    cor.coeff.args = list(method = method, label.x = label.x, label.y = label.y, label.sep = "\n"),
    label = label, repel = TRUE, ...
  )
}

p1 <- plot_scatter(data = summ_ccf %>% mutate(cancer=gsub("TCGA-","",cancer)),
             x="nes",y="Leukocyte Fraction",label.x = -0.8,label.y = 0.1)
p2 <- plot_scatter(data = summ_exp %>% mutate(cancer=gsub("TCGA-","",cancer)),
             x="nes",y="Leukocyte Fraction",label.x = -0.7,label.y = 0.1)


# plot_res <- plot_grid(p1+ggtitle("CCF"),
#                       p2+ggtitle("EXP"),nrow = 2)
# save_plot("~/Immunoediting/results/compare_LF_nes.pdf", plot_res, nrow = 2, ncol = 1)

p1
p2
```

We can see the CCF NES is significantly corrlated with LF, while EXP NES is not.

The negative NES value indicates that samples are under immune negative selection, thus we then consider NES as categorical variable and check the difference between positive and negative NES.

```{r diff_nes,eval=TRUE,message=FALSE,warning=FALSE}

plot_diff_box <- function(data,var,title,laby){
  ggplot(data, aes(x=type, y=get(var), fill=type)) +
  scale_fill_manual(values=c("white", "#FC8D62"))+
  stat_compare_means()+
  geom_violin(width=0.8, alpha=0.2)+
  geom_boxplot(width=0.2) +
  guides(fill=F)+
  labs(title=title,y=laby)+
  theme_bw()+
  theme(axis.title.x=element_blank())
}

pancancer_nes_ccf <- pancancer_nes_ccf %>% 
  mutate(type=ifelse(nes<0,"negative","positive"))
pancancer_nes_exp <- pancancer_nes_exp %>% 
  mutate(type=ifelse(nes<0,"negative","positive"))

pancancer_nes_ccf$type <- factor(pancancer_nes_ccf$type,levels = c("positive","negative"))
pancancer_nes_exp$type <- factor(pancancer_nes_exp$type,levels = c("positive","negative"))  

###LF
p1 <- plot_diff_box(data = pancancer_nes_ccf,var = "Leukocyte Fraction",title = "CCF",
              laby = "Leukocyte Fraction")
p1
p2 <- plot_diff_box(data = pancancer_nes_exp,var = "Leukocyte Fraction",title = "EXP",
              laby = "Leukocyte Fraction")
p2
###T_cells_CD8
p3 <- plot_diff_box(data = pancancer_nes_ccf,var = "T_cells_CD8",title = "CCF",
              laby = "CD8 T Cells")
p3
p4 <- plot_diff_box(data = pancancer_nes_exp,var = "T_cells_CD8",title = "EXP",
              laby = "CD8 T Cells")
p4
###CYT
p5 <- ggplot(pancancer_nes_ccf, aes(x=type, y=log(cyt+1), fill=type)) +
  scale_fill_manual(values=c("white", "#FC8D62"))+
  stat_compare_means()+
  geom_violin(width=0.8, alpha=0.2)+
  geom_boxplot(width=0.2) +
  guides(fill=F)+
  theme_bw()+
  theme(axis.title.x=element_blank())+
  labs(title="CCF",y="log(CYT+1)")
p5
p6 <- ggplot(pancancer_nes_exp, aes(x=type, y=log(cyt+1), fill=type)) +
  scale_fill_manual(values=c("white", "#FC8D62"))+
  stat_compare_means()+
  geom_violin(width=0.8, alpha=0.2)+
  geom_boxplot(width=0.2) +
  theme_bw()+
  theme(axis.title.x=element_blank())+
  guides(fill=F)+
  labs(title="EXP",y="log(CYT+1)")
p6

# plot_res <- plot_grid(p1,p3,p5,nrow = 3)
# save_plot("~/Immunoediting/results/compare_cat_ccf_immune.pdf", plot_res, nrow = 3)
# plot_res1 <- plot_grid(p2,p4,p6,nrow = 3)
# save_plot("~/Immunoediting/results/compare_cat_exp_immune.pdf", plot_res1, nrow = 3)
```

In the categorical aspect, negative and positive show significant difference in those immune scores.

## Neoantigen enrichment score and immune negative selection 
We use a stochastic branching process model (2020 Nature genetics, Eszter Lakatos et.al) to model tumor growth integrating neoantigen-mediated negative selection. The detailed method please see our paper or the [origin NG paper](https://doi.org/10.1038/s41588-020-0687-1)

The Julia scripts used in this simulation (modified from NG paper) can be found in `code/julia`.

The R scripts used to calculate NES of simulation data can be found in `code/R/cal_simulate_es.R`

We can check the relationship between selection coefficient with NES:

```{r cor_s_nes,eval=FALSE}
###bind all data
dt <- vector("list",51)
for (i in 1:51){
  df <- readRDS(paste("~/neo_dep/ssNEA/data/publish/simulation_tumor_growth/s_01_05/",file[i],sep = ""))
  df$selection_strength <- (i-1)*(-0.01)
  df$fdr <- p.adjust(df$p,method = "fdr")
  dt[[i]] <- df
}
all_s <- bind_rows(dt)

all_s <- all_s %>%
  mutate(`Selection strength`=as.character(selection_strength))
all_s$`Selection strength` <- factor(all_s$`Selection strength`,levels = unique(all_s$`Selection strength`))

all_s <- all_s %>%
  filter(selection_strength >=(-0.25))
saveRDS(all_s,file = "~/Immunoediting/data/all_s.rds")
```
Plot

```{r plot_s_nes,eval=TRUE}
all_s <- readRDS("~/Immunoediting/data/all_s.rds")
p1 <- ggplot(data=all_s,aes(x=`Selection strength`,y=nes))+
  geom_violin(mapping =aes(x=`Selection strength`,y=nes,fill=`Selection strength`))+
  stat_summary(fun=median, geom="point", size=1, color="#F9E91E")+
  theme_classic()+
  guides(fill=F)+
  theme(axis.text.x = element_text(angle = 45, vjust = 1, hjust = 1))

p1
# plot_res <- plot_grid(p1)
# save_plot("~/Immunoediting/results/s_nes.pdf", plot_res)
```
For each fixed selection coefficient, the normalized neoantigen enrichment score was calculated, and the resulting the resulting neoantigen enrichment score show near linear correlation with s values. This analysis suggests neoantigen enrichment score can be used to infer the immune selection strength in patient.

## TCGA Pancancer survial analysis
The enrichment score for CCF distribution reflect the status of immune-elimination step. In cancer patient with strong immune-elimination, good prognosis should be expected. Thus we devided patients into two groups based on NES values, and compare their survial probability.
```{r survial_ccf,eval=TRUE,message=FALSE}
library(survival)
library(survminer)
###ccf
pancancer_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds") %>%
  mutate(p_adj=p.adjust(p_value,method = "fdr")) %>%
  mutate(es_type=case_when(
    nes < 0 & p_adj < 0.25 ~ "sig_negative",
    nes < 0 & p_adj >= 0.25 ~ "negative but not sig",
    nes >= 0 ~ "positive"
  ))
pancancer_survial <- readRDS("~/Immunoediting/data/pancancer_survial.rds")
pancancer_ccf <- left_join(pancancer_ccf %>%
                                      mutate(sample=substr(sample,1,12)),
                                    pancancer_survial,by="sample") %>% filter(OS.time<3300)

##plot
fit0 <- survfit(Surv(OS.time,OS)~es_type,data = pancancer_ccf %>% filter(es_type!="negative but not sig"))
p1 <- ggsurvplot(fit0,pval = T,risk.table = T,
                 data = pancancer_ccf %>% filter(es_type!="negative but not sig"),
                 tables.theme = theme_cleantable(),  # Clean risk table
                 palette =c( "black","red"),
                 size=2,title="CCF", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))
p1$plot <- p1$plot+
  theme(legend.text = element_text(size = 10, color = "black"))
p1
```

```{r survial_exp,eval=TRUE,message=FALSE}
###exp
pancancer_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds") %>%
  mutate(p_adj=p.adjust(p_value,method = "fdr")) %>%
  mutate(es_type=case_when(
    nes < 0 & p_adj < 0.25 ~ "sig_negative",
    nes < 0 & p_adj >= 0.25 ~ "negative but not sig",
    nes >= 0 ~ "positive"
  ))
pancancer_exp <- left_join(pancancer_exp %>%
                                      mutate(sample=substr(sample,1,12)),
                                    pancancer_survial,by="sample") %>% filter(OS.time<3300)

##plot
fit1 <- survfit(Surv(OS.time,OS)~es_type,data = pancancer_exp %>% filter(es_type!="negative but not sig"))
p2 <- ggsurvplot(fit1,pval = T,risk.table = T,data = pancancer_exp %>% filter(es_type!="negative but not sig"),
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c( "black","red"),
                  size=2,title="EXP", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))
p2$plot <- p2$plot+
  theme(legend.text = element_text(size = 10, color = "black"))
p2
```
We can see patients with ESccf values (NES<0; FDR<0.25) show improved prognosis compared to remain patients, while expression not.

The patients with NES<0 and FDR<0.25 is small cohort (209 patients), so we could ask this difference is actually caused by NES(ccf) difference or by cancer type difference? (ie. those 209 patients are mainly several cancertype, and those cancer type has better survial than others). Then we do cox multivariate analysis that take cancer type and immune infiltration into account:

```{r cox,eval=TRUE,message=FALSE,warning=FALSE}
library(ezcox)
immue_estimate <- readRDS("~/Immunoediting/data/immue_estimate.rds")
pancancer_ccf$`cancer type` <- getcancer_type(pancancer_ccf$sample)

pancancer_ccf <- left_join(pancancer_ccf,
                           immue_estimate %>% mutate(sample=substr(sample,1,12)),
                           by="sample") %>% distinct(sample,.keep_all = T)
p0 <- show_forest(pancancer_ccf,covariates = c("nes"),controls = c("cancer type","T_cells_CD8"),
            time = "OS.time",status = "OS",vars_to_show="nes",point_size = 8)
p0
###consider only sig-negative+positive
pancancer_ccf1 <- pancancer_ccf %>% filter(es_type!="negative but not sig")
p00 <- show_forest(pancancer_ccf1,covariates = c("nes"),controls = c("cancer type","T_cells_CD8"),
            time = "OS.time",status = "OS",vars_to_show="nes",point_size = 8)
p00

# plot_res <- plot_grid(p0,p00,nrow = 2)
# save_plot("~/Immunoediting/results/cox_multivariate_ccf.pdf", plot_res, nrow = 2,base_asp = 2)
```
So, when we consider cancer type and immune infiltration, the HR is still more than 1 significantly.

In the `Neoantigen ebrichment score is associated with immune cell infiltration` part, we show that NES(ccf) is corrlated with some immune scores. So we want to explore whether prognosis  is different between immune hot and cold tumors or not.

We foucs on CD8 T cells. Let's look at the distribution of CD8 T cells between cancer types:
```{r dis_cd8,eval=TRUE}
immue_estimate$cancer <- gsub("TCGA-","",getcancer_type(immue_estimate$sample))
immue_estimate %>%
  group_by(cancer) %>%
  summarise(median_immune=median(T_cells_CD8)) %>%
  arrange(median_immune) -> immune_pancancer
immue_estimate$cancer <- factor(immue_estimate$cancer,levels = immune_pancancer$cancer)

p000 <- ggboxplot(data = immue_estimate,
    x = "cancer", y = "T_cells_CD8", color = "cancer", add = "jitter", xlab = "TCGA Projects",
    ylab = "CD8 T cells", add.params = list(size = 0.6),
    legend = "none"
  ) +
  rotate_x_text(angle = 45)
p000
# plot_res <- plot_grid(p000)
# save_plot("~/Immunoediting/results/pancancer_immune.pdf", plot_res,base_asp = 2)

hot_cancer <- immune_pancancer$cancer[18:33] 
cold_cancer <- immune_pancancer$cancer[1:16]
dt <- data.frame(hot_cancer=hot_cancer,cold_cancer=cold_cancer)
DT::datatable(dt,
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)

```

Then we do survial analysis for hot tumor and cold tumor.

```{r survial_hot_cold,eval=TRUE,message=FALSE,warning=FALSE}
###ccf
pancancer_ccf <- readRDS("~/Immunoediting/data/pancancer_nes_ccf.rds") %>%
  mutate(p_adj=p.adjust(p_value,method = "fdr")) %>%
  mutate(es_type=case_when(
    nes < 0 & p_adj < 0.25 ~ "sig_negative",
    nes < 0 & p_adj >= 0.25 ~ "negative but not sig",
    nes >= 0 ~ "positive"
  ))
pancancer_survial <- readRDS("~/Immunoediting/data/pancancer_survial.rds")
pancancer_ccf <- left_join(pancancer_ccf %>%
                                      mutate(sample=substr(sample,1,12)),
                                    pancancer_survial,by="sample") %>% filter(OS.time<3300)
pancancer_ccf$`cancer type` <- getcancer_type(pancancer_ccf$sample)

hot <- pancancer_ccf %>% filter(gsub("TCGA-","",`cancer type`) %in% hot_cancer)
fit2 <- survfit(Surv(OS.time,OS)~es_type,data = hot %>% filter(es_type!="negative but not sig"))
p3 <- ggsurvplot(fit2,pval = T,risk.table = T,data = hot %>% filter(es_type!="negative but not sig"),
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c( "black","red"),
                  size=2,title="CCF hot", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))
p3$plot <- p3$plot+
  theme(legend.text = element_text(size = 10, color = "black"))

cold <- pancancer_ccf %>% filter(gsub("TCGA-","",`cancer type`) %in% cold_cancer)
fit3 <- survfit(Surv(OS.time,OS)~es_type,data = cold %>% filter(es_type!="negative but not sig"))
p4 <- ggsurvplot(fit3,pval = T,risk.table = T,
                 data = cold %>% filter(es_type!="negative but not sig"),
                 tables.theme = theme_cleantable(),  # Clean risk table
                 palette =c( "black","red"),
                 size=2,title="CCF cold", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))
p4$plot <- p4$plot+
  theme(legend.text = element_text(size = 10, color = "black"))
p3
p4
#plot_res <- arrange_ggsurvplots(list(p1,p3,p4),nrow = 2,2)

#save_plot("~/Immunoediting/results/survial_ccf.pdf", plot_res, nrow = 2,ncol = 2,base_width = 12,base_height = 8)

###exp
pancancer_exp$cancer <- gsub("TCGA-","",getcancer_type(pancancer_exp$sample))
hot1 <- pancancer_exp %>% filter(cancer %in% hot_cancer)
fit4 <- survfit(Surv(OS.time,OS)~es_type,data = hot1 %>% filter(es_type!="negative but not sig"))
p5 <- ggsurvplot(fit4,pval = T,risk.table = T,data = hot1 %>% filter(es_type!="negative but not sig"),
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c( "black","red"),
                  size=2,title="EXP hot", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))

cold1 <- pancancer_exp %>% filter(cancer %in% cold_cancer)
fit5 <- survfit(Surv(OS.time,OS)~es_type,data = cold1 %>% filter(es_type!="negative but not sig"))
p6 <- ggsurvplot(fit5,pval = T,risk.table = T,data = cold1 %>% filter(es_type!="negative but not sig"),
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c( "black","red"),
                  size=2,title="EXP cold", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NES > 0", "NES < 0 & FDR < 25% "))
p5
p6

#plot_res <- arrange_ggsurvplots(list(p2,p5,p6),nrow = 2,2)

#save_plot("~/Immunoediting/results/survial_exp.pdf", plot_res, nrow = 2,ncol = 2,base_width = 12,base_height = 8)

```
This analysis show that both hot and cold cancer has significantly different prognosis, while both of exp shows no difference. This is because the enrichment score for mRNA expression distribution partially reflect the status of immune-escape, thus we explore other escape mechanisms, including:

- Suppress the transcription/expression of genome alterations encoding high antigenicity (quantified by NES(EXP))
- Antigen presentation pathway gene mutation
- Up-regulate the expression of immune suppressive signaling, like PD-L1, CTLA-4
 
In the `Process immune escape sample` part, we has already find escape samples. Then we can look at the prognosis of escape samples VS no escape samples:

```{r survial_escape,eval=TRUE}
pancancer_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds") %>%
  mutate(p_adj=p.adjust(p_value,method = "fdr")) %>%
  mutate(es_type=case_when(
    nes < 0 & p_adj < 0.25 ~ "sig_negative",
    nes < 0 & p_adj >= 0.25 ~ "negative but not sig",
    nes >= 0 ~ "positive"
  ))
pancancer_exp <- left_join(pancancer_exp %>%
                                      mutate(sample=substr(sample,1,12)),
                                    pancancer_survial,by="sample") %>% 
  filter(OS.time<3300)


pancancer_exp$escape <- mapply(function(x,y,z)
  {case_when(
    NA %in% c(x,y,z) & any(c(x,y,z)=="yes",na.rm = T) ~ "yes",
    NA %in% c(x,y,z) & all(c(x,y,z)=="no",na.rm = T) ~ "NA",
    !(NA %in% c(x,y,z)) & any(c(x,y,z)=="yes",na.rm = T) ~ "yes",
    !(NA %in% c(x,y,z)) & all(c(x,y,z)=="no",na.rm = T) ~ "no"
  )},
  pancancer_exp$apm_mut,
  pancancer_exp$checkpoint_overexpression,
  pancancer_exp$exp_escape)
pancancer_exp <- pancancer_exp %>% 
  filter(escape!="NA")
#saveRDS(pancancer_exp,file = "~/Immunoediting/data/tmp_exp.rds")
```

View the escape samples:

```{r view_escape,eval=TRUE}
a <- table(pancancer_exp$escape) %>% enframe() %>% 
  rename(escape_status=name,sample_counts=value)
DT::datatable(a,
  options = list(scrollX = TRUE, keys = TRUE), rownames = FALSE
)
```

Do survial analysis:
```{r escape_survial,eval=TRUE}
pancancer_exp$cancer <- gsub("TCGA-","",getcancer_type(pancancer_exp$sample))
fit6 <- survfit(Surv(OS.time,OS)~escape,data = pancancer_exp)
p7 <- ggsurvplot(fit6,pval = T,risk.table = T,data = pancancer_exp,
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c("black","red"),
                  size=2,title="Escape of all samples", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NO escape", "Escape"))

hot <- pancancer_exp %>% filter(cancer %in% hot_cancer)
fit7 <- survfit(Surv(OS.time,OS)~escape,data = hot)
p8 <- ggsurvplot(fit7,pval = T,risk.table = T,data = hot,
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c("black","red"),
                  size=2,title="Escape of hot cancers", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NO escape", "Escape"))

cold <- pancancer_exp %>% filter(cancer %in% cold_cancer)
fit8 <- survfit(Surv(OS.time,OS)~escape,data = cold)
p9 <- ggsurvplot(fit8,pval = T,risk.table = T,data = cold,
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c("black","red"),
                  size=2,title="Escape of cold cancers", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("NO escape", "Escape"))

p7
p8
p9

#plot_res <- arrange_ggsurvplots(list(p7,p8,p9),nrow = 2)

#save_plot("~/Immunoediting/results/survial_escape.pdf", plot_res, nrow = 2,ncol = 2,base_width = 12,base_height = 8)
```
 
We can see, in the hot cancers, no escape samples have a better prognosis while in the cold cancers, there are no difference.

We further investigate whether immune hot and no escape samples have a better survial:

```{r two,eval=TRUE}
pancancer_exp <- readRDS("~/Immunoediting/data/pancancer_nes_exp.rds") %>%
  mutate(p_adj=p.adjust(p_value,method = "fdr")) %>%
  mutate(es_type=case_when(
    nes < 0 & p_adj < 0.25 ~ "sig_negative",
    nes < 0 & p_adj >= 0.25 ~ "negative but not sig",
    nes >= 0 ~ "positive"
  ))
pancancer_exp <- left_join(pancancer_exp %>%
                                      mutate(sample=substr(sample,1,12)),
                                    pancancer_survial,by="sample") %>% 
  left_join(.,immue_estimate %>% 
              mutate(sample=substr(sample,1,12)),
            by="sample") %>% 
  filter(OS.time<3300) %>% filter(!is.na(T_cells_CD8))

pancancer_exp$escape <- mapply(function(x,y,z)
  {case_when(
    NA %in% c(x,y,z) & any(c(x,y,z)=="yes",na.rm = T) ~ "yes",
    NA %in% c(x,y,z) & all(c(x,y,z)=="no",na.rm = T) ~ "NA",
    !(NA %in% c(x,y,z)) & any(c(x,y,z)=="yes",na.rm = T) ~ "yes",
    !(NA %in% c(x,y,z)) & all(c(x,y,z)=="no",na.rm = T) ~ "no"
  )},
  pancancer_exp$apm_mut,
  pancancer_exp$checkpoint_overexpression,
  pancancer_exp$exp_escape)
pancancer_exp <- pancancer_exp %>% 
  filter(escape!="NA")

pancancer_exp <- pancancer_exp %>% 
  mutate(immune_type=ifelse(T_cells_CD8<median(T_cells_CD8),"low","high")) %>% 
  mutate(type=ifelse(immune_type=="high" & escape=="no","immune hot and no escape","others"))

fit9 <- survfit(Surv(OS.time,OS)~type,data = pancancer_exp)
p10 <- ggsurvplot(fit9,pval = T,risk.table = T,data = pancancer_exp,
                  tables.theme = theme_cleantable(),  # Clean risk table
                  palette =c("red","black"),
                  size=2,title="Escape of cold cancers", fun = "pct",
                 legend = c(0.8, 0.9), legend.title = element_blank(),
                 legend.labs = c("immune hot and no escape ", "others"))
#plot_res <- arrange_ggsurvplots(list(p10,p10,p10),nrow = 2)

#save_plot("~/Immunoediting/results/survial_no_escape_immune_hot.pdf", plot_res, nrow = 2,ncol = 2,base_width = 12,base_height = 8)
p10
```
